// agnosticui-core@2.0.0-alpha.12/dist/components/Toast/core/_Toast.js
import { LitElement as d, css as c, nothing as u, html as m } from "lit";
import { n as s } from "../../../property-CemaeiRl.js";
import { r as f } from "../../../state-CovhUvdr.js";
import "../../Alert/core/Alert.js";
import "../../shared/CloseButton/CloseButton.js";
var v = Object.defineProperty, e = (l, o, i, n) => {
  for (var a = void 0, r = l.length - 1, h; r >= 0; r--)
    (h = l[r]) && (a = h(o, i, a) || a);
  return a && v(o, i, a), a;
};
const p = class p extends d {
  constructor() {
    super(), this._isHovered = !1, this._autoDismissTimer = null, this._timerStartTime = null, this._remainingTime = null, this._handleCloseButtonClick = () => {
      this._clearTimer(), this.open = !1;
    }, this._handleMouseEnter = () => {
      this.pauseOnHover && this.autoDismiss && (this._isHovered = !0, this._pauseTimer());
    }, this._handleMouseLeave = () => {
      this.pauseOnHover && this.autoDismiss && (this._isHovered = !1, this._resumeTimer());
    }, this._handleKeydown = (o) => {
      o.key === "Escape" && this.showCloseButton && (o.preventDefault(), this._handleCloseButtonClick());
    }, this.open = !1, this.type = "default", this.position = "top-end", this.duration = 5e3, this.autoDismiss = !0, this.showCloseButton = !0, this.pauseOnHover = !0, this.bordered = !1, this.rounded = !0, this.borderedLeft = !1;
  }
  connectedCallback() {
    super.connectedCallback();
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this._clearTimer();
  }
  willUpdate(o) {
    if (o.has("open")) {
      const i = o.get("open");
      if (this.open && !i) {
        const n = new CustomEvent("toast-open", {
          bubbles: !0,
          composed: !0
        });
        this.dispatchEvent(n), this.onToastOpen?.(n), this.autoDismiss && this.duration > 0 && this._startTimer();
      } else if (!this.open && i) {
        this._clearTimer();
        const n = new CustomEvent("toast-close", {
          bubbles: !0,
          composed: !0
        });
        this.dispatchEvent(n), this.onToastClose?.(n);
      }
    }
  }
  _startTimer() {
    this._clearTimer(), this._timerStartTime = Date.now(), this._remainingTime = this.duration, this._autoDismissTimer = window.setTimeout(() => {
      this._handleAutoDismiss();
    }, this.duration);
  }
  _pauseTimer() {
    if (!this._autoDismissTimer || !this._timerStartTime) return;
    const o = Date.now() - this._timerStartTime;
    this._remainingTime = Math.max(0, this.duration - o), this._clearTimer();
  }
  _resumeTimer() {
    !this.autoDismiss || this._remainingTime === null || (this._timerStartTime = Date.now(), this._autoDismissTimer = window.setTimeout(() => {
      this._handleAutoDismiss();
    }, this._remainingTime));
  }
  _clearTimer() {
    this._autoDismissTimer && (window.clearTimeout(this._autoDismissTimer), this._autoDismissTimer = null, this._timerStartTime = null);
  }
  _handleAutoDismiss() {
    const o = new CustomEvent("toast-dismiss", {
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(o), this.onToastDismiss?.(o), this.open = !1;
  }
  _isUrgentType() {
    return this.type === "error" || this.type === "danger" || this.type === "warning";
  }
  render() {
    const o = this._isUrgentType() ? "alert" : "status", i = this._isUrgentType() ? "assertive" : "polite";
    return m`
      <div
        class="toast-container"
        part="ag-toast"
        role="${o}"
        aria-live="${i}"
        aria-atomic="true"
        @mouseenter=${this._handleMouseEnter}
        @mouseleave=${this._handleMouseLeave}
        @keydown=${this._handleKeydown}
      >
        <ag-alert
          part="ag-toast-content"
          .variant=${this.type}
          .bordered=${this.bordered}
          .rounded=${this.rounded}
          .borderedLeft=${this.borderedLeft}
        >
          <div class="toast-inner-layout">
            <div class="toast-content">
              <slot></slot>
            </div>
            ${this.showCloseButton ? m`
              <ag-close-button
                size="sm"
                .variant=${this.type}
                label="Dismiss notification"
                @close-button-click=${this._handleCloseButtonClick}
              ></ag-close-button>
            ` : u}
          </div>
        </ag-alert>
      </div>
    `;
  }
};
p.styles = c`
    :host {
      display: block;
      visibility: hidden;
      position: fixed;
      z-index: var(--ag-z-index-toast, 1000);
      pointer-events: none;
    }

    :host([open]) {
      visibility: visible;
      pointer-events: auto;
    }

    /* Edge positions - full width/height */
    :host([position="top"]) {
      top: var(--ag-space-4);
      inset-inline: 0;
      width: 100%;
    }

    :host([position="bottom"]) {
      bottom: var(--ag-space-4);
      inset-inline: 0;
      width: 100%;
    }

    :host([position="start"]) {
      top: 0;
      inset-inline-start: var(--ag-space-4);
      bottom: 0;
      height: 100%;
    }

    :host([position="end"]) {
      top: 0;
      inset-inline-end: var(--ag-space-4);
      bottom: 0;
      height: 100%;
    }

    /* Corner positions - constrained size */
    :host([position="top-start"]) {
      top: var(--ag-space-4);
      inset-inline-start: var(--ag-space-4);
    }

    :host([position="top-end"]) {
      top: var(--ag-space-4);
      inset-inline-end: var(--ag-space-4);
    }

    :host([position="bottom-start"]) {
      bottom: var(--ag-space-4);
      inset-inline-start: var(--ag-space-4);
    }

    :host([position="bottom-end"]) {
      bottom: var(--ag-space-4);
      inset-inline-end: var(--ag-space-4);
    }

    /* Toast container with animations */
    .toast-container {
      position: relative;
      opacity: 0;
      transform: translateY(-100%);
      transition: opacity var(--ag-motion-fast) ease-out,
                  transform var(--ag-motion-fast) ease-out;
    }

    :host([open]) .toast-container {
      opacity: 1;
      transform: translateY(0);
    }

    /* Transform variations based on position */
    :host([position="bottom"]) .toast-container,
    :host([position="bottom-start"]) .toast-container,
    :host([position="bottom-end"]) .toast-container {
      transform: translateY(100%);
    }

    :host([position="start"]) .toast-container {
      transform: translateX(-100%);
    }

    :host([position="end"]) .toast-container {
      transform: translateX(100%);
    }

    /* RTL Support - flip transform directions for start/end positions */
    :host-context([dir="rtl"]):host([position="start"]) .toast-container {
      transform: translateX(100%);
    }

    :host-context([dir="rtl"]):host([position="end"]) .toast-container {
      transform: translateX(-100%);
    }

    :host([position="bottom"][open]) .toast-container,
    :host([position="bottom-start"][open]) .toast-container,
    :host([position="bottom-end"][open]) .toast-container,
    :host([position="start"][open]) .toast-container,
    :host([position="end"][open]) .toast-container {
      transform: translateX(0) translateY(0);
    }

    /* Size constraints for corner positions */
    :host([position="top-start"]) .toast-container,
    :host([position="top-end"]) .toast-container,
    :host([position="bottom-start"]) .toast-container,
    :host([position="bottom-end"]) .toast-container {
      max-inline-size: 400px;
      max-block-size: 200px;
    }

    /* Inner layout for content and close button */
    .toast-inner-layout {
      display: flex;
      align-items: center;
      gap: var(--ag-space-3);
    }

    .toast-content {
      flex: 1;
      min-width: 0; /* Allows content to shrink and wrap */
      display: flex;
      align-items: center;
      gap: var(--ag-space-2);
    }

    ag-close-button {
      flex-shrink: 0;
    }

    @media (prefers-reduced-motion: reduce) {
      .toast-container {
        transition: opacity var(--ag-motion-fast) ease;
        transform: none !important;
      }

      :host([open]) .toast-container {
        transform: none !important;
      }
    }
  `;
let t = p;
e([
  s({ type: Boolean, reflect: !0 })
], t.prototype, "open");
e([
  s({ type: String })
], t.prototype, "type");
e([
  s({ type: String, reflect: !0 })
], t.prototype, "position");
e([
  s({ type: Number })
], t.prototype, "duration");
e([
  s({ type: Boolean })
], t.prototype, "autoDismiss");
e([
  s({ type: Boolean })
], t.prototype, "showCloseButton");
e([
  s({ type: Boolean })
], t.prototype, "pauseOnHover");
e([
  s({ type: Boolean })
], t.prototype, "bordered");
e([
  s({ type: Boolean })
], t.prototype, "rounded");
e([
  s({ type: Boolean })
], t.prototype, "borderedLeft");
e([
  s({ attribute: !1 })
], t.prototype, "onToastOpen");
e([
  s({ attribute: !1 })
], t.prototype, "onToastClose");
e([
  s({ attribute: !1 })
], t.prototype, "onToastDismiss");
e([
  f()
], t.prototype, "_isHovered");
export {
  t as Toast
};
