// agnosticui-core@2.0.0-alpha.12/dist/components/Toggle/core/_Toggle.js
import { LitElement as b, css as u, html as s } from "lit";
import { n as t } from "../../../property-CemaeiRl.js";
import { o as p } from "../../../if-defined-C8i28hSj.js";
import { createFormControlIds as f, isHorizontalLabel as c, buildAriaDescribedBy as m } from "../../../shared/form-control-utils.js";
import { formControlStyles as y } from "../../../shared/form-control-styles.js";
var v = Object.defineProperty, i = (d, r, a, l) => {
  for (var o = void 0, n = d.length - 1, g; n >= 0; n--)
    (g = d[n]) && (o = g(r, a, o) || o);
  return o && v(r, a, o), o;
};
const h = class h extends b {
  constructor() {
    super(), this._toggleId = "", this._labelId = "", this._helperId = "", this._errorId = "", this.labelPosition = "top", this._handleClick = (a) => {
      this.onClick && this.onClick(a), this._performToggle(a);
    }, this._handleKeydown = (a) => {
      (a.key === " " || a.key === "Enter") && (a.preventDefault(), this._performToggle(a));
    }, this._performToggle = (a) => {
      if (this.disabled || this.readonly) {
        a.preventDefault();
        return;
      }
      this.checked = !this.checked;
      const l = new CustomEvent("toggle-change", {
        detail: {
          checked: this.checked,
          name: this.name,
          value: this.value
        },
        bubbles: !0,
        composed: !0
      });
      this.dispatchEvent(l), this.onToggleChange && this.onToggleChange(l);
    };
    const r = f("toggle");
    this._toggleId = r.inputId, this._labelId = r.labelId, this._helperId = r.helperId, this._errorId = r.errorId, this.label = "", this.labelHidden = !1, this.noLabel = !1, this.checked = !1, this.size = "md", this.variant = "default", this.disabled = !1, this.readonly = !1, this.required = !1, this.invalid = !1, this.errorMessage = "", this.helpText = "", this.name = "", this.value = "";
  }
  firstUpdated() {
    !this.label && !this.noLabel && console.warn("AgToggle: label property is required for accessibility.");
  }
  renderLabel() {
    if (!this.label || this.noLabel) return "";
    const r = [];
    return c(this.labelPosition) ? (r.push("ag-form-control__label--horizontal"), r.push(`ag-form-control__label--${this.labelPosition}`)) : this.labelPosition === "bottom" && r.push(`ag-form-control__label--${this.labelPosition}`), s`
      <label
        id=${this._labelId}
        for=${this._toggleId}
        class="ag-form-control__label ${this.labelHidden ? "ag-form-control__label--hidden" : ""} ${this.required ? "ag-form-control__label--required" : ""} ${r.join(" ")}"
        part="ag-toggle-label"
      >
        ${this.label}
      </label>
    `;
  }
  render() {
    const r = m({
      helperId: this._helperId,
      errorId: this._errorId,
      hasHelper: !!this.helpText && !this.invalid,
      hasError: !!this.invalid && !!this.errorMessage
    }), a = this.helpText && !this.invalid ? s`<div class="ag-form-control__helper" id="${this._helperId}">
          ${this.helpText}
        </div>` : "", l = this.invalid && this.errorMessage ? s`<div class="ag-form-control__error" id="${this._errorId}">
          ${this.errorMessage}
        </div>` : "", o = s`
      <button
        id="${this._toggleId}"
        role="switch"
        aria-checked="${this.checked}"
        aria-labelledby="${p(this.label && !this.noLabel ? this._labelId : void 0)}"
        aria-describedby="${p(r)}"
        ?disabled="${this.disabled}"
        class="ag-toggle ag-toggle--${this.size} ag-toggle--${this.variant}"
        part="ag-toggle-button"
        @click="${this._handleClick}"
        @keydown="${this._handleKeydown}"
      >
        <span class="ag-toggle__track" part="ag-toggle-track">
          <span class="ag-toggle__handle" part="ag-toggle-handle"></span>
        </span>

        <!-- Screen reader state indication -->
        <span class="visually-hidden">
          ${this.checked ? "On" : "Off"}
        </span>
      </button>
    `;
    return c(this.labelPosition) ? s`
        <div class="ag-form-control--horizontal">
          ${this.renderLabel()}
          ${o}
        </div>
        ${a}
        ${l}
      ` : this.labelPosition === "bottom" ? s`
        ${o}
        ${a}
        ${l}
        ${this.renderLabel()}
      ` : s`
      ${this.renderLabel()}
      ${o}
      ${a}
      ${l}
    `;
  }
};
h.styles = [
  y,
  u`
    /* MINIMALIST & THEMEABLE - Styling via --ag-* design tokens */
    :host {
      /* Block for label positioning support */
      display: block;
    }

    button {
      /* Reset browser button styles */
      margin: 0;
      border: none;
      background: transparent;
      cursor: pointer;
      font-family: inherit;
      color: inherit;
      padding: 0;

      /* Flexbox for track/handle layout */
      display: flex;
      align-items: center;
      position: relative;

      /* Touch target accessibility - minimum 44px */
      min-inline-size: 44px;
      min-block-size: 44px;

      /* Focus management */
      border-radius: var(--ag-focus-border-radius);
      transition: all var(--ag-transition-duration) ease;
    }

    button:focus-visible {
      outline: var(--ag-focus-width) solid rgba(var(--ag-focus), 0.5);
      outline-offset: var(--ag-focus-offset);
    }

    button:disabled {
      cursor: not-allowed;
      opacity: 0.6;
      pointer-events: none;
    }

    .ag-toggle__track {
      /* Track container */
      position: relative;
      display: flex;
      align-items: center;
      border-radius: 9999px;
      transition: inherit;

      /* Default size - will be overridden by size variants */
      width: var(--toggle-width, 40px);
      height: var(--toggle-height, 20px);

      /* Functional styling only */
      background: var(--ag-neutral-300);
      border: 1px solid transparent;
    }

    .ag-toggle__handle {
      /* Handle positioning and animation */
      position: absolute;
      top: 50%;
      inset-inline-start: 2px;
      transform: translateY(-50%);
      border-radius: 50%;
      transition: transform var(--ag-transition-duration) ease;

      /* Default handle size */
      width: var(--handle-size, 18px);
      height: var(--handle-size, 18px);

      /* Functional styling only */
      background: var(--ag-white);
      box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
    }

    /* Checked state - handle moves to right */
    :host([checked]) .ag-toggle__handle {
      transform: translateY(-50%) translateX(calc(var(--toggle-width) - var(--handle-size) - 4px));
    }

    /* RTL Support - flip handle movement direction */
    :host-context([dir="rtl"]):host([checked]) .ag-toggle__handle {
      transform: translateY(-50%) translateX(calc(-1 * (var(--toggle-width) - var(--handle-size) - 4px)));
    }

    /* Size variants with 2:1 width:height ratio */
    :host([size="xs"]) {
      --toggle-width: 28px;
      --toggle-height: 14px;
      --handle-size: 12px;
    }

    :host([size="sm"]) {
      --toggle-width: 34px;
      --toggle-height: 17px;
      --handle-size: 15px;
    }

    :host([size="md"]) {
      --toggle-width: 40px;
      --toggle-height: 20px;
      --handle-size: 18px;
    }

    :host([size="lg"]) {
      --toggle-width: 48px;
      --toggle-height: 24px;
      --handle-size: 22px;
    }

    :host([size="xl"]) {
      --toggle-width: 56px;
      --toggle-height: 28px;
      --handle-size: 26px;
    }

    /* Checked state coloring */
    :host([checked]) .ag-toggle__track {
      background: var(--ag-primary);
    }

    :host([checked][variant="success"]) .ag-toggle__track {
      background: var(--ag-success);
    }

    :host([checked][variant="warning"]) .ag-toggle__track {
      background: var(--ag-warning);
    }

    :host([checked][variant="danger"]) .ag-toggle__track {
      background: var(--ag-danger);
    }

    :host([checked][variant="monochrome"]) .ag-toggle__track {
      background: var(--ag-background-primary-inverted);
    }

    :host([checked][variant="monochrome"]) .ag-toggle__handle {
      background: var(--ag-text-primary-inverted);
    }

    /* Readonly state */
    :host([readonly]) button {
      cursor: default;
    }

    /* Screen reader only text */
    .visually-hidden {
      position: absolute !important;
      width: 1px !important;
      height: 1px !important;
      padding: 0 !important;
      margin: -1px !important;
      overflow: hidden !important;
      clip: rect(0, 0, 0, 0) !important;
      white-space: nowrap !important;
      border: 0 !important;
    }
  `
];
let e = h;
i([
  t({ type: String })
], e.prototype, "label");
i([
  t({ type: String, attribute: "label-position" })
], e.prototype, "labelPosition");
i([
  t({ type: Boolean, attribute: "label-hidden" })
], e.prototype, "labelHidden");
i([
  t({ type: Boolean, attribute: "no-label" })
], e.prototype, "noLabel");
i([
  t({ type: Boolean, reflect: !0 })
], e.prototype, "checked");
i([
  t({ type: String, reflect: !0 })
], e.prototype, "size");
i([
  t({ type: String, reflect: !0 })
], e.prototype, "variant");
i([
  t({ type: Boolean, reflect: !0 })
], e.prototype, "disabled");
i([
  t({ type: Boolean, reflect: !0 })
], e.prototype, "readonly");
i([
  t({ type: Boolean, reflect: !0 })
], e.prototype, "required");
i([
  t({ type: Boolean, reflect: !0 })
], e.prototype, "invalid");
i([
  t({ type: String, attribute: "error-message" })
], e.prototype, "errorMessage");
i([
  t({ type: String, attribute: "help-text" })
], e.prototype, "helpText");
i([
  t({ type: String })
], e.prototype, "name");
i([
  t({ type: String })
], e.prototype, "value");
i([
  t({ attribute: !1 })
], e.prototype, "onClick");
i([
  t({ attribute: !1 })
], e.prototype, "onToggleChange");
export {
  e as AgToggle
};
