// agnosticui-core@2.0.0-alpha.12/dist/components/Button/core/_Button.js
import { LitElement as l, css as h, html as c } from "lit";
import { n as r } from "../../../property-CemaeiRl.js";
import { o as s } from "../../../if-defined-C8i28hSj.js";
var b = Object.defineProperty, o = (g, a, e, v) => {
  for (var n = void 0, i = g.length - 1, u; i >= 0; i--)
    (u = g[i]) && (n = u(a, e, n) || n);
  return n && b(a, e, n), n;
};
const d = class d extends l {
  constructor() {
    super(), this.disabled = !1, this.loading = !1, this.toggle = !1, this.pressed = !1, this.bordered = !1, this.ghost = !1, this.link = !1, this.grouped = !1, this.fullWidth = !1, this.type = "button", this.ariaLabel = "", this.variant = "", this.size = "md", this.shape = "";
  }
  _handleClick(a) {
    if (this.onClick && this.onClick(a), this.toggle && !this.disabled && !this.loading && !a.defaultPrevented) {
      this.pressed = !this.pressed;
      const e = new CustomEvent("toggle", {
        detail: { pressed: this.pressed },
        bubbles: !0,
        composed: !0
      });
      this.dispatchEvent(e), this.onToggle && this.onToggle(e);
    }
  }
  _handleFocus(a) {
    this.onFocus && this.onFocus(a);
    const e = new FocusEvent("focus", {
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(e);
  }
  _handleBlur(a) {
    this.onBlur && this.onBlur(a);
    const e = new FocusEvent("blur", {
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(e);
  }
  /**
   * Focus the internal button element
   */
  focus() {
    const a = this.shadowRoot?.querySelector("button");
    a && a.focus();
  }
  /**
   * Blur the internal button element
   */
  blur() {
    const a = this.shadowRoot?.querySelector("button");
    a && a.blur();
  }
  render() {
    const a = this.disabled || this.loading;
    return c`
      <button
        type=${this.type}
        part="ag-button"
        ?disabled=${a}
        aria-disabled=${a ? "true" : "false"}
        aria-busy=${s(this.loading ? "true" : void 0)}
        aria-pressed=${s(this.toggle ? String(this.pressed) : void 0)}
        aria-label=${s(this.ariaLabel || void 0)}
        @click=${this._handleClick}
        @focus=${this._handleFocus}
        @blur=${this._handleBlur}
      >
        <slot></slot>
      </button>
    `;
  }
};
d.styles = h`
    /* MINIMALIST & THEMEABLE - Styling via --ag-* design tokens */
    :host {
      /* Inline-flex for perfect centering while maintaining inline behavior */
      display: inline-flex;
      align-items: center;
      justify-content: center;
    }

    /* Full-width variant */
    :host([full-width]) {
      display: block;
      width: 100%;
    }

    :host([full-width]) button {
      width: 100%;
    }

    button {
      /* Reset browser defaults */
      margin: 0;
      border: 1px solid transparent;
      background: var(--ag-background-tertiary);

      /* Inherit font styling from parent */
      font: inherit;
      color: inherit;

      /* Explicit line-height to prevent extra vertical spacing */
      line-height: 1;

      /* Basic interaction styling */
      cursor: pointer;

      /* Flex layout for perfect content alignment */
      display: flex;
      align-items: center;
      justify-content: center;
      gap: var(--ag-space-1); /* Small gap for icon + text scenarios */

      /* Smooth transitions */
      transition-property: all;
      transition-duration: var(--ag-motion-medium);
    }
    
    /* Secondary inherits font color */
    :host([variant="secondary"]) button {
      background: var(--ag-neutral-400);
      /* We lock the text color at neutral-900 for both light/dark modes */
      color: var(--ag-neutral-900);
    }
    :host([variant="secondary"]) button:hover {
      background: var(--ag-neutral-500);
    }

    /* Primary, success, warning, and danger all have white text and get
      darker on hover */
    :host([variant="primary"]) button {
      background: var(--ag-primary);
      color: var(--ag-white);
    }
    :host([variant="primary"]) button:hover {
      background: var(--ag-primary-dark);
    }
    :host([variant="success"]) button {
      background: var(--ag-success);
      color: var(--ag-white);
    }
    :host([variant="success"]) button:hover {
      background: var(--ag-success-dark);
    }
    :host([variant="warning"]) button {
      background: var(--ag-warning);
      color: var(--ag-neutral-900);
    }
    :host([variant="warning"]) button:hover {
      background: var(--ag-warning-dark);
    }
    :host([variant="danger"]) button {
      background: var(--ag-danger);
      color: var(--ag-white);
    }
    :host([variant="danger"]) button:hover {
      background: var(--ag-danger-dark);
    }

    /* Monochrome variant - adapts to light/dark themes */
    :host([variant="monochrome"]) button {
      background: var(--ag-background-primary-inverted);
      color: var(--ag-text-primary-inverted);
    }
    :host([variant="monochrome"]) button:hover {
      background: var(--ag-background-secondary-inverted);
    }

    /* Size variants */
    :host([size="x-sm"]) button {
      height: var(--ag-space-8);
      min-height: var(--ag-space-8);
      font-size: calc(var(--ag-font-size-base) - 0.375rem);
      padding-inline: var(--ag-space-2);
      padding-block: 0;
    }

    :host([size="sm"]) button {
      height: var(--ag-space-9);
      min-height: var(--ag-space-9);
      font-size: var(--ag-font-size-xs);
      padding-inline: var(--ag-space-3);
      padding-block: 0;
    }

    /* Default size (md) - applies when no size attribute or size="md" */
    button,
    :host([size="md"]) button {
      height: var(--ag-space-10);
      min-height: var(--ag-space-10);
      font-size: var(--ag-font-size-sm);
      padding-inline: var(--ag-space-4);
      padding-block: 0;
    }

    :host([size="lg"]) button {
      height: var(--ag-space-12);
      min-height: var(--ag-space-12);
      font-size: var(--ag-font-size-base);
      padding-inline: var(--ag-space-5);
      padding-block: 0;
    }

    :host([size="xl"]) button {
      height: var(--ag-space-14);
      min-height: var(--ag-space-14);
      font-size: var(--ag-font-size-md);
      padding-inline: var(--ag-space-6);
      padding-block: 0;
    }

    /* Shape variants */
    :host([shape="capsule"]) button {
      border-radius: var(--ag-radius-full);
      padding-inline-start: var(--ag-space-5);
      padding-inline-end: var(--ag-space-5);
    }

    :host([shape="rounded"]) button {
      border-radius: var(--ag-radius-md);
    }

    /* Circle and Square shapes are only intended for single character or icon */
    :host([shape="circle"]) button {
      border-radius: 50%;
      width: var(--ag-space-10);
      height: var(--ag-space-10);
      padding: 0;
      min-height: unset;
    }

    :host([shape="square"]) button {
      border-radius: 0;
      width: var(--ag-space-10);
      height: var(--ag-space-10);
      padding: 0;
      min-height: unset;
    }

    :host([shape="rounded-square"]) button {
      border-radius: var(--ag-radius-md);
      width: var(--ag-space-10);
      height: var(--ag-space-10);
      padding: 0;
      min-height: unset;
    }

    /* Ghost and Link - minimal/invisible button styles */
    :host([ghost]) button,
    :host([link]) button {
      background: transparent;
      border-color: transparent;
      box-shadow: none;
    }

    /* Ghost-specific - minimal padding, respects variant colors */
    :host([ghost]) button {
      padding-inline-start: var(--ag-space-2);
      padding-inline-end: var(--ag-space-2);
    }

    :host([ghost][variant="primary"]) button {
      color: var(--ag-primary);
    }

    :host([ghost][variant="success"]) button {
      color: var(--ag-success);
    }

    :host([ghost][variant="warning"]) button {
      color: var(--ag-warning);
    }

    :host([ghost][variant="danger"]) button {
      color: var(--ag-danger);
    }


    :host([link][variant="danger"]) button:hover,
    :host([link][variant="warning"]) button:hover,
    :host([link][variant="success"]) button:hover,
    :host([link][variant="primary"]) button:hover {
      background: transparent;
    }

    /* Only for ghosted buttons with text that of:
          - "red" danger
          - "yellow" warning
          - "green" success
          - "blue" primary
      do we use an almost white nuetral 100 on hover.
      The default and secondary are not changed here!
    */
    :host([ghost][variant="danger"]) button:hover,
    :host([ghost][variant="warning"]) button:hover,
    :host([ghost][variant="success"]) button:hover,
    :host([ghost][variant="primary"]) button:hover {
      background: var(--ag-neutral-100);
    }

    /* Link-specific - inline style with smooth underline transition */
    :host([link]) button {
      color: var(--ag-primary-text);
      text-decoration: none;
      border-bottom: 1px solid transparent;
      padding: 0;
      min-height: unset;
    }

    :host([link]) button:hover {
      border-bottom-color: var(--ag-primary);
    }

    /* Grouped buttons - for use in button groups */
    :host([grouped]) button {
      border-radius: var(--ag-radius-md);
    }

    :host([grouped]:not(:last-child)) button {
      border-start-end-radius: 0;
      border-end-end-radius: 0;
      margin-inline-end: -1px;
    }

    :host([grouped]:not(:first-child)) button {
      border-start-start-radius: 0;
      border-end-start-radius: 0;
    }

    /* Bordered variant - outline style */
    :host([bordered]) button {
      background: transparent;
      border-color: var(--ag-neutral-500);
    }

    :host([bordered][variant="primary"]) button {
      color: var(--ag-primary-text);
      border-color: var(--ag-primary-text);
    }

    /* Bordered Buttons on Hover have a filled background with white text.  The
      Default Bordered is an exception to this rule. */
    :host([bordered][variant="primary"]) button:hover {
      background: var(--ag-primary);
      color: var(--ag-white);
    }

    :host([bordered][variant="secondary"]) button,
    :host([bordered][variant="secondary"]) button:hover {
      color: var(--ag-text-secondary);
      border-color: var(--ag-text-secondary);
    }

    :host([bordered][variant="success"]) button {
      color: var(--ag-success-text);
      border-color: var(--ag-success-text);
    }

    :host([bordered][variant="success"]) button:hover {
      color: var(--ag-white);
      background: var(--ag-success);
    }

    :host([bordered][variant="warning"]) button {
      color: var(--ag-warning-text);
      border-color: var(--ag-warning-text);
    }

    :host([bordered][variant="warning"]) button:hover {
      background: var(--ag-warning);
      color: var(--ag-neutral-900);
    }

    :host([bordered][variant="danger"]) button {
      color: var(--ag-danger-text);
      border-color: var(--ag-danger-text);
    }

    :host([bordered][variant="danger"]) button:hover {
      background: var(--ag-danger);
      color: var(--ag-white);
    }

    :host([bordered][variant="monochrome"]) button {
      color: var(--ag-text-primary);
      border-color: var(--ag-text-primary);
    }

    :host([bordered][variant="monochrome"]) button:hover {
      background: var(--ag-background-primary-inverted);
      color: var(--ag-text-primary-inverted);
    }

    :host([ghost][variant="monochrome"]) button {
      color: var(--ag-text-primary);
    }

    :host([ghost][variant="monochrome"]) button:hover {
      background: var(--ag-background-secondary);
    }

    /* Disabled states - per-variant to maintain tonal identity */

    /* Filled buttons - use lighter background tokens */
    :host([variant="primary"]) button:disabled {
      background: var(--ag-primary-background);
      color: var(--ag-primary-text);
    }

    :host([variant="success"]) button:disabled {
      background: var(--ag-success-background);
      color: var(--ag-success-text);
    }

    :host([variant="warning"]) button:disabled {
      background: var(--ag-warning-background);
      color: var(--ag-warning-text);
    }

    :host([variant="danger"]) button:disabled {
      background: var(--ag-danger-background);
      color: var(--ag-danger-text);
    }

    :host([variant="secondary"]) button:disabled,
    button:disabled {
      background: var(--ag-background-disabled);
    }

    :host([variant="monochrome"]) button:disabled {
      background: var(--ag-background-tertiary);
    }

    /* Bordered buttons - dim the border and text with opacity */
    :host([bordered]) button:disabled {
      opacity: 60%;
      background: transparent;
    }

    /* Ghost and link buttons - dim with opacity */
    :host([ghost]) button:disabled,
    :host([link]) button:disabled {
      opacity: 60%;
      background: transparent;
    }

    /* All disabled buttons get not-allowed cursor */
    button:disabled {
      cursor: not-allowed;
    }

    button:focus-visible,
    button:focus:not(:hover) {
      /* Softer focus ring using alpha channel */
      outline: var(--ag-focus-width) solid rgba(var(--ag-focus), 0.5);
      outline-offset: var(--ag-focus-offset);
      transition: outline var(--ag-motion-medium) ease;
    }
  `;
let t = d;
o([
  r({ type: String, reflect: !0 })
], t.prototype, "variant");
o([
  r({ type: String, reflect: !0 })
], t.prototype, "size");
o([
  r({ type: String, reflect: !0 })
], t.prototype, "shape");
o([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "bordered");
o([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "ghost");
o([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "link");
o([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "grouped");
o([
  r({ type: Boolean, reflect: !0, attribute: "full-width" })
], t.prototype, "fullWidth");
o([
  r({ type: String })
], t.prototype, "type");
o([
  r({ type: Boolean })
], t.prototype, "disabled");
o([
  r({ type: Boolean })
], t.prototype, "loading");
o([
  r({ type: Boolean })
], t.prototype, "toggle");
o([
  r({ type: Boolean })
], t.prototype, "pressed");
o([
  r({ type: String, reflect: !0, attribute: "aria-label" })
], t.prototype, "ariaLabel");
o([
  r({ attribute: !1 })
], t.prototype, "onClick");
o([
  r({ attribute: !1 })
], t.prototype, "onFocus");
o([
  r({ attribute: !1 })
], t.prototype, "onBlur");
o([
  r({ attribute: !1 })
], t.prototype, "onToggle");
export {
  t as AgButton
};
