// agnosticui-core@2.0.0-alpha.12/dist/components/Fieldset/vue/VueFieldset.js
import { defineComponent as n, ref as t, onMounted as o, createElementBlock as l, openBlock as r, mergeProps as a, renderSlot as i } from "vue";
import "../core/Fieldset.js";
import { _ as s } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const u = n({
  name: "VueFieldset",
  props: {
    legend: {
      type: String,
      default: ""
    },
    bordered: {
      type: Boolean,
      default: !1
    },
    layout: {
      type: String,
      default: "vertical"
    },
    legendHidden: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const d = t(null);
    return o(async () => {
      await customElements.whenDefined("ag-fieldset");
    }), {
      agComponent: d,
      legend: e.legend,
      bordered: e.bordered,
      layout: e.layout,
      legendHidden: e.legendHidden
    };
  }
}), f = ["legend", "bordered", "layout", "legend-hidden"];
function g(e, d, p, m, c, y) {
  return r(), l("ag-fieldset", a({
    ref: "agComponent",
    legend: e.legend,
    bordered: e.bordered || void 0,
    layout: e.layout,
    "legend-hidden": e.legendHidden || void 0
  }, e.$attrs), [
    i(e.$slots, "default")
  ], 16, f);
}
const _ = /* @__PURE__ */ s(u, [["render", g]]);
export {
  _ as default
};
