// agnosticui-core@2.0.0-alpha.12/dist/components/Tabs/core/_Tabs.js
import { LitElement as h, css as u, html as p } from "lit";
import { n as d } from "../../../property-CemaeiRl.js";
import { r as f } from "../../../state-CovhUvdr.js";
import { generateUniqueId as m } from "../../../utils/unique-id.js";
var A = Object.defineProperty, l = (y, t, a, i) => {
  for (var e = void 0, s = y.length - 1, r; s >= 0; s--)
    (r = y[s]) && (e = r(t, a, e) || e);
  return e && A(t, a, e), e;
};
const v = class v extends h {
  constructor() {
    super(), this.panel = "", this.disabled = !1;
  }
  connectedCallback() {
    super.connectedCallback(), this.setAttribute("role", "tab");
  }
  render() {
    return p`<div class="tab" part="ag-tab"><slot></slot></div>`;
  }
};
v.styles = u`
    :host {
      display: inline-flex;
    }

    .tab {
      display: inline-flex;
      align-items: center;
      justify-content: center;
      background: transparent;
      border: none;
      color: var(--ag-text-secondary);
      cursor: pointer;
      font: inherit;
      padding: var(--ag-space-2) var(--ag-space-4);
      border-radius: 0;
      border-bottom: 2px solid transparent;
      text-decoration: none;
    }

    :host([aria-selected="true"]) .tab {
      color: var(--ag-text-primary);
      border-bottom-color: var(--ag-primary);
    }

    .tab:focus-visible {
      outline: var(--ag-focus-width) solid rgba(var(--ag-focus), 0.5);
      outline-offset: var(--ag-focus-offset);
    }

    /* Disabled state styles */
    :host([disabled]) .tab,
    :host([aria-disabled="true"]) .tab {
      opacity: var(--ag-opacity-disabled);
      cursor: not-allowed;
      pointer-events: none;
    }

    .tab:hover {
      background: var(--ag-background-secondary);
      color: var(--ag-text-primary);
    }

    :host([disabled]) .tab:hover,
    :host([aria-disabled="true"]) .tab:hover {
      background: transparent;
      color: var(--ag-text-secondary);
    }
  `;
let c = v;
l([
  d({ type: String })
], c.prototype, "panel");
l([
  d({ type: Boolean, reflect: !0 })
], c.prototype, "disabled");
const g = class g extends h {
  constructor() {
    super();
  }
  connectedCallback() {
    super.connectedCallback(), this.setAttribute("role", "tabpanel"), this.setAttribute("tabindex", "0");
  }
  render() {
    return p`<div class="tab-panel" part="ag-tab-panel"><slot></slot></div>`;
  }
};
g.styles = u`
    :host {
      display: block;
    }

    .tab-panel {
      display: block;
      padding: var(--ag-tabs-panel-padding, 1rem);
    }

    :host([hidden]) {
      display: none;
    }
  `;
let T = g;
const _ = class _ extends h {
  constructor() {
    super(), this.activation = "manual", this.activeTab = 0, this.orientation = "horizontal", this.ariaLabel = "", this._tabs = [], this._panels = [], this._focusedTab = 0;
  }
  firstUpdated() {
    Promise.resolve().then(() => {
      this._updateTabsAndPanels();
    });
  }
  updated(t) {
    t.has("activeTab") && (this._focusedTab = this.activeTab, Promise.resolve().then(() => {
      this._updateTabsAndPanels();
    }));
  }
  connectedCallback() {
    super.connectedCallback(), this.addEventListener("slotchange", () => {
      Promise.resolve().then(() => {
        this._updateTabsAndPanels();
      });
    }), this.addEventListener("keydown", this._handleKeyDown.bind(this)), this.addEventListener("click", this._handleClick.bind(this));
  }
  _updateTabsAndPanels() {
    this._tabs = Array.from(this.querySelectorAll("ag-tab")), this._panels = Array.from(this.querySelectorAll("ag-tab-panel")), this._tabs.forEach((t, a) => {
      const i = t.id || `tab-${m()}`, e = t.panel || this._panels[a]?.id || `panel-${m()}`;
      t.setAttribute("id", i), t.setAttribute("aria-controls", e), t.setAttribute("aria-selected", a === this.activeTab ? "true" : "false"), t.setAttribute("tabindex", a === this._focusedTab ? "0" : "-1"), this._panels[a] && (this._panels[a].setAttribute("id", e), this._panels[a].setAttribute("aria-labelledby", i), a !== this.activeTab ? this._panels[a].setAttribute("hidden", "") : this._panels[a].removeAttribute("hidden"));
    });
  }
  _handleKeyDown(t) {
    if (!this._tabs.length) return;
    const a = t.target, i = a && a.tagName === "AG-TAB", e = a && a.closest("ag-tab-panel");
    if (!i || e)
      return;
    const { key: s } = t, r = this.orientation === "horizontal";
    let o = this._focusedTab, b = !1;
    switch (s) {
      case "ArrowRight":
        r && (o = this._findNextEnabledTab(this._focusedTab, 1), b = this.activation === "automatic", t.preventDefault());
        break;
      case "ArrowLeft":
        r && (o = this._findNextEnabledTab(this._focusedTab, -1), b = this.activation === "automatic", t.preventDefault());
        break;
      case "ArrowDown":
        r || (o = this._findNextEnabledTab(this._focusedTab, 1), b = this.activation === "automatic", t.preventDefault());
        break;
      case "ArrowUp":
        r || (o = this._findNextEnabledTab(this._focusedTab, -1), b = this.activation === "automatic", t.preventDefault());
        break;
      case "Home":
        o = this._findFirstEnabledTab(!0), b = this.activation === "automatic", t.preventDefault();
        break;
      case "End":
        o = this._findFirstEnabledTab(!1), b = this.activation === "automatic", t.preventDefault();
        break;
      case " ":
      case "Enter":
        this.activation === "manual" && (this._activateTab(this._focusedTab), t.preventDefault());
        break;
    }
    o !== this._focusedTab && (this._setFocusedTab(o), b && this._activateTab(o));
  }
  _handleClick(t) {
    let a = t.target;
    for (; a && a.tagName !== "AG-TAB" && (a = a.parentElement, a !== this); )
      ;
    if (a && a.tagName === "AG-TAB") {
      const i = a;
      if (i.hasAttribute("disabled") || i.getAttribute("aria-disabled") === "true")
        return;
      const e = this._tabs.indexOf(i);
      e >= 0 && this._activateTab(e);
    }
  }
  /**
   * Helper method to find the next non-disabled tab in a given direction
   * @param startIndex - The starting index
   * @param direction - 1 for forward, -1 for backward
   * @returns The index of the next non-disabled tab, or startIndex if none found
   */
  _findNextEnabledTab(t, a) {
    const i = this._tabs.length;
    if (i === 0) return t;
    let e = t, s = 0;
    for (; s < i; ) {
      e = a === 1 ? (e + 1) % i : e === 0 ? i - 1 : e - 1;
      const r = this._tabs[e];
      if (!(r.hasAttribute("disabled") || r.getAttribute("aria-disabled") === "true"))
        return e;
      s++;
    }
    return t;
  }
  /**
   * Helper method to find the first non-disabled tab from start
   * @param fromStart - If true, search from beginning; if false, search from end
   * @returns The index of the first non-disabled tab, or 0 if none found
   */
  _findFirstEnabledTab(t) {
    const a = this._tabs.length;
    if (a === 0) return 0;
    if (t)
      for (let i = 0; i < a; i++) {
        const e = this._tabs[i];
        if (!(e.hasAttribute("disabled") || e.getAttribute("aria-disabled") === "true")) return i;
      }
    else
      for (let i = a - 1; i >= 0; i--) {
        const e = this._tabs[i];
        if (!(e.hasAttribute("disabled") || e.getAttribute("aria-disabled") === "true")) return i;
      }
    return 0;
  }
  _setFocusedTab(t) {
    t >= 0 && t < this._tabs.length && (this._focusedTab = t, this._updateTabsAndPanels(), this._tabs[t].focus());
  }
  _activateTab(t) {
    if (t >= 0 && t < this._tabs.length) {
      const a = this.activeTab, i = this._tabs[t];
      if (i.hasAttribute("disabled") || i.getAttribute("aria-disabled") === "true")
        return;
      if (a !== t) {
        this.activeTab = t, this._tabs[t].focus();
        const e = new CustomEvent("tab-change", {
          detail: {
            activeTab: t,
            previousTab: a
          },
          bubbles: !0,
          composed: !0
        });
        this.dispatchEvent(e), this.onTabChange && this.onTabChange(e);
      } else
        this._tabs[t].focus();
    }
  }
  render() {
    return p`
      <div class="tabs-container" part="ag-tabs-container" data-orientation=${this.orientation}>
        <div
          role="tablist"
          part="ag-tabs-tablist"
          aria-orientation=${this.orientation}
          aria-label=${this.ariaLabel || ""}
        >
          <slot name="tab"></slot>
        </div>
        <div class="tab-panels" part="ag-tabs-panels">
          <slot name="panel"></slot>
        </div>
      </div>
    `;
  }
};
_.styles = u`
    :host {
      display: block;
    }

    .tabs-container {
      display: flex;
    }

    .tabs-container[data-orientation="vertical"] {
      flex-direction: row;
    }

    .tabs-container[data-orientation="horizontal"] {
      flex-direction: column;
    }

    [role="tablist"] {
      display: flex;
      gap: var(--ag-space-2);
    }

    [role="tablist"][aria-orientation="horizontal"] {
      flex-direction: row;
      border-bottom: 1px solid var(--ag-border);
    }

    [role="tablist"][aria-orientation="vertical"] {
      flex-direction: column;
      border-inline-end: 1px solid var(--ag-border);
      min-width: 200px;
    }

    .tab-panels {
      flex: 1;
    }

    ::slotted(ag-tab-panel[hidden]) {
      display: none;
    }
  `;
let n = _;
l([
  d({ type: String })
], n.prototype, "activation");
l([
  d({ type: Number, attribute: "active-tab" })
], n.prototype, "activeTab");
l([
  d({ type: String })
], n.prototype, "orientation");
l([
  d({ type: String, reflect: !0, attribute: "aria-label" })
], n.prototype, "ariaLabel");
l([
  d({ attribute: !1 })
], n.prototype, "onTabChange");
l([
  f()
], n.prototype, "_tabs");
l([
  f()
], n.prototype, "_panels");
l([
  f()
], n.prototype, "_focusedTab");
export {
  c as Tab,
  T as TabPanel,
  n as Tabs
};
