// agnosticui-core@2.0.0-alpha.12/dist/components/Badge/core/_Badge.js
import { LitElement as c, css as v, html as u } from "lit";
import { n as a } from "../../../property-CemaeiRl.js";
import { o as n } from "../../../if-defined-C8i28hSj.js";
var p = Object.defineProperty, i = (l, t, r, h) => {
  for (var s = void 0, o = l.length - 1, g; o >= 0; o--)
    (g = l[o]) && (s = g(t, r, s) || s);
  return s && p(t, r, s), s;
};
const d = class d extends c {
  constructor() {
    super(), this.variant = "default", this.size = "md", this.dot = !1, this.value = null, this.max = 99, this.interactive = !1, this.statusLabel = null, this.live = "off", this.hiddenFromAT = !1;
  }
  get _displayValue() {
    return this.value === null || this.value === void 0 ? "" : this.value > this.max ? `${this.max}+` : this.value.toString();
  }
  _handleClick() {
    this.interactive && this.dispatchEvent(new CustomEvent("badge:click", { bubbles: !0, composed: !0 }));
  }
  _handleKeydown(t) {
    this.interactive && (t.key === "Enter" || t.key === " ") && this.dispatchEvent(new CustomEvent("badge:click", { bubbles: !0, composed: !0 }));
  }
  connectedCallback() {
    super.connectedCallback(), this.live !== "off" && this._updateLiveRegion();
  }
  updated(t) {
    super.updated(t), this._checkSingleChar(), t.has("value") && this.live !== "off" && this._updateLiveRegion();
  }
  _checkSingleChar() {
    let t = this._displayValue || "";
    const r = this.shadowRoot?.querySelector("slot");
    if (r) {
      const h = r.assignedNodes({ flatten: !0 });
      t += h.map((s) => s.textContent || "").join("");
    }
    t = t.replace(/\s+/g, ""), t.length === 1 && !this.dot ? this.setAttribute("single-char", "true") : this.removeAttribute("single-char");
  }
  _onSlotChange() {
    this._checkSingleChar();
  }
  _updateLiveRegion() {
    let t = document.getElementById("ag-badge-live");
    t || (t = document.createElement("div"), t.id = "ag-badge-live", t.className = "visually-hidden", t.setAttribute("aria-live", this.live), t.setAttribute("aria-atomic", "true"), document.body.appendChild(t)), t.setAttribute("aria-live", this.live), this.value !== null && this.value !== void 0 ? t.textContent = this.statusLabel ? this.statusLabel : `${this._displayValue} items` : t.textContent = this.statusLabel || "";
  }
  willUpdate(t) {
    (t.has("dot") || t.has("statusLabel")) && (this.hiddenFromAT = this.dot && !this.statusLabel || this.hiddenFromAT);
  }
  render() {
    const t = this.interactive ? "0" : void 0;
    return u`
      <span
        part="ag-badge"
        class="badge"
        role=${n(this.interactive ? "button" : void 0)}
        tabindex=${n(t)}
        aria-label=${n(this.statusLabel || void 0)}
        aria-hidden=${this.hiddenFromAT ? "true" : "false"}
        @click=${this._handleClick}
        @keydown=${this._handleKeydown}
      >
        ${this.dot ? "" : this._displayValue}
        <slot @slotchange=${this._onSlotChange}></slot>
      </span>
    `;
  }
};
d.styles = v`
    :host {
      display: inline-flex;
      align-items: center;
      justify-content: center;
    }
    .badge {
      display: inline-flex;
      align-items: center;
      justify-content: center;
      border-radius: var(--ag-radius-full);
      padding: var(--ag-space-1) var(--ag-space-2);
      font-size: var(--ag-font-size-xs);
      line-height: 1;
      white-space: nowrap;
      transition: all var(--ag-motion-fast);
    }
    :host([dot]) .badge {
      width: var(--ag-space-2);
      height: var(--ag-space-2);
      padding: 0;
      border-radius: 50%;
    }
    :host([size="xs"]) .badge {
      font-size: calc(var(--ag-font-size-xs) - 1px);
    }
    :host([size="xs"]:not([dot])) .badge {
      padding: var(--ag-space-0) var(--ag-space-1);
    }
    :host([size="xs"][dot]) .badge {
      width: var(--ag-space-1);
      height: var(--ag-space-1);
    }
    :host([size="sm"]) .badge {
      font-size: var(--ag-font-size-xs);
    }
    :host([size="sm"]:not([dot])) .badge {
      padding: var(--ag-space-1) var(--ag-space-2);
    }
    :host([size="sm"][dot]) .badge {
      width: var(--ag-space-2);
      height: var(--ag-space-2);
    }
    :host([size="md"]) .badge {
      font-size: var(--ag-font-size-sm);
    }
    :host([size="md"]:not([dot])) .badge {
      padding: var(--ag-space-1) var(--ag-space-3);
    }
    :host([size="md"][dot]) .badge {
      width: var(--ag-space-3);
      height: var(--ag-space-3);
    }
    :host([variant="default"]) .badge {
      background-color: var(--ag-background-tertiary);
      color: var(--ag-text-primary);
    }
    :host([variant="primary"]) .badge {
      background-color: var(--ag-primary);
      color: var(--ag-white);
    }
    :host([variant="success"]) .badge {
      background-color: var(--ag-success);
      color: var(--ag-white);
    }
    :host([variant="warning"]) .badge {
      background-color: var(--ag-warning);
      color: var(--ag-black);
    }
    :host([variant="danger"]) .badge {
      background-color: var(--ag-danger);
      color: var(--ag-white);
    }
    :host([variant="info"]) .badge {
      background-color: var(--ag-info-dark);
      color: var(--ag-white);
    }
    :host([variant="neutral"]) .badge {
      background-color: var(--ag-neutral-200);
      color: var(--ag-neutral-800);
    }
    :host([variant="monochrome"]) .badge {
      background-color: var(--ag-background-primary-inverted);
      color: var(--ag-text-primary-inverted);
    }
    :host([interactive]) .badge {
      cursor: pointer;
    }
    :host([interactive]) .badge:hover {
      filter: brightness(1.1);
    }
    :host([interactive]) .badge:focus-visible {
      outline: var(--ag-focus-width) solid var(--ag-focus);
      outline-offset: var(--ag-focus-offset);
    }
    :host([single-char]) .badge {
      border-radius: 50%;
    }
    :host([size="xs"][single-char]) .badge {
      padding: 1px;
      width: calc(var(--ag-font-size-xs));
      height: calc(var(--ag-font-size-xs));
    }
    :host([size="sm"][single-char]) .badge {
      padding: 2px;
      width: var(--ag-font-size-sm);
      height: var(--ag-font-size-sm);
    }
    :host([size="md"][single-char]) .badge {
      padding: 2px;
      width: var(--ag-font-size-base);
      height: var(--ag-font-size-base);
    }
  `;
let e = d;
i([
  a({ type: String, reflect: !0 })
], e.prototype, "variant");
i([
  a({ type: String, reflect: !0 })
], e.prototype, "size");
i([
  a({ type: Boolean, reflect: !0 })
], e.prototype, "dot");
i([
  a({ type: Number })
], e.prototype, "value");
i([
  a({ type: Number })
], e.prototype, "max");
i([
  a({ type: Boolean, reflect: !0 })
], e.prototype, "interactive");
i([
  a({ type: String, attribute: "status-label" })
], e.prototype, "statusLabel");
i([
  a({ type: String })
], e.prototype, "live");
i([
  a({ type: Boolean, attribute: "hidden-from-at" })
], e.prototype, "hiddenFromAT");
export {
  e as AgBadge
};
