// agnosticui-core@2.0.0-alpha.12/dist/components/ScrollProgress/core/_ScrollProgress.js
import { LitElement as g, css as p, html as c } from "lit";
import { n as i } from "../../../property-CemaeiRl.js";
import "../../Progress/core/Progress.js";
import "../../ProgressRing/core/ProgressRing.js";
import "../../Badge/core/Badge.js";
var h = Object.defineProperty, s = (d, e, l, n) => {
  for (var t = void 0, r = d.length - 1, a; r >= 0; r--)
    (a = d[r]) && (t = a(e, l, t) || t);
  return t && h(e, l, t), t;
};
class o extends g {
  constructor() {
    super(), this._scrollPercent = 0, this._scrollHandler = this._handleScroll.bind(this), this._resizeHandler = this._handleScroll.bind(this), this.mode = "bar", this.target = null, this.orientation = "top", this.dots = 5, this.badgeVariant = "info", this.ringSize = 32, this.ringStrokeWidth = 3, this.ringVariant = "info", this.precision = 0;
  }
  static get styles() {
    return p`
      :host {
        display: block;
      }

      /* Bar mode positioning */
      :host([mode="bar"]) {
        position: fixed;
        inset-inline: 0;
        z-index: var(--ag-z-index-sticky, 100);
      }

      :host([mode="bar"][orientation="top"]) {
        top: 0;
      }

      :host([mode="bar"][orientation="bottom"]) {
        bottom: 0;
      }

      /* Dot trail mode */
      .dot-trail {
        display: flex;
        gap: var(--ag-space-2);
        align-items: center;
        justify-content: center;
      }

      /* Badge mode */
      .badge-wrapper {
        display: inline-flex;
      }

      /* Ring mode */
      .ring-wrapper {
        display: inline-flex;
      }
    `;
  }
  connectedCallback() {
    super.connectedCallback(), this._attachScrollListener(), this._handleScroll();
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this._detachScrollListener();
  }
  _attachScrollListener() {
    this._getScrollTarget().addEventListener("scroll", this._scrollHandler, { passive: !0 }), window.addEventListener("resize", this._resizeHandler, { passive: !0 });
  }
  _detachScrollListener() {
    this._getScrollTarget().removeEventListener("scroll", this._scrollHandler), window.removeEventListener("resize", this._resizeHandler);
  }
  _getScrollTarget() {
    return this.target ? this.target : window;
  }
  _handleScroll() {
    let e, l, n;
    if (this.target)
      e = this.target.scrollTop, l = this.target.scrollHeight, n = this.target.clientHeight;
    else {
      const r = document.scrollingElement || document.documentElement;
      e = r.scrollTop, l = r.scrollHeight, n = r.clientHeight;
    }
    const t = l - n;
    if (t <= 0)
      this._scrollPercent = 0;
    else {
      const r = e / t * 100, a = Math.pow(10, this.precision);
      this._scrollPercent = Math.round(r * a) / a, this._scrollPercent = Math.min(100, Math.max(0, this._scrollPercent));
    }
  }
  _renderBarMode() {
    return c`
      <ag-progress
        part="scroll-progress-bar"
        .value=${this._scrollPercent}
        size="small"
        label="Scroll progress"
      ></ag-progress>
    `;
  }
  _renderDotTrailMode() {
    const e = Math.floor(this._scrollPercent / 100 * this.dots), l = Array.from({ length: this.dots }, (n, t) => {
      const r = t < e;
      return c`
        <ag-badge
          dot
          size="sm"
          variant=${r ? this.badgeVariant : "neutral"}
          hidden-from-at
        ></ag-badge>
      `;
    });
    return c`
      <div class="dot-trail" part="scroll-progress-dots">
        ${l}
      </div>
    `;
  }
  _renderBadgeMode() {
    const e = this._scrollPercent.toFixed(this.precision);
    return c`
      <div class="badge-wrapper">
        <ag-badge
          part="scroll-progress-badge"
          variant=${this.badgeVariant}
          status-label="Scroll progress ${e}%"
        >
          ${e}%
        </ag-badge>
      </div>
    `;
  }
  _renderRingMode() {
    const e = this._scrollPercent.toFixed(this.precision);
    return c`
      <div class="ring-wrapper">
        <ag-progress-ring
          part="scroll-progress-ring"
          .value=${this._scrollPercent}
          size="custom"
          variant=${this.ringVariant}
          label="Scroll progress"
          style="--ag-progress-ring-size: ${this.ringSize}px"
        >
          ${e}%
        </ag-progress-ring>
      </div>
    `;
  }
  render() {
    switch (this.mode) {
      case "bar":
        return this._renderBarMode();
      case "dot-trail":
        return this._renderDotTrailMode();
      case "badge":
        return this._renderBadgeMode();
      case "ring":
        return this._renderRingMode();
      default:
        return this._renderBarMode();
    }
  }
}
s([
  i({ type: String, reflect: !0 })
], o.prototype, "mode");
s([
  i({ attribute: !1 })
], o.prototype, "target");
s([
  i({ type: String, reflect: !0 })
], o.prototype, "orientation");
s([
  i({ type: Number })
], o.prototype, "dots");
s([
  i({ type: String, attribute: "badge-variant" })
], o.prototype, "badgeVariant");
s([
  i({ type: Number, attribute: "ring-size" })
], o.prototype, "ringSize");
s([
  i({ type: Number, attribute: "ring-stroke-width" })
], o.prototype, "ringStrokeWidth");
s([
  i({ type: String, attribute: "ring-variant" })
], o.prototype, "ringVariant");
s([
  i({ type: Number })
], o.prototype, "precision");
s([
  i({ type: Number, state: !0 })
], o.prototype, "_scrollPercent");
export {
  o as ScrollProgress
};
