// agnosticui-core@2.0.0-alpha.12/dist/utils/positioning.js
const i = {
  edgeWidth: "var(--ag-drawer-width)",
  edgeHeight: "var(--ag-sheet-height)",
  cornerMaxWidth: "90vw",
  cornerMaxHeight: "90vh"
};
function f($) {
  return $ === "top" || $ === "bottom" || $ === "start" || $ === "end";
}
function x($) {
  return $ === "top-start" || $ === "top-end" || $ === "bottom-start" || $ === "bottom-end";
}
function w($) {
  return $ ? $ === "top" || $ === "bottom" || $.startsWith("top-") || $.startsWith("bottom-") : !1;
}
function z($) {
  return $ === "start" || $ === "end";
}
function n($, p = "position") {
  return `[${p}="${$}"]`;
}
function v($, p, d = "position", g = i) {
  const o = { ...i, ...g };
  return `
    /* Edge position host sizing */
    ${$}${n("start", d)},
    ${$}${n("end", d)} {
      height: 100%;
    }

    ${$}${n("top", d)},
    ${$}${n("bottom", d)} {
      width: 100%;
    }

    /* Base positioning for all positioned containers */
    ${$}${n("start", d)} ${p},
    ${$}${n("end", d)} ${p},
    ${$}${n("top", d)} ${p},
    ${$}${n("bottom", d)} ${p},
    ${$}${n("top-start", d)} ${p},
    ${$}${n("top-end", d)} ${p},
    ${$}${n("bottom-start", d)} ${p},
    ${$}${n("bottom-end", d)} ${p} {
      box-sizing: border-box;
      position: fixed;
      margin: initial;
      border-radius: 0;
    }

    /* Vertical edge positions (start/end) setup */
    ${$}${n("start", d)} ${p},
    ${$}${n("end", d)} ${p} {
      height: 100%;
      width: ${o.edgeWidth};
      max-block-size: initial;
    }

    /* Horizontal edge positions (top/bottom) setup */
    ${$}${n("top", d)} ${p},
    ${$}${n("bottom", d)} ${p} {
      height: ${o.edgeHeight};
      width: 100%;
      max-inline-size: initial;
    }

    /* Corner positions setup */
    ${$}${n("top-start", d)} ${p},
    ${$}${n("top-end", d)} ${p},
    ${$}${n("bottom-start", d)} ${p},
    ${$}${n("bottom-end", d)} ${p} {
      max-inline-size: ${o.cornerMaxWidth};
      max-block-size: ${o.cornerMaxHeight};
      border-radius: var(--ag-radius-lg);
    }

    /* Top positioning (closed state with transforms) */
    ${$}${n("top", d)} ${p},
    ${$}${n("top-start", d)} ${p},
    ${$}${n("top-end", d)} ${p} {
      top: 0;
      transform: translateY(-100%);
    }

    /* Bottom positioning (closed state with transforms) */
    ${$}${n("bottom", d)} ${p},
    ${$}${n("bottom-start", d)} ${p},
    ${$}${n("bottom-end", d)} ${p} {
      bottom: 0;
      transform: translateY(100%);
    }

    /* Start positioning (closed state with transform) */
    ${$}${n("start", d)} ${p} {
      inset-inline-start: 0;
      transform: translateX(-100%);
    }

    /* End positioning (closed state with transform) */
    ${$}${n("end", d)} ${p} {
      inset-inline-end: 0;
      transform: translateX(100%);
    }

    /* RTL Support - flip transform directions */
    ${$}:host-context([dir="rtl"])${n("start", d)} ${p} {
      transform: translateX(100%);
    }

    ${$}:host-context([dir="rtl"])${n("end", d)} ${p} {
      transform: translateX(-100%);
    }

    /* Corner start (left) side positioning */
    ${$}${n("top-start", d)} ${p},
    ${$}${n("bottom-start", d)} ${p} {
      inset-inline-start: 0;
    }

    /* Corner end (right) side positioning */
    ${$}${n("top-end", d)} ${p},
    ${$}${n("bottom-end", d)} ${p} {
      inset-inline-end: 0;
    }

    /* Open state - remove transforms to slide into view */
    ${$}${n("top", d)}[open] ${p},
    ${$}${n("bottom", d)}[open] ${p},
    ${$}${n("start", d)}[open] ${p},
    ${$}${n("end", d)}[open] ${p},
    ${$}${n("top-start", d)}[open] ${p},
    ${$}${n("top-end", d)}[open] ${p},
    ${$}${n("bottom-start", d)}[open] ${p},
    ${$}${n("bottom-end", d)}[open] ${p} {
      transform: none;
    }
  `;
}
export {
  i as DEFAULT_POSITION_CONFIG,
  v as generatePositionCSS,
  n as getPositionSelector,
  x as isCornerPosition,
  f as isEdgePosition,
  w as isHorizontalPosition,
  z as isVerticalPosition
};
