// agnosticui-core@2.0.0-alpha.12/dist/components/Flex/core/_FlexContainer.js
import { LitElement as p, css as h, html as d } from "lit";
import { n } from "../../../property-CemaeiRl.js";
var u = Object.defineProperty, x = Object.getOwnPropertyDescriptor, s = (f, e, t, o) => {
  for (var i = o > 1 ? void 0 : o ? x(e, t) : e, l = f.length - 1, a; l >= 0; l--)
    (a = f[l]) && (i = (o ? a(e, t, i) : a(i)) || i);
  return o && i && u(e, t, i), i;
};
const c = class c extends p {
  constructor() {
    super(), this._direction = "row", this._wrap = "nowrap", this._justify = "flex-start", this._align = "stretch", this._alignContent = "stretch", this.inline = !1, this.reverse = !1, this.stretchChildren = !1;
  }
  get direction() {
    return this._direction;
  }
  set direction(e) {
    const t = ["row", "row-reverse", "column", "column-reverse"];
    this._direction = e && t.includes(e) ? e : "row", this.requestUpdate();
  }
  get wrap() {
    return this._wrap;
  }
  set wrap(e) {
    const t = ["nowrap", "wrap", "wrap-reverse"];
    this._wrap = e && t.includes(e) ? e : "nowrap", this.requestUpdate();
  }
  get justify() {
    return this._justify;
  }
  set justify(e) {
    const t = ["flex-start", "flex-end", "center", "space-between", "space-around", "space-evenly"];
    this._justify = e && t.includes(e) ? e : "flex-start", this.requestUpdate();
  }
  get align() {
    return this._align;
  }
  set align(e) {
    const t = ["flex-start", "flex-end", "center", "baseline", "stretch"];
    this._align = e && t.includes(e) ? e : "stretch", this.requestUpdate();
  }
  get alignContent() {
    return this._alignContent;
  }
  set alignContent(e) {
    const t = ["flex-start", "flex-end", "center", "space-between", "space-around", "space-evenly", "stretch"];
    this._alignContent = e && t.includes(e) ? e : "stretch", this.requestUpdate();
  }
  updated(e) {
    super.updated(e);
  }
  render() {
    return d`
      <div class="flex-container" part="ag-flex-container">
        <slot></slot>
      </div>
    `;
  }
};
c.styles = h`
    :host {
      display: var(--host-display, contents);
    }

    .flex-container {
      display: flex;
      flex-direction: var(--flex-direction, row);
      flex-wrap: var(--flex-wrap, nowrap);
      justify-content: var(--flex-justify, flex-start);
      align-items: var(--flex-align, stretch);
      align-content: var(--flex-align-content, stretch);
      gap: var(--flex-gap, var(--ag-space-0, 0));
    }

    :host([inline]) .flex-container {
      display: inline-flex;
    }

    :host([stretch-children]) ::slotted(*) {
      flex: 1 1 auto;
    }

    /* Attribute-based CSS custom property defaults */
    /* These set low-specificity defaults that can be easily overridden */

    /* Direction */
    :host([direction="row"]:not([reverse])) {
      --flex-direction: row;
    }
    :host([direction="row"][reverse]) {
      --flex-direction: row-reverse;
    }
    :host([direction="row-reverse"]:not([reverse])) {
      --flex-direction: row-reverse;
    }
    :host([direction="row-reverse"][reverse]) {
      --flex-direction: row;
    }
    :host([direction="column"]:not([reverse])) {
      --flex-direction: column;
    }
    :host([direction="column"][reverse]) {
      --flex-direction: column-reverse;
    }
    :host([direction="column-reverse"]:not([reverse])) {
      --flex-direction: column-reverse;
    }
    :host([direction="column-reverse"][reverse]) {
      --flex-direction: column;
    }

    /* Wrap */
    :host([wrap="nowrap"]) {
      --flex-wrap: nowrap;
    }
    :host([wrap="wrap"]) {
      --flex-wrap: wrap;
    }
    :host([wrap="wrap-reverse"]) {
      --flex-wrap: wrap-reverse;
    }

    /* Justify */
    :host([justify="flex-start"]) {
      --flex-justify: flex-start;
    }
    :host([justify="flex-end"]) {
      --flex-justify: flex-end;
    }
    :host([justify="center"]) {
      --flex-justify: center;
    }
    :host([justify="space-between"]) {
      --flex-justify: space-between;
    }
    :host([justify="space-around"]) {
      --flex-justify: space-around;
    }
    :host([justify="space-evenly"]) {
      --flex-justify: space-evenly;
    }

    /* Align */
    :host([align="flex-start"]) {
      --flex-align: flex-start;
    }
    :host([align="flex-end"]) {
      --flex-align: flex-end;
    }
    :host([align="center"]) {
      --flex-align: center;
    }
    :host([align="baseline"]) {
      --flex-align: baseline;
    }
    :host([align="stretch"]) {
      --flex-align: stretch;
    }

    /* Align Content */
    :host([align-content="flex-start"]) {
      --flex-align-content: flex-start;
    }
    :host([align-content="flex-end"]) {
      --flex-align-content: flex-end;
    }
    :host([align-content="center"]) {
      --flex-align-content: center;
    }
    :host([align-content="space-between"]) {
      --flex-align-content: space-between;
    }
    :host([align-content="space-around"]) {
      --flex-align-content: space-around;
    }
    :host([align-content="space-evenly"]) {
      --flex-align-content: space-evenly;
    }
    :host([align-content="stretch"]) {
      --flex-align-content: stretch;
    }

    /* Gap - Controlled ONLY via --flex-gap CSS custom property (no attribute/prop) */
    /* Gap values are freeform strings, so attribute selectors are not feasible */
    /* (CSS attr() doesn't work in custom properties and we'd need infinite selectors) */
    /* Set --flex-gap via classes, ::part(), or inline styles. Default: var(--ag-space-0, 0) */
  `;
let r = c;
s([
  n({ type: String, reflect: !0 })
], r.prototype, "direction", 1);
s([
  n({ type: String, reflect: !0 })
], r.prototype, "wrap", 1);
s([
  n({ type: String, reflect: !0 })
], r.prototype, "justify", 1);
s([
  n({ type: String, reflect: !0 })
], r.prototype, "align", 1);
s([
  n({ type: String, attribute: "align-content" })
], r.prototype, "alignContent", 1);
s([
  n({ type: Boolean, reflect: !0 })
], r.prototype, "inline", 2);
s([
  n({ type: Boolean, reflect: !0 })
], r.prototype, "reverse", 2);
s([
  n({ type: Boolean, attribute: "stretch-children", reflect: !0 })
], r.prototype, "stretchChildren", 2);
export {
  r as FlexContainer
};
