// typed-htmx@0.3.1/dist/esm/typed-html/jsx-runtime.js
/// <reference types="typed-html" />
/// <reference path="../jsx.d.ts" />
import { createElement } from "typed-html";
import { jsxConfig } from "../index.js";
export class Node {
    constructor(children) {
        this.children = children;
    }
    toString() {
        if (Array.isArray(this.children))
            return this.children.join("\n");
        return this.children.toString();
    }
}
export function Fragment({ children }) {
    if (Array.isArray(children)) {
        const elts = children.map(sanitizer);
        return jsxConfig.trusted ? elts.join("\n") : new Node(elts);
    }
    const elt = sanitizer(children);
    return jsxConfig.trusted ? elt : new Node(elt);
}
function sanitizer(value) {
    const str = value || value === 0 ? value.toString() : "";
    if (!jsxConfig.sanitize || jsxConfig.trusted)
        return str;
    if (value instanceof Node)
        return value;
    return jsxConfig.sanitize(str, typeof value);
}
function expandLiterals(props) {
    for (const attr of jsxConfig.jsonAttributes) {
        if (!(attr in props))
            continue;
        const value = props[attr];
        if (typeof value === "object") {
            props[attr] = { toString: () => JSON.stringify(value) };
        }
    }
}
export function jsx(tag, { children, ...props }) {
    expandLiterals(props);
    const contents = Array.isArray(children) ? children.map(sanitizer) : [sanitizer(children)];
    const elt = createElement(tag, props, ...contents);
    return jsxConfig.trusted ? elt : new Node(elt);
}
export function jsxs(tag, { children, ...props }) {
    expandLiterals(props);
    const elt = createElement(tag, props, ...children.map(sanitizer));
    return jsxConfig.trusted ? elt : new Node(elt);
}
