// agnosticui-core@2.0.0-alpha.12/dist/components/Header/vue/VueHeader.js
import { defineComponent as s, ref as r, onMounted as a, createElementBlock as c, openBlock as i, mergeProps as f, createElementVNode as u, renderSlot as n } from "vue";
import "../core/Header.js";
import { _ as l } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const d = s({
  name: "VueHeader",
  props: {
    sticky: {
      type: Boolean,
      default: !1
    },
    contentJustify: {
      type: String,
      default: "between"
    }
  },
  setup(e, { emit: o }) {
    const t = r(null);
    return a(async () => {
      await customElements.whenDefined("ag-header");
    }), {
      agComponent: t,
      sticky: e.sticky,
      contentJustify: e.contentJustify
    };
  }
}), p = [".sticky", ".contentJustify"], m = { slot: "logo" };
function y(e, o, t, g, k, _) {
  return i(), c("ag-header", f({
    ref: "agComponent",
    ".sticky": e.sticky,
    ".contentJustify": e.contentJustify
  }, e.$attrs), [
    u("span", m, [
      n(e.$slots, "logo")
    ]),
    n(e.$slots, "default")
  ], 48, p);
}
const w = /* @__PURE__ */ l(d, [["render", y]]);
export {
  w as default
};
