// agnosticui-core@2.0.0-alpha.12/dist/components/Pagination/core/_ContentPagination.js
import { LitElement as d, html as n, css as h } from "lit";
import { n as o } from "../../../property-CemaeiRl.js";
var v = Object.defineProperty, r = (l, t, i, c) => {
  for (var a = void 0, s = l.length - 1, g; s >= 0; s--)
    (g = l[s]) && (a = g(t, i, a) || a);
  return a && v(t, i, a), a;
};
const p = class p extends d {
  constructor() {
    super(), this._hasParentAndChild = !1, this.ariaLabel = "content navigation", this.bordered = !1;
  }
  willUpdate(t) {
    (t.has("parent") || t.has("previous") || t.has("next")) && (this._hasParentAndChild = !!this.parent && (!!this.previous || !!this.next));
  }
  _renderChevronIcon(t = "") {
    return n`<svg aria-hidden="true" fill="none" focusable="false" height="20" role="presentation" viewBox="0 0 24 24" width="20" class="${t} text-primary"><path d="M15.5 19l-7-7 7-7" stroke="currentColor" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5"></path></svg>`;
  }
  _handleNavigate(t, i, c) {
    if (!i || i.href && !this.onNavigate)
      return;
    this.onNavigate && c.preventDefault();
    const a = new CustomEvent("navigate", {
      detail: {
        direction: t,
        title: i.title,
        href: i.href
      },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(a), this.onNavigate && this.onNavigate(a);
  }
  render() {
    return n`
      <nav
        class="content-pagination-container"
        part="ag-content-pagination-container"
        aria-label=${this.ariaLabel}
      >
        ${this.parent ? n`
              <div
                class="content-pagination-parent"
                part="ag-content-pagination-parent"
              >
                ${this.parent.href ? n`
                      <a
                        href=${this.parent.href}
                        class="content-pagination-parent-link"
                        part="ag-content-pagination-link"
                        aria-label="Up to: ${this.parent.title}"
                        @click=${(t) => this._handleNavigate("parent", this.parent, t)}
                      >
                        <span class="content-pagination-icon">
                          <slot name="parent-icon">${"↑"}</slot>
                        </span>
                        <span>${this.parent.title}</span>
                      </a>
                    ` : n`
                      <button
                        type="button"
                        class="content-pagination-parent-link"
                        part="ag-content-pagination-link"
                        aria-label="Up to: ${this.parent.title}"
                        @click=${(t) => this._handleNavigate("parent", this.parent, t)}
                      >
                        <span class="content-pagination-icon">
                          <slot name="parent-icon">${"↑"}</slot>
                        </span>
                        <span>${this.parent.title}</span>
                      </button>
                    `}
              </div>
            ` : ""}

        <div
          class="content-pagination-nav ${this._hasParentAndChild ? "content-pagination-nav-with-parent" : ""}"
          part="ag-content-pagination-nav"
        >
          ${this.previous ? n`
                <div class="content-pagination-item content-pagination-item-previous">
                  ${this.previous.href ? n`
                        <a
                          href=${this.previous.href}
                          class="content-pagination-link"
                          part="ag-content-pagination-link"
                          aria-label="Previous: ${this.previous.title}"
                          @click=${(t) => this._handleNavigate("previous", this.previous, t)}
                        >
                          <span class="content-pagination-icon">
                            <slot name="previous-icon">${this._renderChevronIcon()}</slot>
                          </span>
                          <span class="content-pagination-title">${this.previous.title}</span>
                        </a>
                      ` : n`
                        <button
                          type="button"
                          class="content-pagination-link"
                          part="ag-content-pagination-link"
                          aria-label="Previous: ${this.previous.title}"
                          @click=${(t) => this._handleNavigate("previous", this.previous, t)}
                        >
                          <span class="content-pagination-icon">
                            <slot name="previous-icon">${this._renderChevronIcon()}</slot>
                          </span>
                          <span class="content-pagination-title">${this.previous.title}</span>
                        </button>
                      `}
                </div>
              ` : n`<div class="content-pagination-item"></div>`}

          ${this.next ? n`
                <div class="content-pagination-item content-pagination-item-next">
                  ${this.next.href ? n`
                        <a
                          href=${this.next.href}
                          class="content-pagination-link"
                          part="ag-content-pagination-link"
                          aria-label="Next: ${this.next.title}"
                          @click=${(t) => this._handleNavigate("next", this.next, t)}
                        >
                          <span class="content-pagination-title">${this.next.title}</span>
                          <span class="content-pagination-icon">
                            <slot name="next-icon">${this._renderChevronIcon("rotate-180")}</slot>
                          </span>
                        </a>
                      ` : n`
                        <button
                          type="button"
                          class="content-pagination-link"
                          part="ag-content-pagination-link"
                          aria-label="Next: ${this.next.title}"
                          @click=${(t) => this._handleNavigate("next", this.next, t)}
                        >
                          <span class="content-pagination-title">${this.next.title}</span>
                          <span class="content-pagination-icon">
                            <slot name="next-icon">${this._renderChevronIcon("rotate-180")}</slot>
                          </span>
                        </button>
                      `}
                </div>
              ` : n`<div class="content-pagination-item"></div>`}
        </div>
      </nav>
    `;
  }
};
p.styles = h`
    :host {
      display: block;
      width: 100%;
    }

    .rotate-180 {
        transform: rotate(180deg);
    }

    .content-pagination-container {
      display: flex;
      flex-direction: column;
      gap: var(--ag-space-1);
    }

    .content-pagination-parent {
      display: flex;
      align-items: center;
      padding: var(--ag-space-2) var(--ag-space-5);
      border-radius: var(--ag-radius-md);
    }

    .content-pagination-parent-link {
      display: flex;
      align-items: center;
      gap: var(--ag-space-1);
      color: var(--ag-text-primary);
      text-decoration: none;
      font-size: var(--ag-font-size-sm);
      font-weight: 500;
      cursor: pointer;
      background: none;
      border: none;
      font-family: inherit;
    }

    .content-pagination-parent-link:hover {
      color: var(--ag-primary);
    }

    .content-pagination-parent-link:focus-visible {
      outline: var(--ag-focus-width) solid rgba(var(--ag-focus), 0.5);
      outline-offset: var(--ag-focus-offset);
      border-radius: var(--ag-radius-sm);
    }

    .content-pagination-nav {
      display: flex;
      justify-content: space-between;
      gap: var(--ag-space-4);
      padding-block-start: var(--ag-space-2);
      border-top: 0px solid transparent;
    }

    .content-pagination-nav-with-parent {
      border-top: 1px solid var(--ag-border);
    }

    .content-pagination-item {
      display: flex;
      flex: 0 1 auto;
      min-inline-size: 0;
      max-inline-size: 45%;
    }

    .content-pagination-item-previous {
      justify-content: flex-start;
    }

    .content-pagination-item-next {
      justify-content: flex-end;
      margin-inline-start: auto;
    }

    .content-pagination-link {
      display: flex;
      align-items: center;
      gap: var(--ag-space-2, 0.5rem);
      padding: var(--ag-space-3) var(--ag-space-4);
      border-radius: var(--ag-radius-md);
      background-color: var(--ag-background-primary);
      color: var(--ag-text-primary);
      text-decoration: none;
      font-size: var(--ag-font-size-base);
      font-weight: 500;
      cursor: pointer;
      transition: all 0.15s ease-in-out;
      font-family: inherit;
      min-width: 0;
    }

    :host([bordered]) .content-pagination-link {
      border: 1px solid var(--ag-border);
    }
    
    .content-pagination-link:hover {
      background-color: var(--ag-background-secondary);
      color: var(--ag-primary);
    }

    .content-pagination-link:focus-visible {
      outline: var(--ag-focus-width) solid rgba(var(--ag-focus), 0.5);
      outline-offset: var(--ag-focus-offset);
    }

    .content-pagination-link:active {
      transform: translateY(1px);
    }

    button.content-pagination-link {
      background-color: transparent;
      border: 0;
    }

    :host([bordered]) button.content-pagination-link {
      border: 1px solid var(--ag-border);
    }

    .content-pagination-title {
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
      min-width: 0;
    }

    .content-pagination-icon {
      flex-shrink: 0;
      font-size: 1.25em;
      line-height: 1;
    }

    /* Responsive: stack on very small screens */
    @media (max-width: 640px) {
      .content-pagination-nav {
        flex-direction: column;
        gap: var(--ag-space-3);
      }

      .content-pagination-item {
        max-inline-size: 100%;
      }

      .content-pagination-item-next {
        margin-inline-start: 0;
      }
    }

    @media (prefers-reduced-motion) {
      .content-pagination-link {
        transition: none;
      }

      .content-pagination-link:active {
        transform: none;
      }
    }
  `;
let e = p;
r([
  o({ type: Object, attribute: !1 })
], e.prototype, "previous");
r([
  o({ type: Object, attribute: !1 })
], e.prototype, "next");
r([
  o({ type: Object, attribute: !1 })
], e.prototype, "parent");
r([
  o({ type: String, attribute: "aria-label" })
], e.prototype, "ariaLabel");
r([
  o({ type: Boolean, reflect: !0 })
], e.prototype, "bordered");
r([
  o({ attribute: !1 })
], e.prototype, "onNavigate");
r([
  o({ type: Boolean, reflect: !0, attribute: "has-parent" })
], e.prototype, "_hasParentAndChild");
export {
  e as ContentPagination
};
