// agnosticui-core@2.0.0-alpha.12/dist/components/Pagination/core/_Pagination.js
import { LitElement as b, css as f, html as l } from "lit";
import { n as s } from "../../../property-CemaeiRl.js";
var m = Object.defineProperty, r = (d, t, i, e) => {
  for (var a = void 0, o = d.length - 1, g; o >= 0; o--)
    (g = d[o]) && (a = g(t, i, a) || a);
  return a && m(t, i, a), a;
};
const u = class u extends b {
  constructor() {
    super(), this.current = 1, this.totalPages = 1, this.offset = 2, this.justify = "", this.ariaLabel = "pagination", this.bordered = !1, this.firstLastNavigation = !1, this.navigationLabels = {
      first: "First",
      previous: "Previous",
      next: "Next",
      last: "Last"
    };
  }
  get _pages() {
    return this._generatePages();
  }
  updated(t) {
    (t.has("current") || t.has("totalPages") || t.has("offset")) && this.updateComplete.then(() => {
      const i = this.shadowRoot?.querySelector(
        `button[data-page="${this.current}"]`
      );
      i && i.focus();
    });
  }
  _generatePages() {
    return this.totalPages <= 1 ? [1] : this.offset === 1 ? this._generatePagingPaddedByOne(this.current, this.totalPages) : this._generatePagingPaddedByTwo(this.current, this.totalPages);
  }
  _getPaddedArray(t, i, e, a) {
    return i && t.unshift("..."), e && t.push("..."), a <= 1 ? [1] : [1, ...t, a];
  }
  _generatePagingPaddedByOne(t, i) {
    const a = [t - 1, t, t + 1].filter(
      (p) => p > 1 && p < i
    ), o = t > 3, g = t < i - 2;
    return this._getPaddedArray(
      a,
      o,
      g,
      i
    );
  }
  _generatePagingPaddedByTwo(t, i) {
    const a = [t - 2, t - 1, t, t + 1, t + 2].filter(
      (c) => c > 1 && c < i
    ), o = t === 5, g = t === i - 4, p = t > 5, h = t < i - 4;
    return o && a.unshift(2), g && a.push(i - 1), this._getPaddedArray(
      a,
      p,
      h,
      i
    );
  }
  _handlePageChange(t) {
    if (t < 1 || t > this.totalPages || t === this.current)
      return;
    const i = this._pages;
    this.current = t;
    const e = new CustomEvent("page-change", {
      detail: {
        page: t,
        pages: i
      },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(e), this.onPageChange && this.onPageChange(e);
  }
  _getJustifyClass() {
    switch (this.justify) {
      case "start":
        return "pagination-start";
      case "center":
        return "pagination-center";
      case "end":
        return "pagination-end";
      default:
        return "";
    }
  }
  render() {
    const t = this._getJustifyClass(), i = this._pages.length > 0 ? this._pages : [1], e = this.navigationLabels || {
      first: "First",
      previous: "Previous",
      next: "Next",
      last: "Last"
    };
    return l`
      <nav
        class="pagination-container ${t}"
        part="ag-pagination-container"
        aria-label=${this.ariaLabel}
      >
        <ul class="pagination" part="ag-pagination">
          ${this.firstLastNavigation ? l`
                <li
                  class="pagination-item ${this.current === 1 ? "pagination-item-disabled" : ""}"
                  part="ag-pagination-item"
                >
                  <button
                    class="pagination-button pagination-button-first"
                    part="ag-pagination-button"
                    type="button"
                    @click=${() => this._handlePageChange(1)}
                    ?disabled=${this.current === 1}
                    aria-disabled=${this.current === 1 ? "true" : "false"}
                    aria-label="Goto page 1"
                  >
                    ${"«"} ${e.first}
                  </button>
                </li>
              ` : ""}
          <li
            class="pagination-item ${this.current === 1 ? "pagination-item-disabled" : ""}"
            part="ag-pagination-item"
          >
            <button
              class="pagination-button pagination-button-previous"
              part="ag-pagination-button"
              type="button"
              @click=${() => this._handlePageChange(this.current - 1)}
              ?disabled=${this.current === 1}
              aria-disabled=${this.current === 1 ? "true" : "false"}
              aria-label="Goto previous page"
            >
              ${"‹"} ${e.previous}
            </button>
          </li>
          ${i.map(
      (a) => a !== "..." ? l`
                  <li
                    class="pagination-item ${this.current === a ? "pagination-item-active" : ""}"
                    part="ag-pagination-item"
                  >
                    <button
                      data-page="${a}"
                      @click=${() => this._handlePageChange(a)}
                      type="button"
                      class="pagination-button"
                      part="ag-pagination-button"
                      aria-current=${this.current === a ? "page" : "false"}
                      aria-label=${this.current === a ? `Page ${a}, current page` : `Goto page ${a}`}
                    >
                      ${a}
                    </button>
                  </li>
                ` : l`
                  <li
                    class="pagination-item pagination-item-gap"
                    part="ag-pagination-item"
                  >
                    <span>${a}</span>
                  </li>
                `
    )}
          <li
            class="pagination-item ${this.current === this.totalPages ? "pagination-item-disabled" : ""}"
            part="ag-pagination-item"
          >
            <button
              class="pagination-button pagination-button-next"
              part="ag-pagination-button"
              type="button"
              @click=${() => this._handlePageChange(this.current + 1)}
              ?disabled=${this.current === this.totalPages}
              aria-disabled=${this.current === this.totalPages ? "true" : "false"}
              aria-label="Goto next page"
            >
              ${e.next} ${"›"}
            </button>
          </li>
          ${this.firstLastNavigation ? l`
                <li
                  class="pagination-item pagination-item-last ${this.current === this.totalPages ? "pagination-item-disabled" : ""}"
                  part="ag-pagination-item"
                >
                  <button
                    type="button"
                    class="pagination-button pagination-button-last"
                    part="ag-pagination-button"
                    @click=${() => this._handlePageChange(this.totalPages)}
                    ?disabled=${this.current === this.totalPages}
                    aria-disabled=${this.current === this.totalPages ? "true" : "false"}
                    aria-label="Goto last page"
                  >
                    ${e.last} ${"»"}
                  </button>
                </li>
              ` : ""}
        </ul>
      </nav>
    `;
  }
};
u.styles = f`
    :host {
      display: block;
    }

    .pagination-container {
      display: flex;
      font-size: var(--ag-font-size-sm);
    }
    
    :host([justify="start"]),
    :host([justify="center"]),
    :host([justify="end"]) {
      width: 100%;
    }

    .pagination-center {
      justify-content: center;
    }

    .pagination-start {
      justify-content: flex-start;
    }

    .pagination-end {
      justify-content: flex-end;
    }

    .pagination {
      display: flex;
      list-style: none;
      margin: 0;
      padding: 0;
    }

    .pagination-item {
      margin-inline-start: 0.125rem;
      margin-inline-end: 0.125rem;
    }

    /* The first previous next last buttons are slightly tighter */
    .pagination-button-first,
    .pagination-button-previous,
    .pagination-button-next,
    .pagination-button-last {
      padding-inline-start: var(--ag-space-1);
      padding-inline-end: var(--ag-space-1);
    }

    .pagination-button {
      color: var(--ag-primary-text);
      display: inline-block;
      line-height: var(--ag-line-height-4);
      padding-inline-start: var(--ag-space-2);
      padding-inline-end: var(--ag-space-2);
      border-radius: var(--ag-radius-md);
      border: var(--ag-border-width-1) solid transparent;
      background-color: transparent;
      cursor: pointer;
      font: inherit;
    }

    .pagination-button:focus-visible {
      outline: var(--ag-focus-width) solid rgba(var(--ag-focus), 0.5);
      outline-offset: var(--ag-focus-offset);
    }

    @media (prefers-reduced-motion), (update: slow) {
      .pagination-button:focus-visible {
        transition-duration: 0.001ms !important;
      }
    }

    .pagination-item-disabled {
      cursor: not-allowed;
    }

    .pagination-button:disabled,
    .pagination-item-disabled .pagination-button {
      color: var(--ag-text-disabled);
      opacity: 0.8;
      pointer-events: none;
    }

    .pagination-item-active .pagination-button {
      background-color: var(--ag-background-tertiary);
      color: var(--ag-primary-text);
    }

    :host([bordered]) .pagination-item-active .pagination-button {
      background-color: transparent;
      border: var(--ag-border-width-1) solid var(--ag-primary-text);
      color: var(--ag-primary-text);
    }

    .pagination-item:hover .pagination-button {
      text-decoration: none;
    }

    .pagination-item:not(.pagination-item-active):not(.pagination-item-disabled):hover
      .pagination-button {
      background-color: var(--ag-background-secondary);
    }

    .pagination-item-gap {
      display: flex;
      align-items: center;
      transform: translateY(var(--ag-space-1));
    }
  `;
let n = u;
r([
  s({ type: Number })
], n.prototype, "current");
r([
  s({ type: Number, attribute: "total-pages" })
], n.prototype, "totalPages");
r([
  s({ type: Number })
], n.prototype, "offset");
r([
  s({ type: String, reflect: !0 })
], n.prototype, "justify");
r([
  s({ type: String, attribute: "aria-label" })
], n.prototype, "ariaLabel");
r([
  s({ type: Boolean, reflect: !0 })
], n.prototype, "bordered");
r([
  s({ type: Boolean, reflect: !0, attribute: "first-last-navigation" })
], n.prototype, "firstLastNavigation");
r([
  s({ type: Object, attribute: !1 })
], n.prototype, "navigationLabels");
r([
  s({ attribute: !1 })
], n.prototype, "onPageChange");
export {
  n as Pagination
};
