// agnosticui-core@2.0.0-alpha.12/dist/components/EmptyState/core/_EmptyState.js
import { hasSlotContent as l } from "../../../utils/slot.js";
import { LitElement as d, css as u, html as r } from "lit";
import { n as s } from "../../../property-CemaeiRl.js";
var b = Object.defineProperty, a = (p, e, o, n) => {
  for (var i = void 0, c = p.length - 1, g; c >= 0; c--)
    (g = p[c]) && (i = g(e, o, i) || i);
  return i && b(e, o, i), i;
};
const h = class h extends d {
  constructor() {
    super(), this._hasIconSlot = !1, this._hasActionsSlot = !1, this.title = "", this.subtitle = "", this.buttonText = "", this.size = "md", this.bordered = !1, this.rounded = !1;
  }
  _handleSlotChange(e) {
    const o = e.target, n = o.name;
    n === "icon" ? this._hasIconSlot = l(o) : n === "actions" && (this._hasActionsSlot = l(o)), this.requestUpdate();
  }
  firstUpdated() {
    setTimeout(() => {
      const e = this.shadowRoot?.querySelector('slot[name="icon"]'), o = this.shadowRoot?.querySelector('slot[name="actions"]'), n = this._hasIconSlot, i = this._hasActionsSlot;
      this._hasIconSlot = l(e), this._hasActionsSlot = l(o), (n !== this._hasIconSlot || i !== this._hasActionsSlot) && this.requestUpdate();
    }, 0);
  }
  render() {
    const e = [
      "empty",
      this.bordered ? "empty-bordered" : "",
      this.rounded ? "empty-rounded" : ""
    ].filter(Boolean).join(" ");
    return r`
      <!-- Main container with a part for the whole component -->
      <div class="${e}" role="region" aria-label="Empty state" part="ag-empty-state">

        <!-- Icon section part -->
        <div class="icon" part="ag-icon">
          <div class="icon-inner" part="ag-icon-inner">
            <slot name="icon" part="ag-icon-slot" @slotchange=${this._handleSlotChange}></slot>
            ${this._hasIconSlot ? "" : r`
              <!-- Fallback SVG part -->
              <svg
                part="ag-icon-svg"
                xmlns="http://www.w3.org/2000/svg"
                viewBox="0 0 24 24"
                fill="none"
                stroke="currentColor"
                stroke-width="2"
                stroke-linecap="round"
                stroke-linejoin="round"
              >
                <circle cx="12" cy="12" r="10"></circle>
                <line x1="12" y1="8" x2="12" y2="12"></line>
                <line x1="12" y1="16" x2="12.01" y2="16"></line>
              </svg>
            `}
          </div>
        </div>

        <!-- Title part -->
        ${this.title ? r`<h3 class="title" part="ag-title">${this.title}</h3>` : ""}

        <!-- Subtitle part -->
        ${this.subtitle ? r`<p class="subtitle" part="ag-subtitle">${this.subtitle}</p>` : ""}

        <!-- Actions container part -->
        <div class="actions" part="ag-actions">
          <slot name="actions" part="ag-actions-slot" @slotchange=${this._handleSlotChange}>
            ${this.buttonText && !this._hasActionsSlot ? r`
                <!-- Fallback button part -->
                <button type="button" part="ag-actions-button">${this.buttonText}</button>
              ` : ""}
          </slot>
        </div>
      </div>
    `;
  }
};
h.styles = u`
    :host {
      display: block;
    }

    .empty {
      display: flex;
      flex-direction: column;
      align-items: center;
      text-align: center;
      padding: var(--ag-space-8);
      background: var(--ag-background-secondary);
    }

    .empty-bordered {
      background: transparent;
      border: var(--ag-border-width-1) solid var(--ag-border-subtle);
    }

    .empty-rounded {
      border-radius: var(--ag-radius-lg);
    }

    /* Icon sizing and containment */
    .icon {
      margin-block-end: var(--ag-space-1);
      color: var(--ag-neutral-400);
    }

    .icon-inner {
      width: var(--ag-empty-icon-size, var(--ag-space-10));
      height: var(--ag-empty-icon-size, var(--ag-space-10));
      display: inline-flex;
      align-items: center;
      justify-content: center;
    }

    .icon-inner > ::slotted(*) {
      width: 100%;
      height: 100%;
      object-fit: contain;
    }

    :host([size="sm"]) {
      --ag-empty-icon-size: var(--ag-space-8);
    }

    :host([size="lg"]) {
      --ag-empty-icon-size: var(--ag-space-12);
    }

    /* Title */
    .title {
      font-size: var(--ag-font-size-xl);
      margin-block-start: 0;
      margin-block-end: var(--ag-space-2);
      line-height: var(--ag-line-height-lg);
    }

    :host([size="sm"]) .title {
      font-size: var(--ag-font-size-lg);
    }

    :host([size="lg"]) .title {
      font-size: var(--ag-font-size-2x);
    }

    /* Subtitle */
    .subtitle {
      font-size: var(--ag-font-size-base);
      color: var(--ag-text-muted);
      /* The title will provide enough gap */
      margin-block-start: 0;
      margin-block-end: var(--ag-space-4);
      line-height: var(--ag-line-height-base);
    }

    :host([size="sm"]) .subtitle {
      font-size: var(--ag-font-size-sm);
      margin-block-end: var(--ag-space-3);
    }

    :host([size="lg"]) .subtitle {
      font-size: var(--ag-font-size-md);
      margin-block-end: var(--ag-space-6);
    }

    /* Actions */
    .actions {
      display: flex;
      flex-wrap: wrap;
      gap: var(--ag-space-2);
      justify-content: center;
    }
  `;
let t = h;
a([
  s({ type: String })
], t.prototype, "title");
a([
  s({ type: String })
], t.prototype, "subtitle");
a([
  s({ type: String, attribute: "button-text" })
], t.prototype, "buttonText");
a([
  s({ type: String, reflect: !0 })
], t.prototype, "size");
a([
  s({ type: Boolean })
], t.prototype, "bordered");
a([
  s({ type: Boolean })
], t.prototype, "rounded");
a([
  s({ type: Boolean, state: !0 })
], t.prototype, "_hasIconSlot");
a([
  s({ type: Boolean, state: !0 })
], t.prototype, "_hasActionsSlot");
export {
  t as AgEmptyState
};
