// agnosticui-core@2.0.0-alpha.12/dist/components/SkeletonLoader/vue/VueSkeletonLoader.js
import { defineComponent as a, ref as o, onMounted as r, nextTick as f, watchEffect as s, createElementBlock as d, openBlock as l, mergeProps as h, renderSlot as c } from "vue";
import "../core/SkeletonLoader.js";
import { _ as u } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const g = a({
  name: "VueSkeletonLoader",
  props: {
    variant: {
      type: String,
      default: "text"
    },
    effect: {
      type: String,
      default: "pulse"
    },
    intensity: {
      type: String,
      default: "light"
    },
    width: {
      type: String,
      default: void 0
    },
    height: {
      type: String,
      default: void 0
    }
  },
  setup(e) {
    const n = o(null), i = () => {
      const t = n.value;
      t && (t.variant = e.variant, t.effect = e.effect, t.intensity = e.intensity, t.width = e.width, t.height = e.height);
    };
    return r(async () => {
      await customElements.whenDefined("ag-skeleton-loader"), await f(), i();
    }), s(() => {
      n.value && i();
    }), { agComponent: n };
  }
}), p = ["variant", "effect", "intensity", "width", "height"];
function m(e, n, i, t, y, w) {
  return l(), d("ag-skeleton-loader", h({
    ref: "agComponent",
    variant: e.variant,
    effect: e.effect,
    intensity: e.intensity,
    width: e.width,
    height: e.height
  }, e.$attrs), [
    c(e.$slots, "default")
  ], 16, p);
}
const $ = /* @__PURE__ */ u(g, [["render", m]]);
export {
  $ as default
};
