// agnosticui-core@2.0.0-alpha.12/dist/components/Rating/core/_Rating.js
import { LitElement as y, css as w, html as h, svg as g } from "lit";
import { n as o } from "../../../property-CemaeiRl.js";
import { r as p } from "../../../state-CovhUvdr.js";
import { o as f } from "../../../if-defined-C8i28hSj.js";
import { createFormControlIds as $, isHorizontalLabel as m, buildAriaDescribedBy as P } from "../../../shared/form-control-utils.js";
import { formControlStyles as _ } from "../../../shared/form-control-styles.js";
var C = Object.defineProperty, s = (u, e, i, t) => {
  for (var r = void 0, n = u.length - 1, l; n >= 0; n--)
    (l = u[n]) && (r = l(e, i, r) || r);
  return r && C(e, i, r), r;
};
let H = 0;
const v = class v extends y {
  // Flag for pointer down state
  constructor() {
    super(), this.uniqueId = ++H, this._ratingId = "", this._labelId = "", this._helperId = "", this._errorId = "", this.labelPosition = "top", this.hoverValue = 0, this.isHovering = !1, this.isPointerDown = !1;
    const e = $("rating");
    this._ratingId = e.inputId, this._labelId = e.labelId, this._helperId = e.helperId, this._errorId = e.errorId, this.value = 0, this.max = 5, this.precision = "whole", this.readonly = !1, this.allowClear = !1, this.variant = "", this.size = "md", this.name = "", this.label = "", this.labelHidden = !1, this.noLabel = !1, this.required = !1, this.invalid = !1, this.errorMessage = "", this.helpText = "", this.handlePointerMove = this.handlePointerMove.bind(this), this.handlePointerUp = this.handlePointerUp.bind(this), this.handleKeyDown = this.handleKeyDown.bind(this);
  }
  connectedCallback() {
    super.connectedCallback(), this.addEventListener("keydown", this.handleKeyDown);
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this.removeEventListener("keydown", this.handleKeyDown), this.removeGlobalPointerListeners();
  }
  renderLabel() {
    if (!this.label || this.noLabel) return "";
    const e = [];
    return m(this.labelPosition) ? (e.push("ag-form-control__label--horizontal"), e.push(`ag-form-control__label--${this.labelPosition}`)) : this.labelPosition === "bottom" && e.push(`ag-form-control__label--${this.labelPosition}`), h`
      <label
        id=${this._labelId}
        for=${this._ratingId}
        class="ag-form-control__label ${this.labelHidden ? "ag-form-control__label--hidden" : ""} ${this.required ? "ag-form-control__label--required" : ""} ${e.join(" ")}"
        part="ag-rating-label"
      >
        ${this.label}
      </label>
    `;
  }
  render() {
    const e = this.isHovering ? this.hoverValue : this.value, i = Array.from({ length: this.max }, (d, b) => b + 1), t = P({
      helperId: this._helperId,
      errorId: this._errorId,
      hasHelper: !!this.helpText && !this.invalid,
      hasError: !!this.invalid && !!this.errorMessage
    }), r = this.helpText && !this.invalid ? h`<div class="ag-form-control__helper" id="${this._helperId}">
          ${this.helpText}
        </div>` : "", n = this.invalid && this.errorMessage ? h`<div class="ag-form-control__error" id="${this._errorId}">
          ${this.errorMessage}
        </div>` : "", l = h`
      <div
        id="${this._ratingId}"
        class="rating"
        part="base"
        role="slider"
        aria-label="${this.label || "Rating"}"
        aria-labelledby="${f(this.label && !this.noLabel ? this._labelId : void 0)}"
        aria-describedby="${f(t)}"
        aria-valuemin="0"
        aria-valuemax="${this.max}"
        aria-valuenow="${Number(this.value).toFixed(this.precision === "half" ? 1 : 0)}"
        aria-invalid="${this.invalid ? "true" : "false"}"
        aria-required="${this.required ? "true" : "false"}"
        tabindex="${this.readonly ? -1 : 0}"
        @pointerdown="${this.handlePointerDown}"
        @pointermove="${this.handlePointerMoveHost}"
        @pointerleave="${this.handlePointerLeave}"
        @pointerenter="${this.handlePointerEnter}"
      >
        ${i.map((d) => this.renderStar(d, e))}
      </div>
      <span class="visually-hidden" aria-live="polite">
        ${e} of ${this.max}
      </span>
    `;
    return m(this.labelPosition) ? h`
        <div class="ag-form-control--horizontal">
          ${this.renderLabel()}
          ${l}
        </div>
        ${r}
        ${n}
      ` : this.labelPosition === "bottom" ? h`
        ${l}
        ${r}
        ${n}
        ${this.renderLabel()}
      ` : h`
      ${this.renderLabel()}
      ${l}
      ${r}
      ${n}
    `;
  }
  renderStar(e, i) {
    const t = i >= e, r = !t && i >= e - 0.5 && this.precision === "half", n = t || r ? "filled" : "", l = this.isHovering && this.hoverValue >= e ? "hover" : "", c = `ag-rating-half-${this.uniqueId}-${e}`;
    return h`
      <span
        class="star-button"
        part="star-button"
        @click="${(d) => this.handleClickStar(d, e)}"
        aria-label="${e} star"
        title="${e} star"
      >
        <span class="star ${n} ${l}" part="star" data-star="${e}">
          ${this.renderStarSvg(t, r, c)}
        </span>
      </span>
    `;
  }
  renderStarSvg(e, i, t) {
    return i ? g`
        <svg viewBox="0 0 24 24" aria-hidden="true" focusable="false">
          <defs>
            <clipPath id="${t}">
              <rect x="0" y="0" width="12" height="24"></rect>
            </clipPath>
          </defs>
          <path d="M12 .587l3.668 7.431 8.2 1.192-5.934 5.782 1.401 8.174L12 18.896l-7.335 3.86 1.401-8.174L.132 9.21l8.2-1.192z" />
          <path clip-path="url(#${t})" d="M12 .587l3.668 7.431 8.2 1.192-5.934 5.782 1.401 8.174L12 18.896l-7.335 3.86 1.401-8.174L.132 9.21l8.2-1.192z" />
        </svg>
      ` : g`
      <svg viewBox="0 0 24 24" aria-hidden="true" focusable="false">
        <path d="M12 .587l3.668 7.431 8.2 1.192-5.934 5.782 1.401 8.174L12 18.896l-7.335 3.86 1.401-8.174L.132 9.21l8.2-1.192z" />
      </svg>
    `;
  }
  roundToPrecision(e) {
    return this.precision === "half" ? Math.round(e * 2) / 2 : Math.round(e);
  }
  getValueFromClientX(e) {
    const i = this.shadowRoot?.querySelector(".rating");
    if (!i) return 0;
    const t = i.getBoundingClientRect(), r = Math.max(0, Math.min(e - t.left, t.width)), l = (t.width > 0 ? r / t.width : 0) * this.max;
    return this.roundToPrecision(l);
  }
  handleClickStar(e, i) {
    if (e.stopPropagation(), this.readonly) return;
    const t = this.value;
    let r = (this.precision === "half", i);
    this.allowClear && r === t && (r = 0), this.commitValue(r, t);
  }
  handlePointerEnter(e) {
  }
  handlePointerLeave(e) {
    this.isPointerDown || (this.isHovering = !1, this.hoverValue = 0, this.emitHoverEvent("end", this.hoverValue));
  }
  handlePointerDown(e) {
    if (this.readonly) return;
    this.isPointerDown = !0, this.setPointerCapture(e.pointerId);
    const i = e.clientX, t = this.getValueFromClientX(i);
    this.hoverValue = t, this.isHovering = !0, this.emitHoverEvent("start", t), window.addEventListener("pointermove", this.handlePointerMove), window.addEventListener("pointerup", this.handlePointerUp);
  }
  handlePointerMoveHost(e) {
    if (!this.isPointerDown && !this.isHovering) return;
    const i = e.clientX, t = this.getValueFromClientX(i);
    this.hoverValue = t, this.isHovering ? this.emitHoverEvent("move", t) : (this.isHovering = !0, this.emitHoverEvent("start", t));
  }
  handlePointerMove(e) {
    if (this.readonly) return;
    const i = e.clientX, t = this.getValueFromClientX(i);
    t !== this.hoverValue && (this.hoverValue = t, this.emitHoverEvent("move", t));
  }
  handlePointerUp(e) {
    if (this.readonly) return;
    const i = e.clientX, t = this.getValueFromClientX(i), r = this.value;
    let n = t;
    this.allowClear && n === r && (n = 0), this.commitValue(n, r), this.isPointerDown = !1, this.isHovering = !1, this.hoverValue = 0, this.emitHoverEvent("end", t), this.removeGlobalPointerListeners();
  }
  removeGlobalPointerListeners() {
    window.removeEventListener("pointermove", this.handlePointerMove), window.removeEventListener("pointerup", this.handlePointerUp);
  }
  handleKeyDown(e) {
    if (this.readonly) return;
    const i = this.value, t = this.precision === "half" ? 0.5 : 1;
    e.key === "ArrowRight" || e.key === "ArrowUp" ? (e.preventDefault(), this.value = Math.min(this.max, this.value + t), this.commitValue(this.value, i)) : e.key === "ArrowLeft" || e.key === "ArrowDown" ? (e.preventDefault(), this.value = Math.max(0, this.value - t), this.commitValue(this.value, i)) : e.key === "Home" ? (e.preventDefault(), this.value = 0, this.commitValue(this.value, i)) : e.key === "End" && (e.preventDefault(), this.value = this.max, this.commitValue(this.value, i));
  }
  commitValue(e, i) {
    const t = this.roundToPrecision(e);
    this.value = t;
    const r = new CustomEvent("rating-change", {
      detail: { oldValue: i, newValue: t },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(r), this.onRatingChange && this.onRatingChange(r);
  }
  emitHoverEvent(e, i) {
    const t = new CustomEvent("rating-hover", {
      detail: { phase: e, value: i },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(t), this.onRatingHover && this.onRatingHover(t);
  }
};
v.styles = [
  _,
  w`
    :host {
      display: block;
      line-height: 1;
    }

    .rating {
      display: inline-flex;
      gap: var(--ag-space-1);
      align-items: center;
      cursor: pointer;
    }

    :host([readonly]) .rating {
      cursor: default;
      pointer-events: none;
    }

    /* Default (≈ Chakra UI “xs”) */
    .star {
      display: inline-flex;
      align-items: center;
      justify-content: center;
      transition:
        transform var(--ag-motion-medium) ease,
        opacity   var(--ag-motion-medium);
      color: var(--ag-neutral-300);
      width:  var(--ag-space-3);   /* 0.75rem  → xs */
      height: var(--ag-space-3);
    }

    /* size="sm"  → Chakra UI “sm” ≈ 1rem */
    :host([size="sm"]) .star {
      width:  var(--ag-space-4);   /* 1rem */
      height: var(--ag-space-4);
    }

    /* size="md"  → Chakra UI “md” ≈ 1.25rem */
    :host([size="md"]) .star {
      width:  var(--ag-space-5);   /* 1.25rem */
      height: var(--ag-space-5);
    }

    /* size="lg"  → Chakra UI “lg” ≈ 1.5rem */
    :host([size="lg"]) .star {
      width:  var(--ag-space-6);   /* 1.5rem */
      height: var(--ag-space-6);
    }

    .star svg path {
      fill: var(--ag-neutral-300); /* Empty color */
    }
    .star.filled > svg > path:last-of-type,
    .star.hover svg path {
      fill: var(--ag-rating-filled, var(--ag-yellow-400));
    }

    :host([variant="primary"]) .star.filled > svg > path:last-of-type,
    :host([variant="primary"]) .star.hover svg path {
      fill: var(--ag-rating-filled-primary, var(--ag-primary));
    }
    :host([variant="success"]) .star.filled > svg > path:last-of-type,
    :host([variant="success"]) .star.hover svg path {
      fill: var(--ag-rating-filled-success, var(--ag-success));
    }
    :host([variant="warning"]) .star.filled > svg > path:last-of-type,
    :host([variant="warning"]) .star.hover svg path {
      fill: var(--ag-rating-filled-warning, var(--ag-warning));
    }
    :host([variant="danger"]) .star.filled > svg > path:last-of-type,
    :host([variant="danger"]) .star.hover svg path {
      fill: var(--ag-rating-filled-danger, var(--ag-danger));
    }
    :host([variant="secondary"]) .star.filled > svg > path:last-of-type,
    :host([variant="secondary"]) .star.hover svg path {
      fill: var(--ag-rating-filled-secondary, var(--ag-secondary));
    }
    :host([variant="monochrome"]) .star.filled > svg > path:last-of-type,
    :host([variant="monochrome"]) .star.hover svg path {
      fill: var(--ag-text-primary);
    }

    .star-button {
      display: inline-block;
      border: 0;
      background: transparent;
      padding: 0;
      margin: 0;
      line-height: 0;
      cursor: inherit;
    }

    :host(:focus-visible) .rating {
      box-shadow: 0 0 0 var(--ag-focus-offset) rgba(var(--ag-focus), 0.12);
    }

    .visually-hidden {
      position: absolute !important;
      width: 1px;
      height: 1px;
      padding: 0;
      margin: -1px;
      overflow: hidden;
      clip: rect(0 0 0 0);
      white-space: nowrap;
      border: 0;
    }
  `
];
let a = v;
s([
  o({ type: Number })
], a.prototype, "value");
s([
  o({ type: Number })
], a.prototype, "max");
s([
  o({ type: String })
], a.prototype, "precision");
s([
  o({ type: Boolean, reflect: !0 })
], a.prototype, "readonly");
s([
  o({ type: Boolean })
], a.prototype, "allowClear");
s([
  o({ type: String, reflect: !0 })
], a.prototype, "variant");
s([
  o({ type: String, reflect: !0 })
], a.prototype, "size");
s([
  o({ type: String })
], a.prototype, "name");
s([
  o({ type: String })
], a.prototype, "label");
s([
  o({ type: String, attribute: "label-position" })
], a.prototype, "labelPosition");
s([
  o({ type: Boolean, attribute: "label-hidden" })
], a.prototype, "labelHidden");
s([
  o({ type: Boolean, attribute: "no-label" })
], a.prototype, "noLabel");
s([
  o({ type: Boolean, reflect: !0 })
], a.prototype, "required");
s([
  o({ type: Boolean, reflect: !0 })
], a.prototype, "invalid");
s([
  o({ type: String, attribute: "error-message" })
], a.prototype, "errorMessage");
s([
  o({ type: String, attribute: "help-text" })
], a.prototype, "helpText");
s([
  o({ attribute: !1 })
], a.prototype, "onRatingChange");
s([
  o({ attribute: !1 })
], a.prototype, "onRatingHover");
s([
  p()
], a.prototype, "hoverValue");
s([
  p()
], a.prototype, "isHovering");
s([
  p()
], a.prototype, "isPointerDown");
export {
  a as AgRating
};
