// agnosticui-core@2.0.0-alpha.12/dist/components/Flex/vue/VueFlexContainer.js
import { defineComponent as a, ref as i, onMounted as o, nextTick as l, watchEffect as f, createElementBlock as s, openBlock as c, mergeProps as u, renderSlot as d } from "vue";
import "../core/FlexContainer.js";
import { _ as p } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const g = a({
  name: "VueFlexContainer",
  props: {
    direction: {
      type: String,
      default: "row"
    },
    wrap: {
      type: String,
      default: "nowrap"
    },
    justify: {
      type: String,
      default: "flex-start"
    },
    align: {
      type: String,
      default: "stretch"
    },
    alignContent: {
      type: String,
      default: "stretch"
    },
    inline: {
      type: Boolean,
      default: !1
    },
    reverse: {
      type: Boolean,
      default: !1
    },
    stretchChildren: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const n = i(null), r = () => {
      const t = n.value;
      t && (t.direction = e.direction, t.wrap = e.wrap, t.justify = e.justify, t.align = e.align, t.alignContent = e.alignContent, t.inline = e.inline, t.reverse = e.reverse, t.stretchChildren = e.stretchChildren);
    };
    return o(async () => {
      await customElements.whenDefined("ag-flex-container"), await l(), r();
    }), f(() => {
      n.value && r();
    }), { agComponent: n };
  }
});
function m(e, n, r, t, y, C) {
  return c(), s("ag-flex-container", u({ ref: "agComponent" }, e.$attrs), [
    d(e.$slots, "default")
  ], 16);
}
const _ = /* @__PURE__ */ p(g, [["render", m]]);
export {
  _ as default
};
