// agnosticui-core@2.0.0-alpha.12/dist/components/Flex/vue/VueFlexInline.js
import { defineComponent as a, ref as i, onMounted as l, nextTick as o, watchEffect as s, createElementBlock as f, openBlock as c, mergeProps as u, renderSlot as d } from "vue";
import "../core/FlexInline.js";
import { _ as p } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const g = a({
  name: "VueFlexInline",
  props: {
    direction: {
      type: String,
      default: "row"
    },
    wrap: {
      type: String,
      default: "nowrap"
    },
    justify: {
      type: String,
      default: "flex-start"
    },
    align: {
      type: String,
      default: "stretch"
    },
    alignContent: {
      type: String,
      default: "stretch"
    },
    reverse: {
      type: Boolean,
      default: !1
    },
    stretchChildren: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const n = i(null), r = () => {
      const t = n.value;
      t && (t.direction = e.direction, t.wrap = e.wrap, t.justify = e.justify, t.align = e.align, t.alignContent = e.alignContent, t.reverse = e.reverse, t.stretchChildren = e.stretchChildren);
    };
    return l(async () => {
      await customElements.whenDefined("ag-flex-inline"), await o(), r();
    }), s(() => {
      n.value && r();
    }), { agComponent: n };
  }
});
function m(e, n, r, t, y, h) {
  return c(), f("ag-flex-inline", u({ ref: "agComponent" }, e.$attrs), [
    d(e.$slots, "default")
  ], 16);
}
const _ = /* @__PURE__ */ p(g, [["render", m]]);
export {
  _ as default
};
