// agnosticui-core@2.0.0-alpha.12/dist/components/Flex/vue/VueFlexCol.js
import { defineComponent as a, ref as o, onMounted as l, nextTick as s, watchEffect as f, createElementBlock as i, openBlock as c, mergeProps as u, renderSlot as p } from "vue";
import "../core/FlexCol.js";
import { _ as d } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const g = a({
  name: "VueFlexCol",
  props: {
    wrap: {
      type: String,
      default: "nowrap"
    },
    justify: {
      type: String,
      default: "flex-start"
    },
    align: {
      type: String,
      default: "stretch"
    },
    alignContent: {
      type: String,
      default: "stretch"
    },
    reverse: {
      type: Boolean,
      default: !1
    },
    stretchChildren: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const n = o(null), r = () => {
      const t = n.value;
      t && (t.wrap = e.wrap, t.justify = e.justify, t.align = e.align, t.alignContent = e.alignContent, t.reverse = e.reverse, t.stretchChildren = e.stretchChildren);
    };
    return l(async () => {
      await customElements.whenDefined("ag-flex-col"), await s(), r();
    }), f(() => {
      n.value && r();
    }), { agComponent: n };
  }
});
function m(e, n, r, t, C, h) {
  return c(), i("ag-flex-col", u({ ref: "agComponent" }, e.$attrs), [
    p(e.$slots, "default")
  ], 16);
}
const _ = /* @__PURE__ */ d(g, [["render", m]]);
export {
  _ as default
};
