// agnosticui-core@2.0.0-alpha.12/dist/components/Divider/vue/VueDivider.js
import { defineComponent as i, ref as s, onMounted as c, nextTick as r, watchEffect as f, createElementBlock as l, openBlock as u, mergeProps as d, renderSlot as p } from "vue";
import "../core/Divider.js";
import { _ as m } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const v = i({
  name: "VueDivider",
  props: {
    vertical: { type: Boolean, default: !1 },
    justify: {
      type: String,
      default: "center"
    },
    size: {
      type: String,
      default: "default"
    },
    variant: {
      type: String,
      default: "default"
    }
  },
  setup(t) {
    const n = s(null), a = () => {
      const e = n.value;
      e && (e.vertical = t.vertical, e.justify = t.justify, e.size = t.size, e.variant = t.variant);
    };
    return c(async () => {
      await customElements.whenDefined("ag-divider"), await r(), a(), await r();
      const e = n.value;
      if (e) {
        const o = e.shadowRoot?.querySelector("slot");
        o && o.dispatchEvent(new Event("slotchange"));
      }
    }), f(() => {
      n.value && a();
    }), { agComponent: n };
  }
});
function y(t, n, a, e, o, g) {
  return u(), l("ag-divider", d({ ref: "agComponent" }, t.$attrs), [
    p(t.$slots, "default")
  ], 16);
}
const $ = /* @__PURE__ */ m(v, [["render", y]]);
export {
  $ as default
};
