// agnosticui-core@2.0.0-alpha.12/dist/components/ScrollToButton/vue/VueScrollToButton.js
import { defineComponent as l, createElementBlock as o, openBlock as t, createCommentVNode as s, renderSlot as r } from "vue";
import "../core/ScrollToButton.js";
import { _ as a } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const n = l({
  name: "VueScrollToButton",
  props: {
    /**
     * Accessible label for the button
     */
    label: {
      type: String,
      default: "Back to Top"
    },
    /**
     * Show the label text alongside the icon
     */
    showLabel: {
      type: Boolean,
      default: !1
    },
    /**
     * Show/hide the default icon
     */
    icon: {
      type: Boolean,
      default: !0
    },
    /**
     * Scroll threshold in pixels before showing the button
     */
    scrollThreshold: {
      type: Number,
      default: 400
    },
    /**
     * Scroll target: "top", "bottom", or element ID/selector
     */
    target: {
      type: String,
      default: "top"
    },
    /**
     * Arrow direction (auto-detects by default)
     */
    direction: {
      type: String,
      default: "auto"
    },
    /**
     * Enable smooth scrolling animation
     */
    smoothScroll: {
      type: Boolean,
      default: !0
    },
    /**
     * Controls button visibility (usually controlled internally by scroll position)
     */
    visible: {
      type: Boolean,
      default: !1
    },
    /**
     * Button size
     */
    size: {
      type: String,
      default: "md"
    },
    /**
     * Button shape
     */
    shape: {
      type: String,
      default: ""
    }
  }
}), i = ["label", "showLabel", "icon", "scrollThreshold", "target", "direction", "smoothScroll", "visible", "size", "shape"], p = {
  key: 0,
  slot: "icon"
};
function d(e, c, u, h, f, m) {
  return t(), o("ag-scroll-to-button", {
    label: e.label,
    showLabel: e.showLabel,
    icon: e.icon,
    scrollThreshold: e.scrollThreshold,
    target: e.target,
    direction: e.direction,
    smoothScroll: e.smoothScroll,
    visible: e.visible,
    size: e.size,
    shape: e.shape
  }, [
    e.$slots.icon ? (t(), o("span", p, [
      r(e.$slots, "icon")
    ])) : s("", !0)
  ], 8, i);
}
const g = /* @__PURE__ */ a(n, [["render", d]]);
export {
  g as default
};
