// agnosticui-core@2.0.0-alpha.12/dist/components/Avatar/vue/VueAvatar.js
import { defineComponent as r, ref as i, onMounted as o, nextTick as l, watchEffect as c, createElementBlock as f, openBlock as s, mergeProps as u, renderSlot as m } from "vue";
import "../core/Avatar.js";
import { _ as d } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const g = r({
  name: "VueAvatar",
  props: {
    text: { type: String, default: "" },
    imgSrc: { type: String, default: "" },
    imgAlt: { type: String, default: "" },
    size: {
      type: String,
      default: "md"
    },
    shape: {
      type: String,
      default: "circle"
    },
    variant: {
      type: String,
      default: "default"
    },
    ariaLabel: { type: String, default: "" }
  },
  setup(t) {
    const a = i(null), n = () => {
      const e = a.value;
      e && (e.text = t.text, e.imgSrc = t.imgSrc, e.imgAlt = t.imgAlt, e.size = t.size, e.shape = t.shape, e.variant = t.variant, e.ariaLabel = t.ariaLabel);
    };
    return o(async () => {
      await customElements.whenDefined("ag-avatar"), await l(), n();
    }), c(() => {
      a.value && n();
    }), { agComponent: a };
  }
});
function p(t, a, n, e, S, v) {
  return s(), f("ag-avatar", u({ ref: "agComponent" }, t.$attrs), [
    m(t.$slots, "default")
  ], 16);
}
const $ = /* @__PURE__ */ d(g, [["render", p]]);
export {
  $ as default
};
