// agnosticui-core@2.0.0-alpha.12/dist/components/SelectionCardGroup/core/_SelectionCardGroup.js
import { LitElement as u, css as g, nothing as f, html as p } from "lit";
import { n as o } from "../../../property-CemaeiRl.js";
import { r as m } from "../../../state-CovhUvdr.js";
var v = Object.defineProperty, r = (c, e, t, s) => {
  for (var a = void 0, i = c.length - 1, n; i >= 0; i--)
    (n = c[i]) && (a = n(e, t, a) || a);
  return a && v(e, t, a), a;
};
const d = class d extends u {
  constructor() {
    super(), this._handleCardChange = (e) => {
      e.stopPropagation();
      const { value: t, checked: s } = e.detail;
      let a;
      if (this.type === "radio")
        a = s ? [t] : [];
      else {
        const n = [...this._getSelectedValues()];
        if (s)
          n.includes(t) || n.push(t);
        else {
          const h = n.indexOf(t);
          h > -1 && n.splice(h, 1);
        }
        a = n;
      }
      this._internalSelectedValues = a;
      const i = new CustomEvent("selection-change", {
        detail: {
          value: t,
          checked: s,
          selectedValues: a
        },
        bubbles: !0,
        composed: !0
      });
      this.dispatchEvent(i), this.onSelectionChange && this.onSelectionChange(i);
    }, this._handleKeyDown = (e) => {
      const t = this._getCards().filter((n) => !n.disabled);
      if (t.length === 0) return;
      const s = t.find((n) => {
        try {
          return n.shadowRoot?.activeElement || n === document.activeElement;
        } catch {
          return n === document.activeElement;
        }
      }), a = s ? t.indexOf(s) : -1;
      let i = null;
      switch (e.key) {
        case "ArrowDown":
        case "ArrowRight":
          e.preventDefault(), i = a === -1 ? 0 : (a + 1) % t.length;
          break;
        case "ArrowUp":
        case "ArrowLeft":
          e.preventDefault(), i = a === -1 ? t.length - 1 : (a - 1 + t.length) % t.length;
          break;
        case "Home":
          e.preventDefault(), i = 0;
          break;
        case "End":
          e.preventDefault(), i = t.length - 1;
          break;
      }
      if (i !== null) {
        const n = t[i];
        n.focus(), this.type === "radio" && this._handleCardChange(new CustomEvent("selection-card-change", {
          detail: { value: n.value, checked: !0 }
        }));
      }
    }, this.type = "radio", this.name = "", this.legend = "", this.legendHidden = !1, this.theme = "", this.value = "", this.values = [], this.disabled = !1, this._internalSelectedValues = [];
  }
  // Get current selected values (controlled or uncontrolled)
  _getSelectedValues() {
    return this.type === "radio" ? this.value ? [this.value] : this._internalSelectedValues : this.values && this.values.length > 0 ? this.values : this._internalSelectedValues;
  }
  connectedCallback() {
    super.connectedCallback(), this.addEventListener("selection-card-change", this._handleCardChange), this.addEventListener("keydown", this._handleKeyDown);
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this.removeEventListener("selection-card-change", this._handleCardChange), this.removeEventListener("keydown", this._handleKeyDown);
  }
  updated(e) {
    super.updated(e), (e.has("type") || e.has("name") || e.has("theme") || e.has("disabled") || e.has("value") || e.has("values") || e.has("_internalSelectedValues")) && this._syncChildCards();
  }
  firstUpdated() {
    this._syncChildCards();
  }
  _getCards() {
    const e = this.shadowRoot?.querySelector("slot");
    return e ? e.assignedElements({ flatten: !0 }).filter((t) => t.tagName.toLowerCase() === "ag-selection-card") : [];
  }
  _syncChildCards() {
    const e = this._getCards(), t = this._getSelectedValues();
    e.forEach((s) => {
      s._type = this.type, s._name = this.name, s._theme = this.theme, s.checked = t.includes(s.value), this.disabled && (s.disabled = !0);
    });
  }
  _handleSlotChange() {
    this._syncChildCards();
  }
  render() {
    const e = [
      "selection-card-group__legend",
      this.legendHidden ? "selection-card-group__legend--hidden" : ""
    ].filter(Boolean).join(" ");
    return p`
      <fieldset
        class="selection-card-group"
        part="ag-selection-card-group-fieldset"
        role="${this.type === "radio" ? "radiogroup" : "group"}"
        aria-disabled="${this.disabled ? "true" : "false"}"
      >
        ${this.legend ? p`<legend class="${e}" part="ag-selection-card-group-legend">${this.legend}</legend>` : f}
        <div class="selection-card-group__content" part="ag-selection-card-group-content">
          <slot @slotchange="${this._handleSlotChange}"></slot>
        </div>
      </fieldset>
    `;
  }
};
d.styles = g`
    :host {
      display: block;
    }

    .selection-card-group {
      border: 0;
      padding: 0;
      margin: 0;
      min-inline-size: auto;
    }

    .selection-card-group__legend {
      font-weight: 600;
      font-size: var(--ag-font-size-base);
      color: var(--ag-text-primary);
      padding: 0;
      margin-block-end: var(--ag-space-4);
    }

    .selection-card-group__legend--hidden {
      position: absolute !important;
      width: 1px !important;
      height: 1px !important;
      padding: 0 !important;
      margin: -1px !important;
      overflow: hidden !important;
      clip: rect(0, 0, 0, 0) !important;
      white-space: nowrap !important;
      border: 0 !important;
    }

    .selection-card-group__content {
      display: grid;
      grid-template-columns: repeat(auto-fit, minmax(150px, 1fr));
      gap: var(--ag-selection-card-group-gap, var(--ag-space-4));
    }
  `;
let l = d;
r([
  o({ type: String, reflect: !0 })
], l.prototype, "type");
r([
  o({ type: String, reflect: !0 })
], l.prototype, "name");
r([
  o({ type: String })
], l.prototype, "legend");
r([
  o({ type: Boolean, attribute: "legend-hidden" })
], l.prototype, "legendHidden");
r([
  o({ type: String, reflect: !0 })
], l.prototype, "theme");
r([
  o({ type: String })
], l.prototype, "value");
r([
  o({ type: Array })
], l.prototype, "values");
r([
  o({ type: Boolean, reflect: !0 })
], l.prototype, "disabled");
r([
  o({ attribute: !1 })
], l.prototype, "onSelectionChange");
r([
  m()
], l.prototype, "_internalSelectedValues");
export {
  l as AgSelectionCardGroup
};
