// agnosticui-core@2.0.0-alpha.12/dist/components/Breadcrumb/svelte/SvelteBreadcrumb.js
import { e as M, r as Y, u as y, q, S as A, a as L, s as z, g as C, p as U, P as B, b as u, c as G, d as w, D as K, f as $, h as j, i as I, j as R, k as F, l as Z, m as H, n as J, o as Q, t as V, L as W, v as D, w as X, x as k, y as ee, z as re, A as ae, B as ie, C as se, E as te, F as ne, G as ue } from "../../../lifecycle-BbESl1Js.js";
import { createEventDispatcher as fe, onMount as ce, onDestroy as le } from "svelte";
import "../core/_Breadcrumb.js";
function O(e, r) {
  return e === r || e?.[A] === r;
}
function de(e = {}, r, n, f) {
  return M(() => {
    var s, t;
    return Y(() => {
      s = t, t = [], y(() => {
        e !== n(...t) && (r(e, ...t), s && O(n(...s), e) && r(null, ...s));
      });
    }), () => {
      q(() => {
        t && O(n(...t), e) && r(null, ...t);
      });
    };
  }), e;
}
let S = !1;
function _e(e) {
  var r = S;
  try {
    return S = !1, [e(), S];
  } finally {
    S = r;
  }
}
const ve = {
  get(e, r) {
    if (!e.exclude.includes(r))
      return u(e.version), r in e.special ? e.special[r]() : e.props[r];
  },
  set(e, r, n) {
    if (!(r in e.special)) {
      var f = L;
      try {
        R(e.parent_effect), e.special[r] = p(
          {
            get [r]() {
              return e.props[r];
            }
          },
          /** @type {string} */
          r,
          B
        );
      } finally {
        R(f);
      }
    }
    return e.special[r](n), I(e.version), !0;
  },
  getOwnPropertyDescriptor(e, r) {
    if (!e.exclude.includes(r) && r in e.props)
      return {
        enumerable: !0,
        configurable: !0,
        value: e.props[r]
      };
  },
  deleteProperty(e, r) {
    return e.exclude.includes(r) || (e.exclude.push(r), I(e.version)), !0;
  },
  has(e, r) {
    return e.exclude.includes(r) ? !1 : r in e.props;
  },
  ownKeys(e) {
    return Reflect.ownKeys(e.props).filter((r) => !e.exclude.includes(r));
  }
};
function g(e, r) {
  return new Proxy(
    {
      props: e,
      exclude: r,
      special: {},
      version: z(0),
      // TODO this is only necessary because we need to track component
      // destruction inside `prop`, because of `bind:this`, but it
      // seems likely that we can simplify `bind:this` instead
      parent_effect: (
        /** @type {Effect} */
        L
      )
    },
    ve
  );
}
function p(e, r, n, f) {
  var s = !Z || (n & H) !== 0, t = (n & F) !== 0, h = (n & Q) !== 0, l = (
    /** @type {V} */
    f
  ), v = !0, a = () => (v && (v = !1, l = h ? y(
    /** @type {() => V} */
    f
  ) : (
    /** @type {V} */
    f
  )), l), d;
  if (t) {
    var o = A in e || W in e;
    d = C(e, r)?.set ?? (o && r in e ? (i) => e[r] = i : void 0);
  }
  var c, m = !1;
  t ? [c, m] = _e(() => (
    /** @type {V} */
    e[r]
  )) : c = /** @type {V} */
  e[r], c === void 0 && f !== void 0 && (c = a(), d && (s && U(), d(c)));
  var _;
  if (s ? _ = () => {
    var i = (
      /** @type {V} */
      e[r]
    );
    return i === void 0 ? a() : (v = !0, i);
  } : _ = () => {
    var i = (
      /** @type {V} */
      e[r]
    );
    return i !== void 0 && (l = /** @type {V} */
    void 0), i === void 0 ? l : i;
  }, s && (n & B) === 0)
    return _;
  if (d) {
    var N = e.$$legacy;
    return (
      /** @type {() => V} */
      (function(i, P) {
        return arguments.length > 0 ? ((!s || !P || N || m) && d(P ? _() : i), i) : _();
      })
    );
  }
  var E = !1, b = ((n & J) !== 0 ? $ : j)(() => (E = !1, _()));
  t && u(b);
  var T = (
    /** @type {Effect} */
    L
  );
  return (
    /** @type {() => V} */
    (function(i, P) {
      if (arguments.length > 0) {
        const x = P ? u(b) : s && t ? G(i) : i;
        return w(b, x), E = !0, l !== void 0 && (l = x), i;
      }
      return V && E || (T.f & K) !== 0 ? b.v : u(b);
    })
  );
}
var oe = re("<ag-breadcrumb></ag-breadcrumb>", 2);
function Pe(e, r) {
  const n = g(r, ["children", "$$slots", "$$events", "$$legacy"]), f = g(n, ["items", "type", "ariaLabel", "className"]);
  te(r, !1);
  let s = p(r, "items", 24, () => []), t = p(r, "type", 8, "default"), h = p(r, "ariaLabel", 8, "Breadcrumb"), l = p(r, "className", 8, "");
  const v = fe();
  let a = ne();
  function d(c) {
    const m = c.detail;
    v("breadcrumb-click", m);
  }
  ce(() => {
    u(a) && (D(a, u(a).items = s()), u(a).addEventListener("breadcrumb-click", d));
  }), le(() => {
    u(a) && u(a).removeEventListener("breadcrumb-click", d);
  }), X(() => (u(a), ue(s())), () => {
    u(a) && D(a, u(a).items = s());
  }), k(), ee();
  var o = oe();
  ae(o, () => ({
    type: t(),
    "aria-label": h(),
    class: l(),
    ...f
  })), de(o, (c) => w(a, c), () => u(a)), ie(e, o), se();
}
export {
  Pe as default
};
