// agnosticui-core@2.0.0-alpha.12/dist/shared/form-control-utils.js
import { nothing as a, html as l } from "lit";
import { generateUniqueId as n } from "../utils/unique-id.js";
function i(r) {
  const e = n(r);
  return {
    inputId: `${e}-input`,
    labelId: `${e}-label`,
    helperId: `${e}-helper`,
    errorId: `${e}-error`
  };
}
function u(r) {
  const e = [];
  return r.hasHelper && e.push(r.helperId), r.hasError && e.push(r.errorId), e.length > 0 ? e.join(" ") : void 0;
}
function t(r) {
  return r === "start" || r === "end";
}
function c(r) {
  return r === "top" || r === "bottom";
}
function f(r) {
  if (!r.label) return a;
  const e = [];
  return r.position && t(r.position) ? (e.push("ag-form-control__label--horizontal"), e.push(`ag-form-control__label--${r.position}`)) : r.position === "bottom" && e.push(`ag-form-control__label--${r.position}`), l`
    <label
      id="${r.labelId}"
      for="${r.inputId}"
      class="ag-form-control__label ${r.hidden ? "ag-form-control__label--hidden" : ""} ${e.join(" ")}"
      part="ag-label"
    >
      ${r.label}
      ${r.required ? l`
        <span class="ag-form-control__required" part="ag-required" aria-hidden="true">*</span>
      ` : a}
    </label>
  `;
}
function h(r) {
  return r.helpText ? l`
    <div
      id="${r.helperId}"
      class="ag-form-control__helper"
      part="ag-helper"
    >
      ${r.helpText}
    </div>
  ` : a;
}
function b(r) {
  return l`
    <div
      id="${r.errorId}"
      class="ag-form-control__error"
      part="ag-error"
      ?hidden="${!r.invalid || !r.errorMessage}"
    >
      ${r.errorMessage || ""}
    </div>
  `;
}
export {
  u as buildAriaDescribedBy,
  i as createFormControlIds,
  t as isHorizontalLabel,
  c as isVerticalLabel,
  b as renderFormError,
  h as renderFormHelper,
  f as renderFormLabel
};
