// agnosticui-core@2.0.0-alpha.12/dist/components/Tooltip/styled/MinimalTooltip.js
import { LitElement as g, css as v, html as _ } from "lit";
import { n as r } from "../../../property-CemaeiRl.js";
import { r as b } from "../../../state-CovhUvdr.js";
import { e as m } from "../../../query-BykXNUlT.js";
import { a as w, c as y, o as E, f as x, s as k, b as L } from "../../../floating-ui.dom-muqvZJIo.js";
var C = Object.defineProperty, i = (d, t, o, n) => {
  for (var s = void 0, a = d.length - 1, l; a >= 0; a--)
    (l = d[a]) && (s = l(t, o, s) || s);
  return s && C(t, o, s), s;
};
const h = class h extends g {
  constructor() {
    super(), this._handleMouseEnter = () => {
      this.disabled || (clearTimeout(this._hideTimeout), this.show());
    }, this._handleMouseLeave = () => {
      this.disabled || (this._hideTimeout = window.setTimeout(() => {
        this.hide();
      }, 100));
    }, this._handleFocus = () => {
      this.disabled || this.show();
    }, this._handleBlur = () => {
      this.disabled || this.hide();
    }, this._handleClick = () => {
      this.disabled || (this._open ? this.hide() : this.show());
    }, this._handleDocumentKeyDown = (t) => {
      t.key === "Escape" && this._open && this.hide();
    }, this.show = () => {
      this._open || this.disabled || (this._open = !0);
    }, this.hide = () => {
      this._open && (this._open = !1);
    }, this.toggle = () => {
      this._open ? this.hide() : this.show();
    }, this.content = "", this.placement = "top", this.distance = 8, this.skidding = 0, this.trigger = "hover focus", this.disabled = !1, this._open = !1;
  }
  firstUpdated() {
    this._setupEventListeners(), this._setupARIA();
  }
  _setupEventListeners() {
    const t = this.firstElementChild;
    t && (this.trigger.includes("hover") && (this.addEventListener("mouseenter", this._handleMouseEnter), this.addEventListener("mouseleave", this._handleMouseLeave)), this.trigger.includes("focus") && (t.addEventListener("focus", this._handleFocus), t.addEventListener("blur", this._handleBlur)), this.trigger.includes("click") && t.addEventListener("click", this._handleClick));
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this._cleanup?.(), this._removeEventListeners(), document.removeEventListener("keydown", this._handleDocumentKeyDown);
  }
  _setupARIA() {
    const t = this.firstElementChild;
    t && t.setAttribute("aria-describedby", "tooltip");
  }
  _removeEventListeners() {
    const t = this.firstElementChild;
    t && (this.removeEventListener("mouseenter", this._handleMouseEnter), this.removeEventListener("mouseleave", this._handleMouseLeave), t.removeEventListener("focus", this._handleFocus), t.removeEventListener("blur", this._handleBlur), t.removeEventListener("click", this._handleClick));
  }
  updated(t) {
    super.updated(t), t.has("_open") && (this._open ? (document.addEventListener("keydown", this._handleDocumentKeyDown), this.updateComplete.then(() => {
      this._startPositioning();
    }), this.dispatchEvent(new CustomEvent("show"))) : (this._stopPositioning(), document.removeEventListener("keydown", this._handleDocumentKeyDown), this.dispatchEvent(new CustomEvent("hide"))));
  }
  _startPositioning() {
    const t = this.shadowRoot?.querySelector("#tooltip"), o = this.shadowRoot?.querySelector("#arrow");
    if (!t || !o) return;
    const n = this.firstElementChild;
    n && (this._updatePosition(), this._cleanup = w(n, t, () => {
      this._updatePosition();
    }));
  }
  _stopPositioning() {
    this._cleanup?.(), this._cleanup = void 0;
  }
  async _updatePosition() {
    const t = this.shadowRoot?.querySelector("#tooltip"), o = this.shadowRoot?.querySelector("#arrow");
    if (!t || !o) return;
    const n = this.firstElementChild;
    if (!n) return;
    const { x: s, y: a, placement: l, middlewareData: u } = await y(
      n,
      t,
      {
        placement: this.placement,
        middleware: [
          E({ mainAxis: this.distance, crossAxis: this.skidding }),
          x(),
          k({ padding: 8 }),
          L({ element: o })
        ]
      }
    );
    Object.assign(t.style, {
      left: `${s}px`,
      top: `${a}px`
    });
    const { x: p, y: c } = u.arrow ?? {}, f = {
      top: "bottom",
      right: "left",
      bottom: "top",
      left: "right"
    }[l.split("-")[0]];
    Object.assign(o.style, {
      left: p != null ? `${p}px` : "",
      top: c != null ? `${c}px` : "",
      right: "",
      bottom: "",
      [f]: "-4px"
    });
  }
  render() {
    return _`
      <slot></slot>
      <div
        id="tooltip"
        part="base"
        role="tooltip"
        class="tooltip"
        ?data-show=${this._open}
        @mouseenter="${() => clearTimeout(this._hideTimeout)}"
        @mouseleave="${this.hide}"
      >
        <slot name="content">${this.content}</slot>
        <div id="arrow" part="arrow"></div>
      </div>
    `;
  }
};
h.styles = v`
    :host {
      display: inline-block;
    }

    #tooltip {
      display: none;
      position: absolute;
      z-index: var(--ag-z-index-modal, 1000);
      pointer-events: none;
      opacity: 0;
      transition: opacity var(--ag-motion-fast, 0.15s) ease-in-out;

      /* Minimal styling */
      background: var(--tooltip-bg, #1f2937);
      color: var(--tooltip-color, #ffffff);
      padding: 8px 12px;
      border-radius: 6px;
      font-size: 14px;
      font-weight: 500;
      max-inline-size: 250px;
      box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
      line-height: 1.4;
      text-align: center;
    }

    #tooltip[data-show] {
      display: block;
      opacity: 1;
      pointer-events: auto;
    }

    #arrow {
      position: absolute;
      background: var(--tooltip-bg, #1f2937);
      width: var(--ag-space-2, 0.5rem);
      height: var(--ag-space-2, 0.5rem);
      transform: rotate(45deg);
      z-index: -1;
    }

    /* Theme support using CSS variables from playground */
    [data-theme="dark"] minimal-tooltip #tooltip,
    :host-context([data-theme="dark"]) #tooltip {
      --tooltip-bg: #374151;
      --tooltip-color: #f9fafb;
      --tooltip-border: #4b5563;
    }

    [data-theme="light"] minimal-tooltip #tooltip,
    :host-context([data-theme="light"]) #tooltip,
    #tooltip {
      --tooltip-bg: #1f2937;
      --tooltip-color: #ffffff;
      --tooltip-border: #374151;
    }

    /* System preference fallback */
    @media (prefers-color-scheme: dark) {
      :root:not([data-theme]) minimal-tooltip #tooltip,
      :host-context(:root:not([data-theme])) #tooltip {
        --tooltip-bg: #374151;
        --tooltip-color: #f9fafb;
        --tooltip-border: #4b5563;
      }
    }

    /* High contrast mode */
    @media (prefers-contrast: high) {
      #tooltip {
        border: 1px solid var(--tooltip-color, #ffffff);
        font-weight: 600;
      }
    }

    /* Reduced motion support */
    @media (prefers-reduced-motion: reduce) {
      #tooltip {
        transition: none;
      }
    }
  `;
let e = h;
i([
  r({ type: String })
], e.prototype, "content");
i([
  r({ type: String, reflect: !0 })
], e.prototype, "placement");
i([
  r({ type: Number })
], e.prototype, "distance");
i([
  r({ type: Number })
], e.prototype, "skidding");
i([
  r({ type: String })
], e.prototype, "trigger");
i([
  r({ type: Boolean, reflect: !0 })
], e.prototype, "disabled");
i([
  b()
], e.prototype, "_open");
i([
  m("#tooltip")
], e.prototype, "tooltipElement");
i([
  m("#arrow")
], e.prototype, "arrowElement");
customElements.define("minimal-tooltip", e);
export {
  e as MinimalTooltip
};
