// agnosticui-core@2.0.0-alpha.12/dist/components/ProgressRing/vue/VueProgressRing.js
import { defineComponent as a, createElementBlock as n, openBlock as t, renderSlot as o, createTextVNode as r, toDisplayString as i } from "vue";
import "../core/ProgressRing.js";
import { _ as s } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const l = a({
  name: "VueProgressRing",
  props: {
    /**
     * Progress value (0–100)
     */
    value: {
      type: Number,
      default: 0
    },
    /**
     * Size preset
     */
    size: {
      type: String,
      default: "medium"
    },
    /**
     * Color variant
     */
    variant: {
      type: String,
      default: "primary"
    },
    /**
     * Accessible label
     */
    label: {
      type: String,
      default: "Progress"
    },
    /**
     * Disable animation (also respects prefers-reduced-motion)
     */
    noAnimation: {
      type: Boolean,
      default: !1
    }
  }
}), p = [".value", ".size", ".variant", ".label", ".noAnimation"];
function u(e, m, d, f, g, v) {
  return t(), n("ag-progress-ring", {
    ".value": e.value,
    ".size": e.size,
    ".variant": e.variant,
    ".label": e.label,
    ".noAnimation": e.noAnimation
  }, [
    o(e.$slots, "default", {}, () => [
      r(i(e.value) + "%", 1)
    ])
  ], 40, p);
}
const S = /* @__PURE__ */ s(l, [["render", u]]);
export {
  S as default
};
