// agnosticui-core@2.0.0-alpha.12/dist/components/Card/core/_Card.js
import { LitElement as g, css as p, html as v } from "lit";
import { n as d } from "../../../property-CemaeiRl.js";
import { hasSlotContent as n } from "../../../utils/slot.js";
var m = Object.defineProperty, s = (t, a, r, i) => {
  for (var e = void 0, c = t.length - 1, h; c >= 0; c--)
    (h = t[c]) && (e = h(a, r, e) || e);
  return e && m(a, r, e), e;
};
const u = {
  fromAttribute(t) {
    return t === "" || t === "true" ? "md" : t === "false" || t === null ? "" : t || "";
  },
  toAttribute(t) {
    return typeof t == "boolean" ? t ? "md" : null : t || null;
  }
}, l = class l extends g {
  constructor() {
    super(), this._hasHeaderSlotContent = !1, this._hasFooterSlotContent = !1, this.stacked = !1, this.shadow = !1, this.animated = !1, this.rounded = "", this.variant = "";
  }
  /**
   * Handle slot changes to detect if header/footer are empty
   */
  _handleSlotChange(a) {
    const r = a.target, i = r.name;
    i === "header" ? this._hasHeaderSlotContent = n(r) : i === "footer" && (this._hasFooterSlotContent = n(r)), this.requestUpdate();
  }
  firstUpdated() {
    const a = this.shadowRoot?.querySelector('slot[name="header"]'), r = this.shadowRoot?.querySelector('slot[name="footer"]');
    this._hasHeaderSlotContent = n(a), this._hasFooterSlotContent = n(r), this.requestUpdate();
  }
  render() {
    const a = this._hasHeaderSlotContent ? "card-header" : "card-header empty", r = this._hasFooterSlotContent ? "card-footer" : "card-footer empty";
    return v`
      <div class="card-wrapper" part="ag-card-wrapper">
        <div class="${a}" part="ag-card-header">
          <slot name="header" @slotchange=${this._handleSlotChange}></slot>
        </div>
        <div class="card-content" part="ag-card-content">
          <slot></slot>
        </div>
        <div class="${r}" part="ag-card-footer">
          <slot name="footer" @slotchange=${this._handleSlotChange}></slot>
        </div>
      </div>
    `;
  }
};
l.styles = p`
    :host {
      display: block;
      position: relative;
      box-sizing: border-box;
      width: 100%;
      /* Use the global token directly for padding */
      --card-padding: var(--ag-space-6, --ag-card-padding);
      background-color: var(--ag-background-primary);
      border: var(--ag-border-width-1) solid var(--ag-border);
    }

    /* Rounded variants - no rounding by default */
    :host([rounded="sm"]) {
      border-radius: var(--ag-radius-sm);
    }

    :host([rounded="md"]) {
      border-radius: var(--ag-radius-md);
    }

    :host([rounded="lg"]) {
      border-radius: var(--ag-radius-lg);
    }

    :host([shadow]) {
      box-shadow: var(--ag-shadow-lg);
      overflow: hidden;
    }

    /* Enhanced hover effect for shadow cards */
    :host([shadow]:hover) {
      box-shadow: var(--ag-shadow-xl);
    }

    /* Animated cards - smooth transitions */
    :host([animated]) {
      transition:
        box-shadow var(--ag-timing-fast, 150ms) ease-out,
        transform var(--ag-timing-fast, 150ms) cubic-bezier(0.39, 0.575, 0.565, 1);
    }

    :host([animated]:hover) {
      transform: translateY(-3px);
      transition:
        box-shadow var(--ag-timing-fast, 150ms) ease-out,
        transform var(--ag-timing-slow, 300ms) cubic-bezier(0.39, 0.575, 0.565, 1);
    }

    /* Respect reduced motion preferences */
    @media (prefers-reduced-motion), (update: slow) {
      :host([animated]),
      :host([animated]:hover) {
        transition-duration: 0.001ms !important;
        transform: none !important;
      }
    }

    /* Variant colors */
    :host([variant="success"]) {
      background-color: var(--ag-success-background);
      color: var(--ag-success-text);
    }

    :host([variant="info"]) {
      background-color: var(--ag-info-background);
      color: var(--ag-info-text);
    }

    :host([variant="error"]) {
      background-color: var(--ag-danger-background);
      color: var(--ag-danger-text);
    }

    :host([variant="warning"]) {
      background-color: var(--ag-warning-background);
      color: var(--ag-warning-text);
    }

    :host([variant="monochrome"]) {
      background-color: var(--ag-background-primary-inverted);
      color: var(--ag-text-primary-inverted);
    }

    .card-header,
    .card-footer {
      color: var(--ag-text-primary);
    }

    .card-header {
      padding: var(--ag-space-4) var(--card-padding);
      border-bottom: var(--ag-border-width-1) solid var(--ag-border);
    }

    .card-footer {
      padding: var(--ag-space-4) var(--card-padding);
      border-top: var(--ag-border-width-1) solid var(--ag-border);
    }

    /* Hide header/footer when empty class is applied */
    .card-header.empty,
    .card-footer.empty {
      display: none;
    }

    .card-content {
      padding: var(--card-padding);
    }

    :host([stacked]) .card-content > ::slotted(*:not(:last-child)) {
      margin-block-end: var(--ag-space-8);
    }

    /* The accessible clickable card trick */
    ::slotted(a.card-primary-action)::after {
      content: '';
      position: absolute;
      inset: 0;
      z-index: 1;
      cursor: pointer;
    }

    /* Ensure content and other actions are selectable/clickable */
    .card-content,
    ::slotted(h1), ::slotted(h2), ::slotted(h3), ::slotted(h4), ::slotted(h5), ::slotted(h6),
    ::slotted(p), ::slotted(div) {
      position: relative;
      z-index: 2;
    }

    ::slotted(.card-secondary-action) {
      position: relative;
      z-index: 2;
    }
  `;
let o = l;
s([
  d({ type: Boolean, reflect: !0 })
], o.prototype, "stacked");
s([
  d({ type: Boolean, reflect: !0 })
], o.prototype, "shadow");
s([
  d({ type: Boolean, reflect: !0 })
], o.prototype, "animated");
s([
  d({ converter: u, reflect: !0 })
], o.prototype, "rounded");
s([
  d({ type: String, reflect: !0 })
], o.prototype, "variant");
export {
  o as AgCard
};
