// agnosticui-core@2.0.0-alpha.12/dist/components/Drawer/core/_drawer.js
import { LitElement as l, html as d } from "lit";
import { n as t } from "../../../property-CemaeiRl.js";
import "../../Dialog/core/Dialog.js";
var h = Object.defineProperty, o = (r, n, e, c) => {
  for (var i = void 0, p = r.length - 1, a; p >= 0; p--)
    (a = r[p]) && (i = a(n, e, i) || i);
  return i && h(n, e, i), i;
};
class s extends l {
  constructor() {
    super(), this.open = !1, this.heading = "", this.description = "", this.position = "bottom", this.noCloseOnEscape = !1, this.noCloseOnBackdrop = !1, this.showCloseButton = !1;
  }
  firstUpdated(n) {
    super.firstUpdated(n), this._dialogElement = this.shadowRoot?.querySelector("ag-dialog"), this._dialogElement && (this._dialogElement.addEventListener("dialog-open", () => {
      const e = new CustomEvent("drawer-open", {
        bubbles: !0,
        composed: !0
      });
      this.dispatchEvent(e), this.onDrawerOpen?.(e);
    }), this._dialogElement.addEventListener("dialog-close", () => {
      const e = new CustomEvent("drawer-close", {
        bubbles: !0,
        composed: !0
      });
      this.dispatchEvent(e), this.onDrawerClose?.(e), this.open && (this.open = !1);
    }), this._dialogElement.addEventListener("dialog-cancel", () => {
      const e = new CustomEvent("drawer-cancel", {
        bubbles: !0,
        composed: !0
      });
      this.dispatchEvent(e), this.onDrawerCancel?.(e), this.open && (this.open = !1);
    }));
  }
  updated(n) {
    super.updated(n), n.has("open") && this._dialogElement && this._dialogElement.open !== this.open && (this._dialogElement.open = this.open);
  }
  /*
  Property Bindings
  In Lit, the dot (.) prefix is used to create a property binding, which sets a JavaScript property on an element instead of an HTML attribute. This is a critical distinction because properties can hold any JavaScript data type, including objects, arrays, and booleans, while HTML attributes can only store strings.
  https://lit.dev/docs/v1/lit-html/template-reference/#binding-types
  https://vaadin.com/docs/latest/hilla/lit/components/create#binding-to-a-property
  */
  render() {
    return d`
      <ag-dialog
        .open=${this.open}
        .heading=${this.heading}
        .description=${this.description}
        .noCloseOnEscape=${this.noCloseOnEscape}
        .noCloseOnBackdrop=${this.noCloseOnBackdrop}
        .showCloseButton=${this.showCloseButton}
        .drawerPosition=${this.position}
      >
        <slot></slot>
      </ag-dialog>
    `;
  }
}
o([
  t({ type: Boolean, reflect: !0 })
], s.prototype, "open");
o([
  t({ type: String })
], s.prototype, "heading");
o([
  t({ type: String })
], s.prototype, "description");
o([
  t({ type: Boolean })
], s.prototype, "noCloseOnEscape");
o([
  t({ type: Boolean })
], s.prototype, "noCloseOnBackdrop");
o([
  t({ type: Boolean })
], s.prototype, "showCloseButton");
o([
  t({ type: String, reflect: !0, attribute: "position" })
], s.prototype, "position");
o([
  t({ attribute: !1 })
], s.prototype, "onDrawerOpen");
o([
  t({ attribute: !1 })
], s.prototype, "onDrawerClose");
o([
  t({ attribute: !1 })
], s.prototype, "onDrawerCancel");
export {
  s as AgnosticDrawer
};
