// agnosticui-core@2.0.0-alpha.12/dist/components/SelectionButtonGroup/core/_SelectionButtonGroup.js
import { LitElement as p, css as g, nothing as f, html as c } from "lit";
import { n as r } from "../../../property-CemaeiRl.js";
import { r as y } from "../../../state-CovhUvdr.js";
var m = Object.defineProperty, o = (h, e, t, s) => {
  for (var i = void 0, l = h.length - 1, n; l >= 0; l--)
    (n = h[l]) && (i = n(e, t, i) || i);
  return i && m(e, t, i), i;
};
const d = class d extends p {
  constructor() {
    super(), this._handleButtonChange = (e) => {
      e.stopPropagation();
      const { value: t, checked: s } = e.detail;
      let i;
      if (this.type === "radio")
        i = s ? [t] : [];
      else {
        const n = [...this._getSelectedValues()];
        if (s)
          n.includes(t) || n.push(t);
        else {
          const u = n.indexOf(t);
          u > -1 && n.splice(u, 1);
        }
        i = n;
      }
      this._internalSelectedValues = i;
      const l = new CustomEvent("selection-change", {
        detail: {
          value: t,
          checked: s,
          selectedValues: i
        },
        bubbles: !0,
        composed: !0
      });
      this.dispatchEvent(l), this.onSelectionChange && this.onSelectionChange(l);
    }, this._handleKeyDown = (e) => {
      const t = this._getButtons().filter((n) => !n.disabled);
      if (t.length === 0) return;
      const s = t.find((n) => {
        try {
          return n.shadowRoot?.activeElement || n === document.activeElement;
        } catch {
          return n === document.activeElement;
        }
      }), i = s ? t.indexOf(s) : -1;
      let l = null;
      switch (e.key) {
        case "ArrowDown":
        case "ArrowRight":
          e.preventDefault(), l = i === -1 ? 0 : (i + 1) % t.length;
          break;
        case "ArrowUp":
        case "ArrowLeft":
          e.preventDefault(), l = i === -1 ? t.length - 1 : (i - 1 + t.length) % t.length;
          break;
        case "Home":
          e.preventDefault(), l = 0;
          break;
        case "End":
          e.preventDefault(), l = t.length - 1;
          break;
      }
      if (l !== null) {
        const n = t[l];
        n.focus(), this.type === "radio" && this._handleButtonChange(new CustomEvent("selection-button-change", {
          detail: { value: n.value, checked: !0 }
        }));
      }
    }, this.type = "radio", this.name = "", this.legend = "", this.legendHidden = !1, this.theme = "", this.size = "md", this.shape = "", this.value = "", this.values = [], this.disabled = !1, this._internalSelectedValues = [];
  }
  // Get current selected values (controlled or uncontrolled)
  _getSelectedValues() {
    return this.type === "radio" ? this.value ? [this.value] : this._internalSelectedValues : this.values && this.values.length > 0 ? this.values : this._internalSelectedValues;
  }
  connectedCallback() {
    super.connectedCallback(), this.addEventListener("selection-button-change", this._handleButtonChange), this.addEventListener("keydown", this._handleKeyDown);
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this.removeEventListener("selection-button-change", this._handleButtonChange), this.removeEventListener("keydown", this._handleKeyDown);
  }
  updated(e) {
    super.updated(e), (e.has("type") || e.has("name") || e.has("theme") || e.has("size") || e.has("shape") || e.has("disabled") || e.has("value") || e.has("values") || e.has("_internalSelectedValues")) && this._syncChildButtons();
  }
  firstUpdated() {
    this._syncChildButtons();
  }
  _getButtons() {
    const e = this.shadowRoot?.querySelector("slot");
    return e ? e.assignedElements({ flatten: !0 }).filter((t) => t.tagName.toLowerCase() === "ag-selection-button") : [];
  }
  _syncChildButtons() {
    const e = this._getButtons(), t = this._getSelectedValues();
    e.forEach((s) => {
      s._type = this.type, s._name = this.name, s._theme = this.theme, s._size = this.size, s._shape = this.shape, s.checked = t.includes(s.value), this.disabled && (s.disabled = !0);
    });
  }
  _handleSlotChange() {
    this._syncChildButtons();
  }
  render() {
    const e = [
      "selection-button-group__legend",
      this.legendHidden ? "selection-button-group__legend--hidden" : ""
    ].filter(Boolean).join(" ");
    return c`
      <fieldset
        class="selection-button-group"
        part="ag-selection-button-group-fieldset"
        role="${this.type === "radio" ? "radiogroup" : "group"}"
        aria-disabled="${this.disabled ? "true" : "false"}"
      >
        ${this.legend ? c`<legend class="${e}" part="ag-selection-button-group-legend">${this.legend}</legend>` : f}
        <div class="selection-button-group__content" part="ag-selection-button-group-content">
          <slot @slotchange="${this._handleSlotChange}"></slot>
        </div>
      </fieldset>
    `;
  }
};
d.styles = g`
    :host {
      display: block;
    }

    .selection-button-group {
      border: 0;
      padding: 0;
      margin: 0;
      min-inline-size: auto;
    }

    .selection-button-group__legend {
      font-weight: 600;
      font-size: var(--ag-font-size-base);
      color: var(--ag-text-primary);
      padding: 0;
      margin-block-end: var(--ag-space-4);
    }

    .selection-button-group__legend--hidden {
      position: absolute !important;
      width: 1px !important;
      height: 1px !important;
      padding: 0 !important;
      margin: -1px !important;
      overflow: hidden !important;
      clip: rect(0, 0, 0, 0) !important;
      white-space: nowrap !important;
      border: 0 !important;
    }

    .selection-button-group__content {
      display: flex;
      flex-wrap: wrap;
      gap: var(--ag-selection-button-group-gap, var(--ag-space-2));
    }
  `;
let a = d;
o([
  r({ type: String, reflect: !0 })
], a.prototype, "type");
o([
  r({ type: String, reflect: !0 })
], a.prototype, "name");
o([
  r({ type: String })
], a.prototype, "legend");
o([
  r({ type: Boolean, attribute: "legend-hidden" })
], a.prototype, "legendHidden");
o([
  r({ type: String, reflect: !0 })
], a.prototype, "theme");
o([
  r({ type: String, reflect: !0 })
], a.prototype, "size");
o([
  r({ type: String, reflect: !0 })
], a.prototype, "shape");
o([
  r({ type: String })
], a.prototype, "value");
o([
  r({ type: Array })
], a.prototype, "values");
o([
  r({ type: Boolean, reflect: !0 })
], a.prototype, "disabled");
o([
  r({ attribute: !1 })
], a.prototype, "onSelectionChange");
o([
  y()
], a.prototype, "_internalSelectedValues");
export {
  a as AgSelectionButtonGroup
};
