// agnosticui-core@2.0.0-alpha.12/dist/components/Link/core/_Link.js
import { LitElement as l, css as v, html as g } from "lit";
import { n as r } from "../../../property-CemaeiRl.js";
import { o as i } from "../../../if-defined-C8i28hSj.js";
var h = Object.defineProperty, o = (u, a, e, p) => {
  for (var n = void 0, s = u.length - 1, c; s >= 0; s--)
    (c = u[s]) && (n = c(a, e, n) || n);
  return n && h(a, e, n), n;
};
const d = class d extends l {
  constructor() {
    super(), this.href = "", this.variant = "", this.isButton = !1, this.buttonSize = "md", this.buttonShape = "", this.buttonBordered = !1, this.external = !1, this.disabled = !1, this.ariaLabel = "";
  }
  /**
   * Lit lifecycle: called when properties change
   * Used to validate that button-specific props are only used with isButton
   */
  willUpdate(a) {
    super.willUpdate(a), this.isButton || (a.has("buttonSize") && this.buttonSize, a.has("buttonShape") && this.buttonShape, a.has("buttonBordered") && this.buttonBordered);
  }
  _handleClick(a) {
    if (this.disabled) {
      a.preventDefault();
      return;
    }
    this.onClick && this.onClick(a);
  }
  _handleFocus(a) {
    this.onFocus && this.onFocus(a);
    const e = new FocusEvent("focus", {
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(e);
  }
  _handleBlur(a) {
    this.onBlur && this.onBlur(a);
    const e = new FocusEvent("blur", {
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(e);
  }
  /**
   * Focus the internal anchor element
   */
  focus() {
    const a = this.shadowRoot?.querySelector("a");
    a && a.focus();
  }
  /**
   * Blur the internal anchor element
   */
  blur() {
    const a = this.shadowRoot?.querySelector("a");
    a && a.blur();
  }
  render() {
    return this.disabled ? g`
        <span
          part="ag-link"
          role="link"
          aria-disabled="true"
          aria-label=${i(this.ariaLabel || void 0)}
        >
          <slot></slot>
        </span>
      ` : g`
      <a
        href=${this.href}
        part="ag-link"
        rel=${i(this.external ? "noopener noreferrer" : void 0)}
        target=${i(this.external ? "_blank" : void 0)}
        aria-label=${i(this.ariaLabel || void 0)}
        @click=${this._handleClick}
        @focus=${this._handleFocus}
        @blur=${this._handleBlur}
      >
        <slot></slot>
      </a>
    `;
  }
};
d.styles = v`
    /* MINIMALIST & THEMEABLE - Styling via --ag-* design tokens */
    :host {
      display: inline;
    }

    a {
      /* Base link styles */
      font-size: var(--ag-font-size-sm);
      color: var(--ag-primary-text);
      text-decoration: underline;
      cursor: pointer;
      transition-property: color, background-color, border-color, text-decoration-color;
      transition-duration: var(--ag-motion-medium);
    }

    a:hover {
      color: var(--ag-primary-text);
      opacity: 0.85;
    }

    a:focus-visible {
      outline: var(--ag-focus-width) solid rgba(var(--ag-focus), 0.5);
      outline-offset: var(--ag-focus-offset);
      transition: outline var(--ag-motion-medium) ease;
    }

    /* Variant styles for standard links */
    :host([variant="primary"]) a {
      color: var(--ag-primary-text);
      font-weight: 600;
    }

    :host([variant="primary"]) a:hover {
      color: var(--ag-primary-text);
      opacity: 0.85;
    }

    :host([variant="success"]) a {
      color: var(--ag-success-text);
      font-weight: 600;
    }

    :host([variant="success"]) a:hover {
      color: var(--ag-success-text);
      opacity: 0.85;
    }

    :host([variant="warning"]) a {
      font-weight: 600;
      color: var(--ag-warning-text);
    }

    :host([variant="warning"]) a:hover {
      color: var(--ag-warning-text);
      opacity: 0.85;
    }

    :host([variant="danger"]) a {
      font-weight: 600;
      color: var(--ag-danger-text);
    }

    :host([variant="danger"]) a:hover {
      color: var(--ag-danger-text);
      opacity: 0.85;
    }

    :host([variant="monochrome"]) a {
      color: var(--ag-text-primary);
      font-weight: 600;
    }

    :host([variant="monochrome"]) a:hover {
      color: var(--ag-text-primary);
      opacity: 0.85;
    }

    /* Button-style link base */
    :host([isButton]) a {
      display: inline-flex;
      align-items: center;
      justify-content: center;
      padding: calc(2px + var(--ag-space-2)) var(--ag-space-4);
      background: var(--ag-primary);
      color: var(--ag-white);
      text-decoration: none;
      border: var(--ag-border-width-1) solid transparent;
      font-size: var(--ag-font-size-sm);
      line-height: 1;
      gap: var(--ag-space-1);
    }

    :host([isButton]) a:hover {
      background: var(--ag-primary-dark);
      color: var(--ag-white);
    }

    /* Button variant styles */
    :host([isButton][variant="primary"]) a {
      background: var(--ag-primary);
      color: var(--ag-white);
    }

    :host([isButton][variant="primary"]) a:hover {
      background: var(--ag-primary-dark);
    }

    :host([isButton][variant="success"]) a {
      background: var(--ag-success);
      color: var(--ag-white);
    }

    :host([isButton][variant="success"]) a:hover {
      background: var(--ag-success-dark);
    }

    :host([isButton][variant="warning"]) a {
      background: var(--ag-warning);
      color: var(--ag-neutral-900);
    }

    :host([isButton][variant="warning"]) a:hover {
      background: var(--ag-warning-dark);
    }

    :host([isButton][variant="danger"]) a {
      background: var(--ag-danger);
      color: var(--ag-white);
    }

    :host([isButton][variant="danger"]) a:hover {
      background: var(--ag-danger-dark);
    }

    :host([isButton][variant="monochrome"]) a {
      background: var(--ag-background-primary-inverted);
      color: var(--ag-text-primary-inverted);
      font-weight: 400;
    }

    :host([isButton][variant="monochrome"]) a:hover {
      background: var(--ag-background-secondary-inverted);
    }

    /* Button size variants */
    :host([isButton][buttonSize="x-sm"]) a {
      font-size: calc(var(--ag-font-size-base) - 0.375rem);
      padding: var(--ag-space-1) var(--ag-space-2);
    }

    :host([isButton][buttonSize="sm"]) a {
      font-size: var(--ag-font-size-xs);
      padding: var(--ag-space-2) var(--ag-space-3);
    }

    :host([isButton][buttonSize="md"]) a {
      font-size: var(--ag-font-size-sm);
      padding: calc(2px + var(--ag-space-2)) var(--ag-space-4);
    }

    :host([isButton][buttonSize="lg"]) a {
      font-size: var(--ag-font-size-base);
      padding: var(--ag-space-3) var(--ag-space-5);
    }

    :host([isButton][buttonSize="xl"]) a {
      font-size: var(--ag-font-size-md);
      padding: var(--ag-space-3) var(--ag-space-6);
    }

    /* Button shape variants */
    :host([isButton][buttonShape="capsule"]) a {
      border-radius: var(--ag-radius-full);
      padding-inline-start: var(--ag-space-5);
      padding-inline-end: var(--ag-space-5);
    }

    :host([isButton][buttonShape="rounded"]) a {
      border-radius: var(--ag-radius-md);
    }

    :host([isButton][buttonShape="circle"]) a {
      border-radius: 50%;
      width: var(--ag-space-10);
      height: var(--ag-space-10);
      padding: 0;
    }

    :host([isButton][buttonShape="square"]) a {
      border-radius: 0;
      width: var(--ag-space-10);
      height: var(--ag-space-10);
      padding: 0;
    }

    :host([isButton][buttonShape="rounded-square"]) a {
      border-radius: var(--ag-radius-md);
      width: var(--ag-space-10);
      height: var(--ag-space-10);
      padding: 0;
    }

    /* Button bordered variant */
    :host([isButton][buttonBordered]) a {
      background: transparent;
      border: 1px solid var(--ag-neutral-500);
      color: inherit;
    }

    :host([isButton][buttonBordered][variant="primary"]) a {
      color: var(--ag-primary-text);
      border-color: var(--ag-primary-text);
      background: transparent;
    }

    :host([isButton][buttonBordered][variant="primary"]) a:hover {
      background: var(--ag-primary);
      color: var(--ag-white);
    }

    :host([isButton][buttonBordered][variant="success"]) a {
      color: var(--ag-success-text);
      border-color: var(--ag-success-text);
      background: transparent;
    }

    :host([isButton][buttonBordered][variant="success"]) a:hover {
      background: var(--ag-success);
      color: var(--ag-white);
    }

    :host([isButton][buttonBordered][variant="warning"]) a {
      color: var(--ag-warning-text);
      border-color: var(--ag-warning-text);
      background: transparent;
    }

    :host([isButton][buttonBordered][variant="warning"]) a:hover {
      background: var(--ag-warning);
      color: var(--ag-neutral-900);
    }

    :host([isButton][buttonBordered][variant="danger"]) a {
      color: var(--ag-danger-text);
      border-color: var(--ag-danger-text);
      background: transparent;
    }

    :host([isButton][buttonBordered][variant="danger"]) a:hover {
      background: var(--ag-danger);
      color: var(--ag-white);
    }

    :host([isButton][buttonBordered][variant="monochrome"]) a {
      color: var(--ag-text-primary);
      border-color: var(--ag-text-primary);
      background: transparent;
    }

    :host([isButton][buttonBordered][variant="monochrome"]) a:hover {
      background: var(--ag-background-primary-inverted);
      color: var(--ag-text-primary-inverted);
    }

    /* Disabled state */
    :host([disabled]) {
      cursor: not-allowed;
    }

    :host([disabled]) a,
    :host([disabled]) span {
      color: var(--ag-text-muted);
      cursor: not-allowed;
      opacity: 0.6;
      text-decoration: none;
      pointer-events: none;
    }

    :host([disabled][isButton]) a,
    :host([disabled][isButton]) span {
      background: var(--ag-background-disabled);
      color: var(--ag-text-muted);
    }
  `;
let t = d;
o([
  r({ type: String })
], t.prototype, "href");
o([
  r({ type: String, reflect: !0 })
], t.prototype, "variant");
o([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "isButton");
o([
  r({ type: String, reflect: !0 })
], t.prototype, "buttonSize");
o([
  r({ type: String, reflect: !0 })
], t.prototype, "buttonShape");
o([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "buttonBordered");
o([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "external");
o([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "disabled");
o([
  r({ type: String, reflect: !0, attribute: "aria-label" })
], t.prototype, "ariaLabel");
o([
  r({ attribute: !1 })
], t.prototype, "onClick");
o([
  r({ attribute: !1 })
], t.prototype, "onFocus");
o([
  r({ attribute: !1 })
], t.prototype, "onBlur");
export {
  t as AgLink
};
