// agnosticui-core@2.0.0-alpha.12/dist/components/BadgeFx/core/_BadgeFx.js
import { css as p } from "lit";
import { n as i } from "../../../property-CemaeiRl.js";
import { AgBadge as l } from "../../Badge/core/_Badge.js";
import { motionStyles as h } from "../../../styles/motion.styles.js";
var x = Object.defineProperty, o = (g, e, a, r) => {
  for (var t = void 0, n = g.length - 1, d; n >= 0; n--)
    (d = g[n]) && (t = d(e, a, t) || t);
  return t && x(e, a, t), t;
};
const f = class f extends l {
  constructor() {
    super(), this._observer = null, this.fxSpeed = "md", this.fxEase = "ease", this.fxDisabled = !1;
  }
  connectedCallback() {
    super.connectedCallback(), this._updateTheme(), this._observer = new MutationObserver((e) => {
      e.forEach((a) => {
        a.type === "attributes" && a.attributeName === "data-theme" && this._updateTheme();
      });
    }), this._observer.observe(document.documentElement, {
      attributes: !0,
      attributeFilter: ["data-theme"]
    });
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this._observer && (this._observer.disconnect(), this._observer = null);
  }
  _updateTheme() {
    const e = document.documentElement.getAttribute("data-theme") === "dark", a = this.shadowRoot?.querySelector(".badge");
    a instanceof HTMLElement && (a.style.setProperty("--ag-fx-shadow-opacity", e ? "0.8" : "0.4"), e && this.variant === "monochrome" ? a.style.setProperty("--ag-fx-sweep-color", "rgba(0, 0, 0, 0.6)") : a.style.setProperty("--ag-fx-sweep-color", "rgba(255, 255, 255, 0.5)"));
  }
  firstUpdated(e) {
    super.firstUpdated(e), this._applyFxClasses(), this._applyFxCustomProperties(), this._updateTheme();
  }
  updated(e) {
    super.updated(e), (e.has("fx") || e.has("fxDisabled")) && this._applyFxClasses(), e.has("variant") && this._updateTheme(), (e.has("fxSpeed") || e.has("fxEase")) && this._applyFxCustomProperties();
  }
  _applyFxClasses() {
    const e = this.shadowRoot?.querySelector(".badge");
    if (e) {
      const a = [];
      e.classList.forEach((r) => {
        r.startsWith("ag-fx-") && a.push(r);
      }), a.forEach((r) => e.classList.remove(r)), this.fx && !this.fxDisabled && e.classList.add(`ag-fx-${this.fx}`), this.fxDisabled ? e.classList.add("ag-fx-disabled") : e.classList.remove("ag-fx-disabled");
    }
  }
  _applyFxCustomProperties() {
    const e = this.shadowRoot?.querySelector(".badge");
    e instanceof HTMLElement && (e.style.setProperty("--ag-fx-duration", `var(--ag-fx-duration-${this.fxSpeed})`), e.style.setProperty("--ag-fx-ease", `var(--ag-fx-ease-${this.fxEase})`));
  }
};
f.styles = [
  h,
  l.styles,
  p`
      /* ========================================
         OVERRIDE PARENT HOVER STATES
         Neutralize competing hover effects from core
         ======================================== */

      /* Badge has a brightness filter on hover when interactive */
      :host([interactive]) .badge:hover {
        filter: none;
      }

      /* ========================================
         FX EFFECT SETUP
         Add necessary CSS properties for effects that need them
         ======================================== */

      /* Shimmer needs a gradient mask */
      .badge.ag-fx-shimmer {
        overflow: hidden;
        -webkit-mask-image: linear-gradient(
          to right,
        rgba(0, 0, 0, 0) 0%,      /* transparent left edge */
        rgba(0, 0, 0, 0.9) 40%,     /* fade in complete at 40% */
        rgba(0, 0, 0, 0.9) 60%,     /* fade out starts at 60% */
        rgba(0, 0, 0, 0) 100%     /* transparent right edge */
        );
        mask-image: linear-gradient(
          to right,
        rgba(0, 0, 0, 0) 0%,      /* transparent left edge */
        rgba(0, 0, 0, 1) 40%,     /* fade in complete at 40% */
        rgba(0, 0, 0, 1) 60%,     /* fade out starts at 60% */
        rgba(0, 0, 0, 0) 100%     /* transparent right edge */
        );
        /* 40-60% above creates a 20% wide "visible window" with soft 40% fade zones on each side. */
        -webkit-mask-size: 250% 100%;
        mask-size: 250% 100%;
        -webkit-mask-position: 215% 0;
        mask-position: 215% 0;
      }

      /* Ripple needs overflow visible to show expanding ring */
      .badge.ag-fx-ripple {
        overflow: visible;
      }

      /* Highlight Sweep needs a gradient background overlay */
      .badge.ag-fx-highlight-sweep {
        position: relative;
        overflow: hidden;
      }

      .badge.ag-fx-highlight-sweep::before {
        content: '';
        position: absolute;
        inset: 0;
        background: linear-gradient(
          90deg,
          transparent 0%,
          transparent 30%,
          var(--ag-fx-sweep-color, rgba(255, 255, 255, 0.5)) 50%,
          transparent 70%,
          transparent 100%
        );
        transform: translateX(-100%);
        pointer-events: none;
      }

      /* ========================================
         FX EFFECT CLASSES
         Applied to target element (.badge)
         ======================================== */

      /* 
        Prevent hover flicker: 
        - Use @host for hover detection (doesn't move with animation)
        - Or add animation-play-state to prevent retriggering
      */
      :host([fx="bounce"]) .badge,
      :host([fx="flip"]) .badge,
      :host([fx="jelly"]) .badge {
        /* Animation only plays once per hover */
        animation-iteration-count: 1;
      }

      /* Hover-triggered effects - now triggered by :host instead of .badge */
      :host([fx="bounce"]:hover) .badge {
        animation: ag-fx-bounce var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      :host([fx="pulse"]:hover) .badge {
        animation: ag-fx-pulse var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      :host([fx="jelly"]:hover) .badge {
        animation: ag-fx-jelly var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      :host([fx="shimmer"]:hover) .badge {
        animation: ag-fx-shimmer var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      :host([fx="glow"]:hover) .badge {
        animation: ag-fx-glow var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      :host([fx="flip"]:hover) .badge {
        animation: ag-fx-flip var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      :host([fx="ripple"]:hover) .badge {
        animation: ag-fx-ripple var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      :host([fx="highlight-sweep"]:hover) .badge::before {
        animation: ag-fx-highlight-sweep var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      /* Active/press effects - require interactive or explicit cursor */
      .badge.ag-fx-press-pop {
        cursor: pointer;
      }

      .badge.ag-fx-press-pop:active {
        animation: ag-fx-press-pop var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      /* Entrance effects */
      .badge.ag-fx-slide-in {
        animation: ag-fx-slide-in var(--ag-fx-duration, 200ms) var(--ag-fx-ease, ease);
      }

      /* Disabled state */
      .badge.ag-fx-disabled {
        animation: none !important;
      }

      .badge.ag-fx-disabled::before {
        animation: none !important;
      }

      /* Reduced motion support */
      @media (prefers-reduced-motion: reduce) {
        .badge.ag-fx-bounce,
        .badge.ag-fx-pulse,
        .badge.ag-fx-jelly,
        .badge.ag-fx-shimmer,
        .badge.ag-fx-glow,
        .badge.ag-fx-flip,
        .badge.ag-fx-ripple,
        .badge.ag-fx-highlight-sweep,
        .badge.ag-fx-press-pop,
        .badge.ag-fx-slide-in {
          animation-duration: 0.01ms !important;
          transition-duration: 0.01ms !important;
        }
      }
    `
];
let s = f;
o([
  i({ type: String, reflect: !0 })
], s.prototype, "fx");
o([
  i({ type: String, attribute: "fx-speed", reflect: !0 })
], s.prototype, "fxSpeed");
o([
  i({ type: String, attribute: "fx-ease", reflect: !0 })
], s.prototype, "fxEase");
o([
  i({ type: Boolean, attribute: "fx-disabled", reflect: !0 })
], s.prototype, "fxDisabled");
export {
  s as BadgeFx
};
