// agnosticui-core@2.0.0-alpha.12/dist/components/Timeline/core/_Timeline.js
import { LitElement as h, css as f, nothing as c, html as p } from "lit";
import { n as e } from "../../../property-CemaeiRl.js";
var v = Object.defineProperty, a = (u, t, s, i) => {
  for (var r = void 0, l = u.length - 1, m; l >= 0; l--)
    (m = u[l]) && (r = m(t, s, r) || r);
  return r && v(t, s, r), r;
};
const g = class g extends h {
  // ──────────────────────────────────────────────────────────────
  // Constructor
  // ──────────────────────────────────────────────────────────────
  constructor() {
    super(), this.orientation = "horizontal", this.variant = "", this.compact = !1;
  }
  // ──────────────────────────────────────────────────────────────
  // Lifecycle
  // ──────────────────────────────────────────────────────────────
  updated(t) {
    super.updated(t), (t.has("orientation") || t.has("variant")) && this._updateChildItems();
  }
  _updateChildItems() {
    const t = this.shadowRoot?.querySelector("slot");
    if (!t) return;
    const s = t.assignedElements({ flatten: !0 }).filter((i) => i.tagName.toLowerCase() === "ag-timeline-item");
    s.forEach((i, r) => {
      i.setAttribute("orientation", this.orientation), this.variant && i.setAttribute("variant", this.variant), i.toggleAttribute("first", r === 0), i.toggleAttribute("last", r === s.length - 1);
    });
  }
  // ──────────────────────────────────────────────────────────────
  // Main Render
  // ──────────────────────────────────────────────────────────────
  render() {
    return p`
      <ul 
        class="timeline-container"
        part="ag-timeline-container"
        role="list"
        aria-label=${this.ariaLabel ?? c}
      >
        <slot @slotchange=${this._updateChildItems}></slot>
      </ul>
    `;
  }
};
g.styles = f`
    :host {
      display: block;
    }

    .timeline-container {
      list-style: none;
      margin: 0;
      padding: 0;
      display: flex;
      position: relative;
    }

    slot {
      display: contents;
    }

    /* Horizontal orientation (default) */
    :host([orientation="horizontal"]) .timeline-container {
      flex-direction: row;
      align-items: stretch;
    }

    /* Vertical orientation */
    :host([orientation="vertical"]) .timeline-container {
      flex-direction: column;
      align-items: stretch;
    }
  `;
let n = g;
a([
  e({ type: String, reflect: !0 })
], n.prototype, "orientation");
a([
  e({ type: String, reflect: !0 })
], n.prototype, "variant");
a([
  e({ type: Boolean })
], n.prototype, "compact");
a([
  e({ type: String, attribute: "aria-label" })
], n.prototype, "ariaLabel");
const d = class d extends h {
  // ──────────────────────────────────────────────────────────────
  // Constructor
  // ──────────────────────────────────────────────────────────────
  constructor() {
    super(), this.orientation = "horizontal", this.variant = "", this.first = !1, this.last = !1;
  }
  // ──────────────────────────────────────────────────────────────
  // Main Render
  // ──────────────────────────────────────────────────────────────
  render() {
    return p`
      <li class="item-container" part="ag-item-container" role="listitem">
        <div class="connector connector-before" part="ag-connector" aria-hidden="true"></div>
        
        <div class="ag-start" part="ag-start">
          <slot name="ag-start"></slot>
        </div>
        
        <div 
          class="ag-marker" 
          part="ag-marker"
          role=${this.markerAriaLabel ? "img" : c}
          aria-label=${this.markerAriaLabel ?? c}
        >
          <slot name="ag-marker"></slot>
        </div>
        
        <div class="ag-end" part="ag-end">
          <slot name="ag-end"></slot>
        </div>
        
        <div class="connector connector-after" part="ag-connector" aria-hidden="true"></div>
      </li>
    `;
  }
};
d.styles = f`
    :host {
      display: block;
      position: relative;
    }

    :host([orientation="horizontal"]) {
      flex: 1;
    }

    .item-container {
      display: grid;
      position: relative;
      align-items: center;
    }

    /* Horizontal layout - 3x3 grid */
    :host([orientation="horizontal"]) .item-container {
      grid-template-columns: minmax(0, 1fr) auto minmax(0, 1fr);
      grid-template-rows: auto auto auto;
      justify-items: center;
      align-items: center;
    }

    /* Vertical layout - 3x3 grid */
    /* Vertical layout - Flush alignment */
    :host([orientation="vertical"]) .item-container {
      grid-template-columns: minmax(0, 1fr) auto minmax(0, 1fr);
      /* Rows: connector-before (0), marker (auto), connector-after + spacing (minmax ensures minimum gap between items) */
      grid-template-rows: 0 auto minmax(var(--ag-timeline-vertical-spacing, var(--ag-space-4)), 1fr);
      align-items: start;
      justify-items: center;
    }

    /* Start slot positioning */
    .ag-start {
      margin: var(--ag-space-1);
      /* Reset line-height for slotted content */
      line-height: normal;
    }

    :host([orientation="horizontal"]) .ag-start {
      grid-column: 1 / 4;
      grid-row: 1 / 2;
      align-self: end;
      justify-self: center;
    }

    :host([orientation="vertical"]) .ag-start {
      grid-column: 1 / 2;
      grid-row: 2 / 4;
      align-self: var(--ag-timeline-start-align, start);
      justify-self: end;
      margin: 0 var(--ag-space-2) 0 0;
    }

    /* Marker slot positioning */
    .ag-marker {
      grid-column: 2 / 3;
      grid-row: 2 / 3;
      display: flex;
      align-items: center;
      justify-content: center;
      z-index: 1;
      /* Reset line-height for slotted content */
      line-height: normal;
    }

    /* End slot positioning */
    .ag-end {
      margin: var(--ag-space-1);
      /* Reset line-height for slotted content */
      line-height: normal;
    }

    :host([orientation="horizontal"]) .ag-end {
      grid-column: 1 / 4;
      grid-row: 3 / 4;
      align-self: start;
      justify-self: center;
    }

    :host([orientation="vertical"]) .ag-end {
      grid-column: 3 / 4;
      grid-row: 2 / 4;
      align-self: var(--ag-timeline-end-align, start);
      justify-self: start;
      margin: 0 0 0 var(--ag-space-2);
    }

    /* Connector lines */
    .connector {
      border: none;
      background: var(--ag-timeline-connector-color, var(--ag-border));
      margin: 0;
    }

    .connector-before {
      grid-column: 1 / 2;
      grid-row: 2 / 3;
    }

    .connector-after {
      grid-column: 3 / 4;
      grid-row: 2 / 3;
    }

    /* Horizontal connectors */
    :host([orientation="horizontal"]) .connector {
      width: 100%;
      height: var(--ag-timeline-connector-width, 2px);
    }

    :host([orientation="horizontal"][first]) .connector-before {
      border-radius: var(--ag-radius-full) 0 0 var(--ag-radius-full);
    }

    :host([orientation="horizontal"][last]) .connector-after {
      border-radius: 0 var(--ag-radius-full) var(--ag-radius-full) 0;
    }

    /* Vertical connectors */
    :host([orientation="vertical"]) .connector {
      height: 100%;
      width: var(--ag-timeline-connector-width, 2px);
    }

    :host([orientation="vertical"]) .connector-before {
      grid-column: 2 / 3;
      grid-row: 1 / 2;
    }

    :host([orientation="vertical"]) .connector-after {
      grid-column: 2 / 3;
      grid-row: 2 / 4;
      z-index: 0;
    }

    :host([orientation="vertical"][first]) .connector-before {
      border-radius: var(--ag-radius-full) var(--ag-radius-full) 0 0;
    }

    :host([orientation="vertical"][last]) .connector-after {
      border-radius: 0 0 var(--ag-radius-full) var(--ag-radius-full);
    }

    /* Hide first connector on first item */
    :host([first]) .connector-before {
      display: none;
    }

    /* Hide last connector on last item */
    :host([last]) .connector-after {
      display: none;
    }

    /* Variant colors */
    :host([variant="primary"]) .connector {
      background: var(--ag-primary);
    }

    :host([variant="success"]) .connector {
      background: var(--ag-success);
    }

    :host([variant="warning"]) .connector {
      background: var(--ag-warning);
    }

    :host([variant="danger"]) .connector {
      background: var(--ag-danger);
    }

    :host([variant="monochrome"]) .connector {
      background: var(--ag-text-muted);
    }
  `;
let o = d;
a([
  e({ type: String, reflect: !0 })
], o.prototype, "orientation");
a([
  e({ type: String, reflect: !0 })
], o.prototype, "variant");
a([
  e({ type: String, attribute: "marker-aria-label" })
], o.prototype, "markerAriaLabel");
a([
  e({ type: Boolean, reflect: !0 })
], o.prototype, "first");
a([
  e({ type: Boolean, reflect: !0 })
], o.prototype, "last");
export {
  n as AgTimeline,
  o as AgTimelineItem
};
