// agnosticui-core@2.0.0-alpha.12/dist/components/Header/core/_Header.js
import { LitElement as l, css as p, html as h } from "lit";
import { n as d } from "../../../property-CemaeiRl.js";
var f = Object.defineProperty, c = (o, r, i, g) => {
  for (var e = void 0, n = o.length - 1, s; n >= 0; n--)
    (s = o[n]) && (e = s(r, i, e) || e);
  return e && f(r, i, e), e;
};
const a = class a extends l {
  constructor() {
    super(...arguments), this.sticky = !1, this.contentJustify = "between";
  }
  render() {
    return h`
      <header class="header" part="ag-header">
        <div class="header-content" part="ag-header-content">
          <div class="header-logo">
            <slot name="logo"></slot>
          </div>
          <div class="header-nav">
            <slot></slot>
          </div>
        </div>
      </header>
    `;
  }
};
a.styles = p`
    :host {
      display: block;
    }

    /* Sticky positioning - must be on :host to work with light DOM scroll containers */
    :host([sticky]) {
      position: sticky;
      top: 0;
      z-index: var(--ag-z-index-dropdown);
    }

    .header {
      font-family: inherit;
      background-color: var(--ag-background-primary);
      box-shadow: 0 1px 5px 2px rgb(0 0 0 / 10%);
      border-bottom: var(--ag-border-width-1) solid var(--ag-border);
      padding-block-start: var(--ag-space-2);
      padding-block-end: var(--ag-space-2);
      padding-inline-start: var(--ag-space-6);
      padding-inline-end: var(--ag-space-6);
    }

    /* Mobile-first: column layout */
    .header-content {
      max-inline-size: 100%;
      margin: 0 auto;
      display: flex;
      flex-direction: column;
      align-items: center;
      gap: var(--ag-space-4);
    }

    .header-logo {
      display: flex;
      align-items: center;
    }

    .header-nav {
      display: flex;
      align-items: center;
    }

    /* Ensure slotted nav elements inherit flex + center alignment */
    .header-nav ::slotted(nav) {
      display: flex;
      align-items: center;
    }

    /* Desktop: row layout with configurable justification */
    @media (min-width: 960px) {
      .header-content {
        flex-direction: row;
        width: 960px;
      }

      :host([contentJustify="between"]) .header-content {
        justify-content: space-between;
      }

      :host([contentJustify="start"]) .header-content {
        justify-content: flex-start;
      }

      :host([contentJustify="end"]) .header-content {
        justify-content: flex-end;
      }

      :host([contentJustify="around"]) .header-content {
        justify-content: space-around;
      }

      :host([contentJustify="center"]) .header-content {
        justify-content: center;
      }
    }
  `;
let t = a;
c([
  d({ type: Boolean, reflect: !0 })
], t.prototype, "sticky");
c([
  d({ type: String, reflect: !0 })
], t.prototype, "contentJustify");
export {
  t as Header
};
