// agnosticui-core@2.0.0-alpha.12/dist/components/Link/vue/VueLink.js
import { defineComponent as l, ref as u, onMounted as o, watchEffect as r, createElementBlock as i, openBlock as d, renderSlot as f } from "vue";
import "../core/Link.js";
import { _ as s } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const c = l({
  name: "VueLink",
  props: {
    href: {
      type: String,
      default: ""
    },
    variant: {
      type: String,
      default: ""
    },
    isButton: {
      type: Boolean,
      default: !1
    },
    buttonSize: {
      type: String,
      default: "md"
    },
    buttonShape: {
      type: String,
      default: ""
    },
    buttonBordered: {
      type: Boolean,
      default: !1
    },
    external: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    ariaLabel: {
      type: String,
      default: ""
    }
  },
  emits: ["click", "focus", "blur"],
  setup(t, { emit: n }) {
    const e = u(null);
    return o(() => {
      e.value && (e.value.addEventListener("click", (a) => n("click", a)), e.value.addEventListener("focus", (a) => n("focus", a)), e.value.addEventListener("blur", (a) => n("blur", a)));
    }), r(() => {
      e.value && (e.value.href = t.href, e.value.variant = t.variant, e.value.isButton = t.isButton, e.value.buttonSize = t.buttonSize, e.value.buttonShape = t.buttonShape, e.value.buttonBordered = t.buttonBordered, e.value.external = t.external, e.value.disabled = t.disabled, e.value.ariaLabel = t.ariaLabel);
    }), {
      agComponent: e
    };
  }
}), v = { ref: "agComponent" };
function b(t, n, e, a, p, B) {
  return d(), i("ag-link", v, [
    f(t.$slots, "default")
  ], 512);
}
const h = /* @__PURE__ */ s(c, [["render", b]]);
export {
  h as default
};
