// agnosticui-core@2.0.0-alpha.12/dist/components/Menu/core/_Menu.js
import { LitElement as p, css as b, html as u, nothing as x } from "lit";
import { n as a } from "../../../property-CemaeiRl.js";
import { r as m } from "../../../state-CovhUvdr.js";
import { e as I } from "../../../query-BykXNUlT.js";
import "../../Button/core/Button.js";
var w = Object.defineProperty, n = (y, e, t, o) => {
  for (var i = void 0, r = y.length - 1, d; r >= 0; r--)
    (d = y[r]) && (i = d(e, t, i) || i);
  return i && w(e, t, i), i;
};
const g = class g extends p {
  constructor() {
    super(), this.size = "md", this.shape = "rounded", this.bordered = !1, this.ghost = !1, this.link = !1, this.grouped = !1, this.disabled = !1, this.loading = !1, this.ariaLabel = "", this.menuVariant = "chevron", this.buttonVariant = "", this.unicode = "", this._menuOpen = !1, this.menuAlign = "left", this.additionalGutter = "", this._clickOutsideHandler = this._handleClickOutside.bind(this);
  }
  connectedCallback() {
    super.connectedCallback(), document.addEventListener("click", this._clickOutsideHandler);
  }
  disconnectedCallback() {
    super.disconnectedCallback(), document.removeEventListener("click", this._clickOutsideHandler);
  }
  firstUpdated() {
    this._updateMenuReference();
  }
  updated(e) {
    super.updated(e), e.has("_menuOpen") && this._updateMenuReference();
  }
  _updateMenuReference() {
    this._menu = this.querySelector("ag-menu");
  }
  _handleClickOutside(e) {
    !this._menuOpen || e.composedPath().includes(this) || this._closeMenu();
  }
  _handleClick(e) {
    if (this.disabled) return;
    this.onClick && this.onClick(e);
    const t = new MouseEvent("click", {
      bubbles: !0,
      composed: !0,
      cancelable: e.cancelable
    });
    this.dispatchEvent(t), this._menuOpen ? this._closeMenu() : this._openMenu();
  }
  _handleFocus(e) {
    this.onFocus && this.onFocus(e);
    const t = new FocusEvent("focus", {
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(t);
  }
  _handleBlur(e) {
    this.onBlur && this.onBlur(e);
    const t = new FocusEvent("blur", {
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(t);
  }
  _handleKeydown(e) {
    switch (e.key) {
      case "Enter":
      case " ":
        e.preventDefault(), this._openMenu();
        break;
      case "ArrowDown":
        e.preventDefault(), this._openMenu(!0);
        break;
      case "ArrowUp":
        e.preventDefault(), this._openMenu(!1);
        break;
    }
  }
  // Add method to update menu position based on button height:
  _updateMenuPosition() {
    if (!this._menu || !this._trigger) return;
    const e = this._trigger.shadowRoot?.querySelector("button");
    if (!e) return;
    const t = e.offsetHeight;
    this.additionalGutter ? this._menu.style.top = `calc(${t}px + ${this.additionalGutter})` : this._menu.style.top = `${t}px`;
  }
  _openMenu(e = !0) {
    if (this._menuOpen) return;
    this._menuOpen = !0, this._updateMenuReference(), this._menu && (this._menu.open = !0, this._updateMenuPosition(), this._menu._updateMenuItems(), requestAnimationFrame(() => {
      e ? this._menu?._focusFirstItem() : this._menu?._focusLastItem();
    }));
    const t = new CustomEvent("menu-open", {
      detail: { open: !0 },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(t), this.onMenuOpen && this.onMenuOpen(t);
  }
  _closeMenu() {
    if (!this._menuOpen) return;
    this._menuOpen = !1, this._menu && (this._menu.open = !1), this._trigger?.focus();
    const e = new CustomEvent("menu-close", {
      detail: { open: !1 },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(e), this.onMenuClose && this.onMenuClose(e);
  }
  _renderChevronIcon() {
    return u`
      <svg class="chevron-icon" part="ag-menu-chevron-icon" viewBox="0 0 20 20" fill="currentColor">
        <path
          fill-rule="evenodd"
          d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
          clip-rule="evenodd"
        />
      </svg>
    `;
  }
  render() {
    this.setAttribute("data-menu-open", this._menuOpen ? "true" : "false");
    const e = this.unicode ? u`<span class="unicode-icon" part="ag-menu-unicode-icon">${this.unicode}</span>` : u`<slot></slot>`, t = this.menuVariant === "icon";
    return u`
      <ag-button
        .variant=${this.buttonVariant}
        .size=${this.size}
        .shape=${t ? this.shape || "square" : this.shape}
        ?bordered=${this.bordered}
        ?ghost=${this.ghost}
        ?link=${this.link}
        ?grouped=${this.grouped}
        ?disabled=${this.disabled}
        ?loading=${this.loading}
        aria-haspopup="menu"
        aria-expanded="${this._menuOpen}"
        .ariaLabel=${this.ariaLabel || "Menu"}
        @click=${this._handleClick}
        @focus=${this._handleFocus}
        @blur=${this._handleBlur}
        @keydown=${this._handleKeydown}
      >
        ${this.menuVariant === "chevron" ? u`<span class="label" part="ag-menu-label">${e}</span> ${this._renderChevronIcon()}` : e}
      </ag-button>
      <slot name="menu"></slot>
    `;
  }
};
g.styles = b`
    :host {
      display: inline-flex;
      position: relative;
      background-color: inherit;
    }

    .chevron-icon {
      width: var(--ag-space-4);
      height: var(--ag-space-4);
      color: var(--ag-text-secondary);
      transition: transform var(--ag-motion-fast) ease;
      flex-shrink: 0;
    }

    :host([button-variant="primary"]) .chevron-icon,
    :host([button-variant="secondary"]) .chevron-icon,
    :host([button-variant="success"]) .chevron-icon,
    :host([button-variant="warning"]) .chevron-icon,
    :host([button-variant="danger"]) .chevron-icon,
    :host([button-variant="monochrome"]) .chevron-icon {
      color: var(--ag-white);
    }

    /* When a button is ghost, we want the chevron to be the variant color */
    :host([ghost][button-variant="primary"]) .chevron-icon {
      color: var(--ag-primary);
    }
    :host([ghost][button-variant="secondary"]) .chevron-icon {
      color: var(--ag-secondary);
    }
    :host([ghost][button-variant="success"]) .chevron-icon {
      color: var(--ag-success);
    }
    :host([ghost][button-variant="warning"]) .chevron-icon {
      color: var(--ag-warning);
    }
    :host([ghost][button-variant="danger"]) .chevron-icon {
      color: var(--ag-danger);
    }
    :host([ghost][button-variant="monochrome"]) .chevron-icon {
      color: var(--ag-text-primary);
    }

    .label {
      display: inline-flex;
      align-items: center;
      gap: var(--ag-space-2);
    }

    ag-button[aria-expanded="true"] .chevron-icon {
      transform: rotate(180deg);
    }

    ::slotted(ag-menu) {
      position: absolute;
      background-color: var(--ag-background-primary);
      margin-block-start: var(--ag-space-1);
      z-index: var(--ag-z-index-dropdown);
    }
    /* Left alignment - menu left aligns with button left */
    :host([menu-align="left"]) ::slotted(ag-menu) {
      inset-inline-start: 0;
      inset-inline-end: auto;
    }

    /* Right alignment - menu right aligns with button right */
    :host([menu-align="right"]) ::slotted(ag-menu) {
      inset-inline-end: 0;
      inset-inline-start: auto;
    }
  `;
let s = g;
n([
  I("ag-button")
], s.prototype, "_trigger");
n([
  a({ type: String, reflect: !0 })
], s.prototype, "size");
n([
  a({ type: String, reflect: !0 })
], s.prototype, "shape");
n([
  a({ type: Boolean, reflect: !0 })
], s.prototype, "bordered");
n([
  a({ type: Boolean, reflect: !0 })
], s.prototype, "ghost");
n([
  a({ type: Boolean, reflect: !0 })
], s.prototype, "link");
n([
  a({ type: Boolean, reflect: !0 })
], s.prototype, "grouped");
n([
  a({ type: Boolean })
], s.prototype, "disabled");
n([
  a({ type: Boolean })
], s.prototype, "loading");
n([
  a({ type: String, reflect: !0, attribute: "aria-label" })
], s.prototype, "ariaLabel");
n([
  a({ type: String, reflect: !0, attribute: "menu-variant" })
], s.prototype, "menuVariant");
n([
  a({ type: String, reflect: !0, attribute: "button-variant" })
], s.prototype, "buttonVariant");
n([
  a({ type: String })
], s.prototype, "unicode");
n([
  a({ type: String, reflect: !0, attribute: "menu-align" })
], s.prototype, "menuAlign");
n([
  a({ type: String, reflect: !0, attribute: "additional-gutter" })
], s.prototype, "additionalGutter");
n([
  a({ attribute: !1 })
], s.prototype, "onMenuOpen");
n([
  a({ attribute: !1 })
], s.prototype, "onMenuClose");
n([
  a({ attribute: !1 })
], s.prototype, "onClick");
n([
  a({ attribute: !1 })
], s.prototype, "onFocus");
n([
  a({ attribute: !1 })
], s.prototype, "onBlur");
n([
  m()
], s.prototype, "_menuOpen");
const f = class f extends p {
  constructor() {
    super(), this.open = !1, this.placement = "bottom-start", this.ariaLabel = "", this.type = "default", this.checkHiddenItems = !1, this._focusedIndex = 0, this._menuItems = [];
  }
  connectedCallback() {
    super.connectedCallback(), this.setAttribute("role", "menu"), this.setAttribute("aria-orientation", "vertical"), this.addEventListener("keydown", this._handleKeydown), this.addEventListener("menu-select", this._handleMenuSelect);
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this.removeEventListener("keydown", this._handleKeydown), this.removeEventListener("menu-select", this._handleMenuSelect);
  }
  firstUpdated() {
    this.open || this.setAttribute("hidden", "");
  }
  updated(e) {
    e.has("open") && (this.open ? this.removeAttribute("hidden") : (this.setAttribute("hidden", ""), this.type === "default" && (this.selectedValue = void 0, this._updateSelection()))), e.has("selectedValue") && this._updateSelection();
  }
  _handleMenuSelect(e) {
    const t = e.target;
    this.type === "single-select" && (this.selectedValue = t.value);
  }
  _updateSelection() {
    this._menuItems.forEach((e) => {
      e.checked = this.selectedValue !== void 0 && e.value === this.selectedValue;
    });
  }
  /**
   * Check if a menu item is actually navigable based on its computed styles.
   * This handles Tailwind's responsive utilities (sm:hidden, md:hidden, etc.)
   * which are applied to wrapper divs around menu items.
   * 
   * Only called when checkHiddenItems prop is true.
   */
  _isElementNavigable(e) {
    if (e.disabled || e.getAttribute("aria-hidden") === "true")
      return !1;
    if (!this.checkHiddenItems)
      return !0;
    const t = window.getComputedStyle(e);
    if (t.display === "none" || t.visibility === "hidden")
      return !1;
    let o = e.parentElement;
    for (; o && o !== this; ) {
      const i = window.getComputedStyle(o);
      if (i.display === "none" || i.visibility === "hidden")
        return !1;
      o = o.parentElement;
    }
    return !0;
  }
  /**
   * Get currently navigable items (filtered for visibility).
   * When checkHiddenItems is true, recalculates on every call to handle dynamic viewport changes.
   */
  _getNavigableItems() {
    return this.checkHiddenItems ? this._menuItems.filter((e) => this._isElementNavigable(e)) : this._menuItems.filter(
      (e) => !e.disabled && e.getAttribute("aria-hidden") !== "true"
    );
  }
  _updateMenuItems() {
    this._menuItems = Array.from(this.querySelectorAll("ag-menu-item")), this._updateTabIndex(), this._updateSelection();
  }
  _updateTabIndex() {
    this._menuItems.forEach((e, t) => {
      e.setAttribute("tabindex", t === this._focusedIndex ? "0" : "-1");
    });
  }
  _focusFirstItem() {
    const e = this._getNavigableItems();
    if (e.length === 0) return;
    const t = e[0];
    this._focusedIndex = this._menuItems.indexOf(t), this._updateTabIndex(), t?.focus();
  }
  _focusLastItem() {
    const e = this._getNavigableItems();
    if (e.length === 0) return;
    const t = e[e.length - 1];
    this._focusedIndex = this._menuItems.indexOf(t), this._updateTabIndex(), t?.focus();
  }
  _focusNextItem() {
    const e = this._getNavigableItems();
    if (e.length === 0) return;
    const t = this._menuItems[this._focusedIndex], o = e.indexOf(t), i = o >= 0 ? o : -1;
    for (let r = 1; r <= e.length; r++) {
      const d = (i + r) % e.length, h = e[d];
      this._focusedIndex = this._menuItems.indexOf(h), this._updateTabIndex(), h?.focus();
      return;
    }
  }
  _focusPreviousItem() {
    const e = this._getNavigableItems();
    if (e.length === 0) return;
    const t = this._menuItems[this._focusedIndex], o = e.indexOf(t), i = o >= 0 ? o : e.length;
    for (let r = 1; r <= e.length; r++) {
      const d = (i - r + e.length) % e.length, h = e[d];
      this._focusedIndex = this._menuItems.indexOf(h), this._updateTabIndex(), h?.focus();
      return;
    }
  }
  _handleKeydown(e) {
    switch (this.onKeyDown && this.onKeyDown(e), e.key) {
      case "ArrowDown":
        e.preventDefault(), this._focusNextItem();
        break;
      case "ArrowUp":
        e.preventDefault(), this._focusPreviousItem();
        break;
      case "Home":
        e.preventDefault(), this._focusFirstItem();
        break;
      case "End":
        e.preventDefault(), this._focusLastItem();
        break;
      case "Enter": {
        e.preventDefault();
        const t = this._getNavigableItems(), o = this._menuItems[this._focusedIndex];
        t.includes(o) && o?.click();
        break;
      }
      case "Escape":
      case "Tab":
        e.preventDefault(), this._closeMenu();
        break;
    }
  }
  _closeMenu() {
    const e = this.closest("ag-menu-button");
    e && e._closeMenu();
  }
  render() {
    return u`<div class="menu" part="ag-menu"><slot></slot></div>`;
  }
};
f.styles = b`
    :host {
      position: absolute;
      background-color: var(--ag-background-primary);
      border: 1px solid var(--ag-border-subtle);
      border-radius: var(--ag-radius-md);
      box-shadow: 0 4px 6px -1px rgba(0, 0, 0, 0.1), 0 2px 4px -1px rgba(0, 0, 0, 0.06);
      margin-block-start: var(--ag-space-1);
      min-inline-size: 12rem;
      max-inline-size: 16rem;
      width: max-content;
      z-index: var(--ag-z-index-dropdown);
      overflow: hidden;
      text-overflow: ellipsis;
    }

    .menu {
      padding: var(--ag-space-2);
    }

    :host([hidden]) {
      display: none;
    }
  `;
let c = f;
n([
  a({ type: Boolean })
], c.prototype, "open");
n([
  a()
], c.prototype, "placement");
n([
  a({ reflect: !0, attribute: "aria-label" })
], c.prototype, "ariaLabel");
n([
  a({ attribute: "selected-value" })
], c.prototype, "selectedValue");
n([
  a({ type: String })
], c.prototype, "type");
n([
  a({ type: Boolean, attribute: "check-hidden-items" })
], c.prototype, "checkHiddenItems");
n([
  a({ attribute: !1 })
], c.prototype, "onKeyDown");
n([
  m()
], c.prototype, "_focusedIndex");
n([
  m()
], c.prototype, "_menuItems");
const v = class v extends p {
  constructor() {
    super(), this._menu = null, this.value = "", this.disabled = !1, this.href = "", this.target = "", this.checked = !1, this.variant = "default";
  }
  connectedCallback() {
    super.connectedCallback(), this._menu = this.closest("ag-menu");
    const e = this._menu?.type === "single-select" ? "menuitemradio" : "menuitem";
    this.setAttribute("role", e), this.setAttribute("tabindex", "-1"), this.addEventListener("click", this._handleClick);
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this.removeEventListener("click", this._handleClick);
  }
  updated(e) {
    e.has("checked") && this.setAttribute("aria-checked", this.checked ? "true" : "false");
  }
  _handleClick(e) {
    if (this.disabled) {
      e.preventDefault(), e.stopPropagation();
      return;
    }
    this.onClick && e instanceof MouseEvent && this.onClick(e);
    const t = this.closest("ag-menu-button"), o = new CustomEvent("menu-select", {
      detail: { value: this.value },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(o), this.onMenuSelect && this.onMenuSelect(o), t && !this.href && t._closeMenu();
  }
  focus() {
    const e = this.shadowRoot?.querySelector("button, a");
    e && e.focus();
  }
  render() {
    return this.href ? u`
        <a
          href="${this.href}"
          target="${this.target || x}"
          part="ag-menu-item-link ag-menu-item"
          ?disabled="${this.disabled}"
        >
          <slot></slot>
        </a>
      ` : u`
      <button ?disabled="${this.disabled}" part="ag-menu-item-button ag-menu-item">
        <slot></slot>
      </button>
    `;
  }
};
v.styles = b`
    :host {
      display: block;
    }

    button {
      width: 100%;
    }
    button,
    a {
      display: block;
      background-color: transparent;
      color: var(--ag-text-primary);
      border: none;
      border-radius: var(--ag-radius-sm);
      padding: var(--ag-space-2) var(--ag-space-3);
      text-align: start;
      text-decoration: none;
      font-size: inherit;
      line-height: 1.25;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      cursor: pointer;
      transition: all var(--ag-motion-fast) ease;
    }

    button:hover:not([disabled]),
    a:hover:not([disabled]) {
      background-color: var(--ag-background-secondary);
      color: var(--ag-text-primary);
    }

    button:focus,
    a:focus {
      background-color: var(--ag-background-secondary);
      outline: var(--ag-focus-width) solid var(--ag-focus);
      outline-offset: 0;
    }

    button:active:not([disabled]),
    a:active:not([disabled]) {
      background-color: var(--ag-background-tertiary);
    }

    :host([aria-checked='true']) button:not([disabled]),
    :host([aria-checked='true']) a:not([disabled]) {
      background-color: var(--ag-menu-item-selected-bg, var(--ag-primary));
      color: var(--ag-white);
    }

    :host([aria-checked='true']) button:focus:not([disabled]),
    :host([aria-checked='true']) a:focus:not([disabled]),
    :host([aria-checked='true']) button:hover:not([disabled]),
    :host([aria-checked='true']) a:hover:not([disabled]) {
      background-color: var(--ag-menu-item-selected-active-bg, var(--ag-primary-dark));
    }


    :host([variant="monochrome"][aria-checked='true']) button:not([disabled]),
    :host([variant="monochrome"][aria-checked='true']) a:not([disabled]) {
      background-color: var(--ag-background-primary-inverted);
      color: var(--ag-text-primary-inverted);
    }

    /* hover/active + selected gets --ag-background-secondary-inverted affordance */
    :host([variant="monochrome"][aria-checked='true']) button:focus:not([disabled]),
    :host([variant="monochrome"][aria-checked='true']) a:focus:not([disabled]),
    :host([variant="monochrome"][aria-checked='true']) button:hover:not([disabled]),
    :host([variant="monochrome"][aria-checked='true']) a:hover:not([disabled]) {
      background-color: var(--ag-background-secondary-inverted);
    }

    button[disabled],
    a[disabled] {
      background-color: transparent;
      color: var(--ag-text-muted);
      cursor: not-allowed;
      opacity: 0.6;
    }
  `;
let l = v;
n([
  a()
], l.prototype, "value");
n([
  a({ type: Boolean })
], l.prototype, "disabled");
n([
  a()
], l.prototype, "href");
n([
  a()
], l.prototype, "target");
n([
  a({ type: Boolean })
], l.prototype, "checked");
n([
  a({ type: String, reflect: !0 })
], l.prototype, "variant");
n([
  a({ attribute: !1 })
], l.prototype, "onClick");
n([
  a({ attribute: !1 })
], l.prototype, "onMenuSelect");
const _ = class _ extends p {
  connectedCallback() {
    super.connectedCallback(), this.setAttribute("role", "separator");
  }
  render() {
    return u`<div class="separator" part="ag-menu-separator"></div>`;
  }
};
_.styles = b`
    :host {
      display: block;
    }
    .separator {
      border-top: 1px solid var(--ag-border);
      margin: var(--ag-space-2) 0;
    }
  `;
let k = _;
export {
  c as AgMenu,
  s as AgMenuButton,
  l as AgMenuItem,
  k as AgMenuSeparator
};
