// agnosticui-core@2.0.0-alpha.12/dist/VueAlert.vue_vue_type_script_setup_true_lang-BSEoqQWL.js
import { defineComponent as a, ref as l, createElementBlock as n, openBlock as f, mergeProps as m, renderSlot as u } from "vue";
import "./components/Alert/core/Alert.js";
const b = ["variant", "bordered", "rounded", "borderedLeft", "dismissible"], B = /* @__PURE__ */ a({
  __name: "VueAlert",
  props: {
    variant: { default: "default" },
    bordered: { type: Boolean, default: !1 },
    rounded: { type: Boolean, default: !1 },
    borderedLeft: { type: Boolean, default: !1 },
    dismissible: { type: Boolean, default: !1 }
  },
  emits: ["alert-dismiss"],
  setup(e, { emit: d }) {
    const s = d, r = l(), i = (t) => {
      s("alert-dismiss", t.detail);
    };
    return (t, o) => (f(), n("ag-alert", m({
      ref_key: "alertRef",
      ref: r,
      variant: e.variant,
      bordered: e.bordered || void 0,
      rounded: e.rounded || void 0,
      borderedLeft: e.borderedLeft || void 0,
      dismissible: e.dismissible || void 0,
      onAlertDismiss: i
    }, t.$attrs), [
      u(t.$slots, "default")
    ], 16, b));
  }
});
export {
  B as _
};
