// agnosticui-core@2.0.0-alpha.12/dist/VueAccordionItem.vue_vue_type_script_setup_true_lang-iL8z7vQz.js
import { defineComponent as u, ref as r, onMounted as s, onUnmounted as i, createElementBlock as c, openBlock as f, mergeProps as g, renderSlot as m } from "vue";
import "./components/Accordion/core/Accordion.js";
const v = [".open", ".disabled", ".useChevron", ".useX", ".useMinus", ".noIndicator", ".background", ".bordered", "headingLevel"], p = /* @__PURE__ */ u({
  __name: "VueAccordionItem",
  props: {
    open: { type: Boolean, default: !1 },
    headingLevel: { default: 3 },
    disabled: { type: Boolean, default: !1 },
    background: { type: Boolean, default: !1 },
    bordered: { type: Boolean, default: !1 },
    useChevron: { type: Boolean, default: !0 },
    useX: { type: Boolean, default: !1 },
    useMinus: { type: Boolean, default: !1 },
    noIndicator: { type: Boolean, default: !1 }
  },
  emits: ["toggle", "update:open"],
  setup(e, { emit: l }) {
    const a = l, n = r(), d = (o) => {
      const t = o.detail;
      a("toggle", t), a("update:open", t.open);
    };
    return s(async () => {
      await customElements.whenDefined("ag-accordion-item"), n.value && n.value.addEventListener("toggle", d);
    }), i(() => {
      n.value && n.value.removeEventListener("toggle", d);
    }), (o, t) => (f(), c("ag-accordion-item", g({
      ref_key: "accordionItemRef",
      ref: n,
      ".open": e.open,
      ".disabled": e.disabled,
      ".useChevron": e.useChevron,
      ".useX": e.useX,
      ".useMinus": e.useMinus,
      ".noIndicator": e.noIndicator,
      ".background": e.background,
      ".bordered": e.bordered,
      headingLevel: e.headingLevel
    }, o.$attrs), [
      m(o.$slots, "default")
    ], 48, v));
  }
});
export {
  p as _
};
