// agnosticui-core@2.0.0-alpha.12/dist/VueButtonFx.vue_vue_type_script_setup_true_lang-BRrt6Nxs.js
import { defineComponent as r, ref as g, watch as c, createElementBlock as b, openBlock as x, mergeProps as h, renderSlot as y } from "vue";
import "./components/ButtonFx/core/ButtonFx.js";
const B = [".disabled", ".loading", ".toggle", ".pressed", ".bordered", ".ghost", ".link", ".grouped", ".ariaLabel", ".fx", ".fxDisabled", "type", "variant", "size", "shape", "fx-speed", "fx-ease"], k = /* @__PURE__ */ r({
  __name: "VueButtonFx",
  props: {
    variant: { default: "" },
    size: { default: "md" },
    shape: { default: "" },
    bordered: { type: Boolean, default: !1 },
    ghost: { type: Boolean, default: !1 },
    link: { type: Boolean, default: !1 },
    grouped: { type: Boolean, default: !1 },
    fullWidth: { type: Boolean },
    type: { default: "button" },
    disabled: { type: Boolean, default: !1 },
    loading: { type: Boolean, default: !1 },
    toggle: { type: Boolean, default: !1 },
    pressed: { type: Boolean, default: !1 },
    ariaLabel: { default: "" },
    fx: {},
    fxSpeed: { default: "md" },
    fxEase: { default: "ease" },
    fxDisabled: { type: Boolean, default: !1 }
  },
  emits: ["click", "focus", "blur", "toggle", "update:pressed"],
  setup(e, { emit: s }) {
    const o = e, t = s, l = g(), n = (a) => {
      t("click", a);
    }, f = (a) => {
      t("focus", a);
    }, i = (a) => {
      t("blur", a);
    }, u = (a) => {
      const d = a;
      t("toggle", d.detail), t("update:pressed", d.detail.pressed);
    };
    return c(
      () => o.pressed,
      (a) => {
        l.value && o.toggle && (l.value.pressed = a);
      }
    ), (a, d) => (x(), b("ag-button-fx", h({
      ref_key: "buttonFxRef",
      ref: l,
      ".disabled": e.disabled,
      ".loading": e.loading,
      ".toggle": e.toggle,
      ".pressed": e.pressed,
      ".bordered": e.bordered,
      ".ghost": e.ghost,
      ".link": e.link,
      ".grouped": e.grouped,
      ".ariaLabel": e.ariaLabel,
      ".fx": e.fx,
      ".fxDisabled": e.fxDisabled,
      type: e.type,
      variant: e.variant,
      size: e.size,
      shape: e.shape,
      "fx-speed": e.fxSpeed,
      "fx-ease": e.fxEase,
      onClick: n,
      onFocus: f,
      onBlur: i,
      onToggle: u
    }, a.$attrs), [
      y(a.$slots, "default")
    ], 48, B));
  }
});
export {
  k as _
};
