// agnosticui-core@2.0.0-alpha.12/dist/VueBreadcrumb.vue_vue_type_script_setup_true_lang-Dp6vVm0c.js
import { defineComponent as u, ref as o, watch as s, onMounted as d, createElementBlock as b, openBlock as f, mergeProps as p } from "vue";
import "./components/Breadcrumb/core/Breadcrumb.js";
const y = ["type", ".primary", ".ariaLabel"], h = /* @__PURE__ */ u({
  __name: "VueBreadcrumb",
  props: {
    items: {},
    type: { default: "default" },
    primary: { type: Boolean, default: !1 },
    ariaLabel: { default: "Breadcrumb" }
  },
  emits: ["breadcrumb-click"],
  setup(e, { emit: m }) {
    const t = e, i = m, a = o(), n = (r) => {
      i("breadcrumb-click", r.detail);
    }, c = () => {
      a.value && (a.value.items = t.items);
    };
    return s(() => t.items, c, { deep: !0 }), d(async () => {
      await customElements.whenDefined("ag-breadcrumb"), a.value && c();
    }), (r, l) => (f(), b("ag-breadcrumb", p({
      ref_key: "breadcrumbRef",
      ref: a,
      type: e.type,
      ".primary": e.primary,
      ".ariaLabel": e.ariaLabel,
      onBreadcrumbClick: n
    }, r.$attrs), null, 48, y));
  }
});
export {
  h as _
};
