// agnosticui-core@2.0.0-alpha.12/dist/VueButton.vue_vue_type_script_setup_true_lang-D1jGatE9.js
import { defineComponent as r, ref as g, watch as c, createElementBlock as h, openBlock as b, mergeProps as p, renderSlot as y } from "vue";
import "./components/Button/core/Button.js";
const B = [".disabled", ".loading", ".toggle", ".pressed", ".bordered", ".ghost", ".link", ".grouped", ".fullWidth", ".ariaLabel", "type", "variant", "size", "shape"], v = /* @__PURE__ */ r({
  __name: "VueButton",
  props: {
    variant: { default: "" },
    size: { default: "md" },
    shape: { default: "" },
    bordered: { type: Boolean, default: !1 },
    ghost: { type: Boolean, default: !1 },
    link: { type: Boolean, default: !1 },
    grouped: { type: Boolean, default: !1 },
    fullWidth: { type: Boolean, default: !1 },
    type: { default: "button" },
    disabled: { type: Boolean, default: !1 },
    loading: { type: Boolean, default: !1 },
    toggle: { type: Boolean, default: !1 },
    pressed: { type: Boolean, default: !1 },
    ariaLabel: { default: "" }
  },
  emits: ["click", "focus", "blur", "toggle", "update:pressed"],
  setup(e, { emit: n }) {
    const d = e, l = n, a = g(), s = (t) => {
      l("click", t);
    }, u = (t) => {
      l("focus", t);
    }, i = (t) => {
      l("blur", t);
    }, f = (t) => {
      const o = t;
      l("toggle", o.detail), l("update:pressed", o.detail.pressed);
    };
    return c(
      () => d.pressed,
      (t) => {
        a.value && d.toggle && (a.value.pressed = t);
      }
    ), (t, o) => (b(), h("ag-button", p({
      ref_key: "buttonRef",
      ref: a,
      ".disabled": e.disabled,
      ".loading": e.loading,
      ".toggle": e.toggle,
      ".pressed": e.pressed,
      ".bordered": e.bordered,
      ".ghost": e.ghost,
      ".link": e.link,
      ".grouped": e.grouped,
      ".fullWidth": e.fullWidth,
      ".ariaLabel": e.ariaLabel,
      type: e.type,
      variant: e.variant,
      size: e.size,
      shape: e.shape,
      onClick: s,
      onFocus: u,
      onBlur: i,
      onToggle: f
    }, t.$attrs), [
      y(t.$slots, "default")
    ], 48, B));
  }
});
export {
  v as _
};
