// agnosticui-core@2.0.0-alpha.12/dist/VueBadgeFx.vue_vue_type_script_setup_true_lang-B5Bex8CH.js
import { defineComponent as l, ref as f, createElementBlock as d, openBlock as n, mergeProps as i, renderSlot as u } from "vue";
import "./components/BadgeFx/core/BadgeFx.js";
const s = ["variant", "size", ".dot", ".interactive", ".hiddenFromAT", "value", "max", "status-label", "live", "fx", "fx-speed", "fx-ease", ".fxDisabled"], r = /* @__PURE__ */ l({
  __name: "VueBadgeFx",
  props: {
    variant: { default: "default" },
    size: { default: "md" },
    dot: { type: Boolean, default: !1 },
    value: { default: null },
    max: { default: 99 },
    interactive: { type: Boolean, default: !1 },
    statusLabel: { default: null },
    live: { default: "off" },
    hiddenFromAT: { type: Boolean, default: !1 },
    fx: {},
    fxSpeed: { default: "md" },
    fxEase: { default: "ease" },
    fxDisabled: { type: Boolean, default: !1 }
  },
  setup(e) {
    const t = f(null);
    return (a, o) => (n(), d("ag-badge-fx", i({
      ref_key: "agComponent",
      ref: t,
      variant: e.variant,
      size: e.size,
      ".dot": e.dot,
      ".interactive": e.interactive,
      ".hiddenFromAT": e.hiddenFromAT,
      value: e.value,
      max: e.max,
      "status-label": e.statusLabel,
      live: e.live,
      fx: e.fx,
      "fx-speed": e.fxSpeed,
      "fx-ease": e.fxEase,
      ".fxDisabled": e.fxDisabled
    }, a.$attrs), [
      u(a.$slots, "default")
    ], 48, s));
  }
});
export {
  r as _
};
