// agnosticui-core@2.0.0-alpha.12/dist/VueBadge.vue_vue_type_script_setup_true_lang-DAPP5_-i.js
import { defineComponent as l, ref as n, createElementBlock as d, openBlock as i, mergeProps as o, renderSlot as u } from "vue";
import "./components/Badge/core/Badge.js";
const f = ["variant", "size", ".dot", ".interactive", ".hiddenFromAT", "value", "max", "status-label", "live"], v = /* @__PURE__ */ l({
  __name: "VueBadge",
  props: {
    variant: { default: "default" },
    size: { default: "md" },
    dot: { type: Boolean, default: !1 },
    value: { default: null },
    max: { default: 99 },
    interactive: { type: Boolean, default: !1 },
    statusLabel: { default: null },
    live: { default: "off" },
    hiddenFromAT: { type: Boolean, default: !1 }
  },
  setup(e) {
    const t = n(null);
    return (a, r) => (i(), d("ag-badge", o({
      ref_key: "agComponent",
      ref: t,
      variant: e.variant,
      size: e.size,
      ".dot": e.dot,
      ".interactive": e.interactive,
      ".hiddenFromAT": e.hiddenFromAT,
      value: e.value,
      max: e.max,
      "status-label": e.statusLabel,
      live: e.live
    }, a.$attrs), [
      u(a.$slots, "default")
    ], 48, f));
  }
});
export {
  v as _
};
