// agnosticui-core@2.0.0-alpha.12/dist/VueCombobox.vue_vue_type_script_setup_true_lang-C7gDMrYJ.js
import { defineComponent as m, ref as v, createElementBlock as h, openBlock as x, renderSlot as g } from "vue";
import "./components/Combobox/core/Combobox.js";
const y = [".labelHidden", ".labelPosition", ".noLabel", ".disabled", ".invalid", ".clearable", ".readonly", ".required", ".loading", ".multiple", ".closeOnSelect", ".options", ".value", ".defaultValue", ".maxOptionsVisible", "placeholder", "label", "ariaLabel", "helpText", "errorMessage", "autocomplete", "filterMode", "size", "variant", "loadingText", "noResultsText"], T = /* @__PURE__ */ m({
  __name: "VueCombobox",
  props: {
    options: { default: () => [] },
    value: { default: void 0 },
    defaultValue: { default: void 0 },
    placeholder: { default: "" },
    label: { default: "" },
    labelPosition: { default: "top" },
    labelHidden: { type: Boolean, default: !1 },
    noLabel: { type: Boolean, default: !1 },
    ariaLabel: { default: null },
    helpText: { default: "" },
    errorMessage: { default: "" },
    id: {},
    autocomplete: { default: "list" },
    filterMode: { default: "startsWith" },
    clearable: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    readonly: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    invalid: { type: Boolean, default: !1 },
    size: { default: "default" },
    maxVisibleOptions: {},
    closeOnSelect: { type: Boolean, default: !0 },
    variant: { default: "default" },
    multiple: { type: Boolean, default: !1 },
    maxOptionsVisible: { default: 3 },
    loading: { type: Boolean, default: !1 },
    loadingText: { default: "Loading..." },
    noResultsText: { default: "No results found" }
  },
  emits: ["change", "select", "search", "open", "close", "focus", "blur", "update:value"],
  setup(e, { expose: n, emit: d }) {
    const a = d, t = v(), i = (l) => {
      const o = l;
      a("change", o.detail), a("update:value", o.detail.value);
    }, u = (l) => {
      a("select", l.detail);
    }, s = (l) => {
      a("search", l.detail);
    }, c = (l) => {
      a("open", l.detail);
    }, f = (l) => {
      a("close", l.detail);
    }, r = (l) => {
      a("focus", l);
    }, b = (l) => {
      a("blur", l);
    };
    return n({
      focus: () => t.value?.focus?.(),
      blur: () => t.value?.blur?.(),
      open: () => t.value?.open?.(),
      close: () => t.value?.close?.(),
      toggle: () => t.value?.toggle?.(),
      selectOption: (l) => t.value?.selectOption?.(l),
      clearSelection: () => t.value?.clearSelection?.()
    }), (l, o) => (x(), h("ag-combobox", {
      ref_key: "comboboxRef",
      ref: t,
      ".labelHidden": e.labelHidden,
      ".labelPosition": e.labelPosition,
      ".noLabel": e.noLabel,
      ".disabled": e.disabled,
      ".invalid": e.invalid,
      ".clearable": e.clearable,
      ".readonly": e.readonly,
      ".required": e.required,
      ".loading": e.loading,
      ".multiple": e.multiple,
      ".closeOnSelect": e.closeOnSelect,
      ".options": e.options,
      ".value": e.value,
      ".defaultValue": e.defaultValue,
      ".maxOptionsVisible": e.maxOptionsVisible,
      placeholder: e.placeholder,
      label: e.label,
      ariaLabel: e.ariaLabel,
      helpText: e.helpText,
      errorMessage: e.errorMessage,
      autocomplete: e.autocomplete,
      filterMode: e.filterMode,
      size: e.size,
      variant: e.variant,
      loadingText: e.loadingText,
      noResultsText: e.noResultsText,
      onChange: i,
      onSelect: u,
      onSearch: s,
      onOpen: c,
      onClose: f,
      onFocus: r,
      onBlur: b
    }, [
      g(l.$slots, "default")
    ], 40, y));
  }
});
export {
  T as _
};
