// agnosticui-core@2.0.0-alpha.12/dist/VueCheckbox.vue_vue_type_script_setup_true_lang-VtQfJ-6f.js
import { defineComponent as c, createElementBlock as o, openBlock as u, mergeProps as r, renderSlot as s } from "vue";
import "./components/Checkbox/core/Checkbox.js";
const m = [".checked", ".indeterminate", ".disabled", ".required", ".invalid", "name", "value", "size", "theme", "labelText", "labelPosition", "errorMessage", "helpText"], k = /* @__PURE__ */ c({
  __name: "VueCheckbox",
  props: {
    name: { default: "" },
    value: { default: "" },
    checked: { type: Boolean, default: !1 },
    indeterminate: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    size: { default: "medium" },
    theme: { default: "default" },
    labelText: { default: "" },
    labelPosition: { default: "end" },
    required: { type: Boolean, default: !1 },
    invalid: { type: Boolean, default: !1 },
    errorMessage: { default: "" },
    helpText: { default: "" }
  },
  emits: ["click", "change", "update:checked", "update:indeterminate"],
  setup(e, { emit: d }) {
    const t = d, i = (a) => {
      t("click", a);
    }, n = (a) => {
      const l = a;
      t("change", l.detail), t("update:checked", l.detail.checked), t("update:indeterminate", l.detail.indeterminate);
    };
    return (a, l) => (u(), o("ag-checkbox", r({
      ref: "checkboxRef",
      ".checked": e.checked,
      ".indeterminate": e.indeterminate,
      ".disabled": e.disabled,
      ".required": e.required,
      ".invalid": e.invalid,
      name: e.name,
      value: e.value,
      size: e.size,
      theme: e.theme,
      labelText: e.labelText,
      labelPosition: e.labelPosition,
      errorMessage: e.errorMessage,
      helpText: e.helpText,
      onClick: i,
      onChange: n
    }, a.$attrs), [
      s(a.$slots, "default")
    ], 48, m));
  }
});
export {
  k as _
};
