// agnosticui-core@2.0.0-alpha.12/dist/VueCollapsible.vue_vue_type_script_setup_true_lang-2DYXeukJ.js
import { defineComponent as r, ref as u, createElementBlock as d, openBlock as i, mergeProps as f, createElementVNode as c, renderSlot as t } from "vue";
import "./components/Collapsible/core/Collapsible.js";
const p = [".open", ".bordered", ".shadow", ".useChevron", ".useX", ".useMinus", ".noIndicator"], m = { slot: "summary" }, B = /* @__PURE__ */ r({
  __name: "VueCollapsible",
  props: {
    open: { type: Boolean, default: !1 },
    bordered: { type: Boolean, default: !1 },
    shadow: { type: Boolean, default: !1 },
    useChevron: { type: Boolean, default: !0 },
    useX: { type: Boolean, default: !1 },
    useMinus: { type: Boolean, default: !1 },
    noIndicator: { type: Boolean, default: !1 }
  },
  emits: ["toggle", "update:open"],
  setup(e, { emit: a }) {
    const n = a, l = u(), s = (o) => {
      n("toggle", o), n("update:open", o.detail.open);
    };
    return (o, h) => (i(), d("ag-collapsible", f({
      ref_key: "collapsibleRef",
      ref: l,
      ".open": e.open,
      ".bordered": e.bordered,
      ".shadow": e.shadow,
      ".useChevron": e.useChevron,
      ".useX": e.useX,
      ".useMinus": e.useMinus,
      ".noIndicator": e.noIndicator,
      ".onToggle": s
    }, o.$attrs), [
      c("span", m, [
        t(o.$slots, "summary")
      ]),
      t(o.$slots, "default")
    ], 48, p));
  }
});
export {
  B as _
};
