// agnosticui-core@2.0.0-alpha.12/dist/VueCopyButton.vue_vue_type_script_setup_true_lang-BZi5H42P.js
import { defineComponent as c, ref as u, createElementBlock as r, openBlock as s, mergeProps as d, createCommentVNode as n, renderSlot as i } from "vue";
import "./components/CopyButton/core/CopyButton.js";
const m = ["text", "label", "success-label", "error-label", "timeout", "size", "variant"], p = {
  key: 0,
  slot: "icon-copy"
}, y = {
  key: 1,
  slot: "icon-copied"
}, b = {
  key: 2,
  slot: "icon-error"
}, $ = /* @__PURE__ */ c({
  __name: "VueCopyButton",
  props: {
    text: {},
    label: { default: "Copy to clipboard" },
    successLabel: { default: "Copied!" },
    errorLabel: { default: "Copy failed" },
    timeout: { default: 1e3 },
    size: { default: "md" },
    variant: { default: "ghost" }
  },
  emits: ["copy", "copy-error"],
  setup(o) {
    const a = u(null);
    return (e, t) => (s(), r("ag-copy-button", d({
      ref_key: "agComponent",
      ref: a,
      text: o.text,
      label: o.label,
      "success-label": o.successLabel,
      "error-label": o.errorLabel,
      timeout: o.timeout,
      size: o.size,
      variant: o.variant
    }, e.$attrs, {
      onCopy: t[0] || (t[0] = (l) => e.$emit("copy", l)),
      onCopyError: t[1] || (t[1] = (l) => e.$emit("copy-error", l))
    }), [
      e.$slots["icon-copy"] ? (s(), r("span", p, [
        i(e.$slots, "icon-copy")
      ])) : n("", !0),
      e.$slots["icon-copied"] ? (s(), r("span", y, [
        i(e.$slots, "icon-copied")
      ])) : n("", !0),
      e.$slots["icon-error"] ? (s(), r("span", b, [
        i(e.$slots, "icon-error")
      ])) : n("", !0)
    ], 16, m));
  }
});
export {
  $ as _
};
