// agnosticui-core@2.0.0-alpha.12/dist/VueContentPagination.vue_vue_type_script_setup_true_lang-DSQQBA4M.js
import { defineComponent as f, useAttrs as v, computed as g, ref as b, createElementBlock as n, openBlock as o, mergeProps as h, createCommentVNode as s, renderSlot as r } from "vue";
import "./components/Pagination/core/ContentPagination.js";
const k = ["aria-label", ".bordered"], y = {
  key: 0,
  slot: "previous-icon"
}, _ = {
  key: 1,
  slot: "next-icon"
}, $ = {
  key: 2,
  slot: "parent-icon"
}, B = /* @__PURE__ */ f({
  __name: "VueContentPagination",
  props: {
    previous: {},
    next: {},
    parent: {},
    bordered: { type: Boolean, default: !1 },
    ariaLabel: { default: "content navigation" }
  },
  emits: ["navigate"],
  setup(a, { emit: c }) {
    const l = v(), p = g(() => {
      const e = { ...l };
      return t.previous && (e.previous = t.previous), t.next && (e.next = t.next), t.parent && (e.parent = t.parent), e;
    }), t = a, d = c, u = b(), m = (e) => {
      const i = e.detail;
      d("navigate", i);
    };
    return (e, i) => (o(), n("ag-content-pagination", h({
      ref_key: "contentPaginationRef",
      ref: u,
      "aria-label": a.ariaLabel,
      ".bordered": a.bordered,
      onNavigate: m
    }, p.value), [
      e.$slots["previous-icon"] ? (o(), n("span", y, [
        r(e.$slots, "previous-icon")
      ])) : s("", !0),
      e.$slots["next-icon"] ? (o(), n("span", _, [
        r(e.$slots, "next-icon")
      ])) : s("", !0),
      e.$slots["parent-icon"] ? (o(), n("span", $, [
        r(e.$slots, "parent-icon")
      ])) : s("", !0)
    ], 48, k));
  }
});
export {
  B as _
};
