// agnosticui-core@2.0.0-alpha.12/dist/VueDialog.vue_vue_type_script_setup_true_lang-1MD8xeGw.js
import { defineComponent as i, ref as c, createElementBlock as p, openBlock as g, mergeProps as r, renderSlot as f } from "vue";
import "./components/Dialog/core/Dialog.js";
const u = [".open", ".noCloseOnEscape", ".noCloseOnBackdrop", ".showCloseButton", "heading", "description"], m = /* @__PURE__ */ i({
  __name: "VueDialog",
  props: {
    open: { type: Boolean, default: !1 },
    heading: {},
    description: {},
    noCloseOnEscape: { type: Boolean, default: !1 },
    noCloseOnBackdrop: { type: Boolean, default: !1 },
    showCloseButton: { type: Boolean, default: !1 }
  },
  emits: ["dialog-open", "dialog-close", "dialog-cancel", "update:open"],
  setup(o, { emit: a }) {
    const n = a, l = c(), t = (e) => {
      n("dialog-open", e), n("update:open", !0);
    }, s = (e) => {
      n("dialog-close", e), n("update:open", !1);
    }, d = (e) => {
      n("dialog-cancel", e), n("update:open", !1);
    };
    return (e, C) => (g(), p("ag-dialog", r({
      ref_key: "dialogRef",
      ref: l,
      ".open": o.open,
      ".noCloseOnEscape": o.noCloseOnEscape,
      ".noCloseOnBackdrop": o.noCloseOnBackdrop,
      ".showCloseButton": o.showCloseButton,
      heading: o.heading,
      description: o.description,
      onDialogOpen: t,
      onDialogClose: s,
      onDialogCancel: d
    }, e.$attrs), [
      f(e.$slots, "default")
    ], 48, u));
  }
});
export {
  m as _
};
