// agnosticui-core@2.0.0-alpha.12/dist/VueDrawer.vue_vue_type_script_setup_true_lang-ofDg73MZ.js
import { defineComponent as d, ref as p, createElementBlock as c, openBlock as i, mergeProps as f, renderSlot as u } from "vue";
import "./components/Drawer/core/Drawer.js";
const w = [".open", ".noCloseOnEscape", ".noCloseOnBackdrop", ".showCloseButton", ".position", "heading", "description"], m = /* @__PURE__ */ d({
  __name: "VueDrawer",
  props: {
    open: { type: Boolean, default: !1 },
    heading: { default: "" },
    description: { default: "" },
    noCloseOnEscape: { type: Boolean, default: !1 },
    noCloseOnBackdrop: { type: Boolean, default: !1 },
    showCloseButton: { type: Boolean, default: !1 },
    position: { default: "bottom" }
  },
  emits: ["drawer-open", "drawer-close", "drawer-cancel", "update:open"],
  setup(o, { emit: a }) {
    const n = a, t = p(), s = (e) => {
      n("drawer-open", e), n("update:open", !0);
    }, r = (e) => {
      n("drawer-close", e), n("update:open", !1);
    }, l = (e) => {
      n("drawer-cancel", e), n("update:open", !1);
    };
    return (e, C) => (i(), c("ag-drawer", f({
      ref_key: "drawerRef",
      ref: t,
      ".open": o.open,
      ".noCloseOnEscape": o.noCloseOnEscape,
      ".noCloseOnBackdrop": o.noCloseOnBackdrop,
      ".showCloseButton": o.showCloseButton,
      ".position": o.position,
      heading: o.heading,
      description: o.description,
      onDrawerOpen: s,
      onDrawerClose: r,
      onDrawerCancel: l
    }, e.$attrs), [
      u(e.$slots, "default")
    ], 48, w));
  }
});
export {
  m as _
};
