// agnosticui-core@2.0.0-alpha.12/dist/VueEmptyState.vue_vue_type_script_setup_true_lang-BS6VTxrx.js
import { defineComponent as r, ref as a, createElementBlock as o, openBlock as n, mergeProps as i, createCommentVNode as s, renderSlot as d } from "vue";
import "./components/EmptyState/core/EmptyState.js";
const u = [".bordered", ".rounded", "title", "subtitle", "buttonText", "size"], m = {
  key: 0,
  slot: "icon"
}, c = {
  key: 1,
  slot: "actions"
}, y = /* @__PURE__ */ r({
  __name: "VueEmptyState",
  props: {
    title: { default: "" },
    subtitle: { default: "" },
    buttonText: { default: "" },
    size: { default: "md" },
    bordered: { type: Boolean, default: !1 },
    rounded: { type: Boolean, default: !1 }
  },
  setup(e) {
    const l = a();
    return (t, f) => (n(), o("ag-empty-state", i({
      ref_key: "agComponent",
      ref: l,
      ".bordered": e.bordered,
      ".rounded": e.rounded,
      title: e.title,
      subtitle: e.subtitle,
      buttonText: e.buttonText,
      size: e.size
    }, t.$attrs), [
      t.$slots.icon ? (n(), o("span", m, [
        d(t.$slots, "icon")
      ])) : s("", !0),
      t.$slots.actions ? (n(), o("span", c, [
        d(t.$slots, "actions")
      ])) : s("", !0)
    ], 48, u));
  }
});
export {
  y as _
};
