// agnosticui-core@2.0.0-alpha.12/dist/VueInput.vue_vue_type_script_setup_true_lang-cOALNmlO.js
import { defineComponent as b, createElementBlock as n, openBlock as d, mergeProps as y, createCommentVNode as o, renderSlot as u } from "vue";
import "./components/Input/core/Input.js";
const g = [".value", ".labelHidden", ".noLabel", ".capsule", ".rounded", ".underlined", ".underlinedWithBackground", ".inline", ".required", ".disabled", ".readonly", ".invalid", "type", "placeholder", "label", "ariaLabel", "helpText", "errorMessage", "size", "rows", "cols"], B = {
  key: 0,
  slot: "addon-left"
}, m = {
  key: 1,
  slot: "addon-right"
}, L = /* @__PURE__ */ b({
  __name: "VueInput",
  props: {
    label: { default: "" },
    labelHidden: { type: Boolean, default: !1 },
    labelPosition: {},
    noLabel: { type: Boolean, default: !1 },
    ariaLabel: { default: "" },
    type: { default: "text" },
    value: { default: "" },
    placeholder: { default: "" },
    rows: { default: 4 },
    cols: { default: 50 },
    size: { default: "default" },
    capsule: { type: Boolean, default: !1 },
    rounded: { type: Boolean, default: !1 },
    underlined: { type: Boolean, default: !1 },
    underlinedWithBackground: { type: Boolean, default: !1 },
    inline: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    readonly: { type: Boolean, default: !1 },
    invalid: { type: Boolean, default: !1 },
    errorMessage: { default: "" },
    helpText: { default: "" }
  },
  emits: ["click", "input", "change", "focus", "blur", "update:value"],
  setup(e, { emit: i }) {
    const a = i, r = (l) => {
      a("click", l);
    }, s = (l) => {
      a("input", l);
      const t = l.target;
      a("update:value", t.value);
    }, c = (l) => {
      a("change", l);
    }, f = (l) => {
      a("focus", l);
    }, h = (l) => {
      a("blur", l);
    };
    return (l, t) => (d(), n("ag-input", y({
      ".value": e.value,
      ".labelHidden": e.labelHidden,
      ".noLabel": e.noLabel,
      ".capsule": e.capsule,
      ".rounded": e.rounded,
      ".underlined": e.underlined,
      ".underlinedWithBackground": e.underlinedWithBackground,
      ".inline": e.inline,
      ".required": e.required,
      ".disabled": e.disabled,
      ".readonly": e.readonly,
      ".invalid": e.invalid,
      type: e.type,
      placeholder: e.placeholder,
      label: e.label,
      ariaLabel: e.ariaLabel,
      helpText: e.helpText,
      errorMessage: e.errorMessage,
      size: e.size,
      rows: e.rows,
      cols: e.cols,
      onClick: r,
      onInput: s,
      onChange: c,
      onFocus: f,
      onBlur: h
    }, l.$attrs), [
      l.$slots["addon-left"] ? (d(), n("span", B, [
        u(l.$slots, "addon-left")
      ])) : o("", !0),
      l.$slots["addon-right"] ? (d(), n("span", m, [
        u(l.$slots, "addon-right")
      ])) : o("", !0)
    ], 48, g));
  }
});
export {
  L as _
};
