// agnosticui-core@2.0.0-alpha.12/dist/VueImage.vue_vue_type_script_setup_true_lang-SddK93SB.js
import { defineComponent as d, ref as s, onMounted as f, watchEffect as u, createElementBlock as g, openBlock as h, mergeProps as m, renderSlot as o } from "vue";
import "./components/Image/core/Image.js";
const k = ["src", "alt", "width", "height", "aspect-ratio", "fit", "position", "loading", ".fade", "duration", "fallback-src"], E = /* @__PURE__ */ d({
  __name: "VueImage",
  props: {
    sources: {},
    src: { default: "" },
    alt: { default: "" },
    width: {},
    height: {},
    aspectRatio: {},
    fit: { default: "cover" },
    position: { default: "center" },
    loading: { default: "lazy" },
    fade: { type: Boolean, default: !1 },
    duration: { default: 200 },
    fallbackSrc: {}
  },
  emits: ["ag-load", "ag-error"],
  setup(e, { emit: i }) {
    const l = e, a = s(null);
    f(() => {
      u(() => {
        a.value && l.sources && (a.value.sources = l.sources);
      });
    });
    const r = i, n = (t) => {
      r("ag-load");
    }, c = (t) => {
      r("ag-error");
    };
    return (t, v) => (h(), g("ag-image", m({
      ref_key: "agImageRef",
      ref: a,
      src: e.src,
      alt: e.alt,
      width: e.width,
      height: e.height,
      "aspect-ratio": e.aspectRatio,
      fit: e.fit,
      position: e.position,
      loading: e.loading,
      ".fade": e.fade,
      duration: e.duration,
      "fallback-src": e.fallbackSrc,
      onAgLoad: n,
      onAgError: c
    }, t.$attrs), [
      o(t.$slots, "placeholder", { slot: "placeholder" }),
      o(t.$slots, "error", { slot: "error" }),
      o(t.$slots, "default")
    ], 48, k));
  }
});
export {
  E as _
};
