// agnosticui-core@2.0.0-alpha.12/dist/VueIntlFormatter.vue_vue_type_script_setup_true_lang-evMy310u.js
import { defineComponent as o, ref as d, createElementBlock as r, openBlock as l, mergeProps as c } from "vue";
import "./components/IntlFormatter/core/IntlFormatter.js";
const f = ["type", "value", "lang", "date", "weekday", "era", "year", "month", "day", "hour", "minute", "second", "timeZoneName", "timeZone", "hourFormat", "dateStyle", "timeStyle", ".noGrouping", "currency", "currencyDisplay", "minimumIntegerDigits", "minimumFractionDigits", "maximumFractionDigits", "minimumSignificantDigits", "maximumSignificantDigits"], v = /* @__PURE__ */ o({
  __name: "VueIntlFormatter",
  props: {
    type: { default: "date" },
    value: { default: void 0 },
    lang: { default: void 0 },
    date: { default: void 0 },
    weekday: { default: void 0 },
    era: { default: void 0 },
    year: { default: void 0 },
    month: { default: void 0 },
    day: { default: void 0 },
    hour: { default: void 0 },
    minute: { default: void 0 },
    second: { default: void 0 },
    timeZoneName: { default: void 0 },
    timeZone: { default: void 0 },
    hourFormat: { default: "auto" },
    dateStyle: { default: void 0 },
    timeStyle: { default: void 0 },
    noGrouping: { type: Boolean, default: !1 },
    currency: { default: "USD" },
    currencyDisplay: { default: "symbol" },
    minimumIntegerDigits: { default: void 0 },
    minimumFractionDigits: { default: void 0 },
    maximumFractionDigits: { default: void 0 },
    minimumSignificantDigits: { default: void 0 },
    maximumSignificantDigits: { default: void 0 }
  },
  emits: ["format-error"],
  setup(i, { emit: t }) {
    const a = t, m = d(), n = (e) => {
      a("format-error", e.detail);
    };
    return (e, u) => (l(), r("ag-intl-formatter", c({
      ref_key: "formatterRef",
      ref: m,
      type: i.type,
      value: i.value,
      lang: i.lang,
      date: i.date,
      weekday: i.weekday,
      era: i.era,
      year: i.year,
      month: i.month,
      day: i.day,
      hour: i.hour,
      minute: i.minute,
      second: i.second,
      timeZoneName: i.timeZoneName,
      timeZone: i.timeZone,
      hourFormat: i.hourFormat,
      dateStyle: i.dateStyle,
      timeStyle: i.timeStyle,
      ".noGrouping": i.noGrouping,
      currency: i.currency,
      currencyDisplay: i.currencyDisplay,
      minimumIntegerDigits: i.minimumIntegerDigits,
      minimumFractionDigits: i.minimumFractionDigits,
      maximumFractionDigits: i.maximumFractionDigits,
      minimumSignificantDigits: i.minimumSignificantDigits,
      maximumSignificantDigits: i.maximumSignificantDigits,
      onFormatError: n
    }, e.$attrs), null, 48, f));
  }
});
export {
  v as _
};
