// agnosticui-core@2.0.0-alpha.12/dist/VueMessageBubble.vue_vue_type_script_setup_true_lang-CjDTnJfG.js
import { defineComponent as n, ref as f, createElementBlock as a, openBlock as o, mergeProps as u, renderSlot as r, createCommentVNode as s } from "vue";
import "./components/MessageBubble/core/MessageBubble.js";
const m = ["from", "message", "time", "author", "avatar-url", "footer", "variant"], d = {
  key: 1,
  slot: "header"
}, i = {
  key: 2,
  slot: "footer"
}, h = {
  key: 3,
  slot: "avatar"
}, k = /* @__PURE__ */ n({
  __name: "VueMessageBubble",
  props: {
    from: { default: "them" },
    message: { default: "" },
    time: { default: "" },
    author: { default: "" },
    avatarUrl: { default: "" },
    footer: { default: "" },
    variant: { default: "default" }
  },
  setup(t) {
    const l = f(null);
    return (e, v) => (o(), a("ag-message-bubble", u({
      ref_key: "agComponent",
      ref: l,
      from: t.from,
      message: t.message,
      time: t.time,
      author: t.author,
      "avatar-url": t.avatarUrl,
      footer: t.footer,
      variant: t.variant
    }, e.$attrs), [
      e.$slots.default ? r(e.$slots, "default", { key: 0 }) : s("", !0),
      e.$slots.header ? (o(), a("span", d, [
        r(e.$slots, "header")
      ])) : s("", !0),
      e.$slots.footer ? (o(), a("span", i, [
        r(e.$slots, "footer")
      ])) : s("", !0),
      e.$slots.avatar ? (o(), a("span", h, [
        r(e.$slots, "avatar")
      ])) : s("", !0)
    ], 16, m));
  }
});
export {
  k as _
};
