// agnosticui-core@2.0.0-alpha.12/dist/VueMenu.vue_vue_type_script_setup_true_lang-BXc7397e.js
import { defineComponent as h, ref as o, createElementBlock as k, openBlock as p, mergeProps as B, renderSlot as l, createElementVNode as v } from "vue";
import "./components/Menu/core/Menu.js";
const w = [".disabled", ".loading", ".bordered", ".ghost", ".link", ".grouped", "menu-variant", "menu-align", "button-variant", "size", "shape", "ariaLabel", "unicode", "additional-gutter"], L = [".open", ".checkHiddenItems", "placement", "type", "selected-value", "ariaLabel"], C = /* @__PURE__ */ h({
  __name: "VueMenu",
  props: {
    open: { type: Boolean, default: !1 },
    placement: { default: "bottom-start" },
    menuAriaLabel: {},
    menuAlign: { default: "left" },
    menuType: { default: "default" },
    selectedValue: {},
    checkHiddenItems: { type: Boolean, default: !1 },
    menuVariant: {},
    buttonVariant: {},
    unicode: {},
    additionalGutter: {},
    size: {},
    shape: {},
    bordered: { type: Boolean },
    ghost: { type: Boolean },
    link: { type: Boolean },
    grouped: { type: Boolean },
    fullWidth: { type: Boolean },
    type: {},
    disabled: { type: Boolean, default: !1 },
    loading: { type: Boolean },
    toggle: { type: Boolean },
    pressed: { type: Boolean },
    ariaLabel: {},
    onToggle: {}
  },
  emits: ["click", "focus", "blur", "keydown", "menu-keydown", "menu-open", "menu-close", "menu-select", "update:open"],
  setup(e, { emit: u }) {
    const t = u, d = o(), i = o(), c = (n) => {
      t("click", n);
    }, m = (n) => {
      t("focus", n);
    }, s = (n) => {
      t("blur", n);
    }, r = (n) => {
      t("keydown", n);
    }, f = (n) => {
      t("menu-keydown", n);
    }, y = (n) => {
      const a = n;
      t("menu-open", a.detail), t("update:open", a.detail.open);
    }, g = (n) => {
      const a = n;
      t("menu-close", a.detail), t("update:open", a.detail.open);
    }, b = (n) => {
      t("menu-select", n.detail);
    };
    return (n, a) => (p(), k("ag-menu-button", B({
      ref_key: "menuButtonRef",
      ref: d,
      ".disabled": e.disabled,
      ".loading": e.loading,
      ".bordered": e.bordered,
      ".ghost": e.ghost,
      ".link": e.link,
      ".grouped": e.grouped,
      "menu-variant": e.menuVariant,
      "menu-align": e.menuAlign,
      "button-variant": e.buttonVariant,
      size: e.size,
      shape: e.shape,
      ariaLabel: e.ariaLabel,
      unicode: e.unicode,
      "additional-gutter": e.additionalGutter,
      onClick: c,
      onFocus: m,
      onBlur: s,
      onKeydown: r,
      onMenuOpen: y,
      onMenuClose: g,
      onMenuSelect: b
    }, n.$attrs), [
      l(n.$slots, "default"),
      v("ag-menu", {
        ref_key: "menuRef",
        ref: i,
        slot: "menu",
        ".open": e.open,
        ".checkHiddenItems": e.checkHiddenItems,
        placement: e.placement,
        type: e.menuType,
        "selected-value": e.selectedValue,
        ariaLabel: e.menuAriaLabel,
        onKeydown: f
      }, [
        l(n.$slots, "menu")
      ], 40, L)
    ], 48, w));
  }
});
export {
  C as _
};
