// agnosticui-core@2.0.0-alpha.12/dist/VueMenuItem.vue_vue_type_script_setup_true_lang-CXdZ61KO.js
import { defineComponent as o, ref as r, createElementBlock as i, openBlock as f, mergeProps as m, renderSlot as s } from "vue";
import "./components/Menu/core/Menu.js";
const h = ["value", "disabled", "href", "target", "variant", "checked"], g = /* @__PURE__ */ o({
  __name: "VueMenuItem",
  props: {
    selected: { type: Boolean, default: !1 },
    value: { default: "" },
    disabled: { type: Boolean, default: !1 },
    href: {},
    target: {},
    checked: { type: Boolean, default: !1 },
    variant: { default: "default" }
  },
  emits: ["click", "menu-select"],
  setup(e, { emit: l }) {
    const a = l, n = r(), c = (t) => {
      a("click", t);
    }, d = (t) => {
      a("menu-select", t.detail);
    };
    return (t, u) => (f(), i("ag-menu-item", m({
      ref_key: "menuItemRef",
      ref: n,
      value: e.value,
      disabled: e.disabled || void 0,
      href: e.href,
      target: e.target,
      variant: e.variant,
      checked: e.checked || e.selected,
      onClick: c,
      onMenuSelect: d
    }, t.$attrs), [
      s(t.$slots, "default")
    ], 16, h));
  }
});
export {
  g as _
};
