// agnosticui-core@2.0.0-alpha.12/dist/VuePagination.vue_vue_type_script_setup_true_lang-Bbf1X690.js
import { defineComponent as s, ref as f, createElementBlock as l, openBlock as g, mergeProps as u, renderSlot as d } from "vue";
import "./components/Pagination/core/Pagination.js";
const c = [".bordered", ".firstLastNavigation", ".navigationLabels", "current", "total-pages", "offset", "justify", "aria-label"], L = /* @__PURE__ */ s({
  __name: "VuePagination",
  props: {
    current: { default: 1 },
    totalPages: { default: 1 },
    offset: { default: 2 },
    justify: { default: "" },
    ariaLabel: { default: "pagination" },
    bordered: { type: Boolean, default: !1 },
    firstLastNavigation: { type: Boolean, default: !1 },
    navigationLabels: { default: () => ({
      first: "First",
      previous: "Previous",
      next: "Next",
      last: "Last"
    }) }
  },
  emits: ["page-change", "update:current"],
  setup(e, { emit: i }) {
    const n = i, o = f(), r = (a) => {
      const t = a.detail;
      n("page-change", t), n("update:current", t.page);
    };
    return (a, t) => (g(), l("ag-pagination", u({
      ref_key: "paginationRef",
      ref: o,
      ".bordered": e.bordered,
      ".firstLastNavigation": e.firstLastNavigation,
      ".navigationLabels": e.navigationLabels,
      current: e.current,
      "total-pages": e.totalPages,
      offset: e.offset,
      justify: e.justify,
      "aria-label": e.ariaLabel,
      onPageChange: r
    }, a.$attrs), [
      d(a.$slots, "default")
    ], 48, c));
  }
});
export {
  L as _
};
