// agnosticui-core@2.0.0-alpha.12/dist/VueRating.vue_vue_type_script_setup_true_lang-CICncp71.js
import { defineComponent as r, ref as u, createElementBlock as s, openBlock as f, mergeProps as m, renderSlot as c } from "vue";
import "./components/Rating/core/Rating.js";
const v = [".value", ".readonly", ".allowClear", ".required", ".invalid", "max", "precision", "variant", "size", "name", "label", "labelPosition", "labelHidden", "noLabel", "errorMessage", "helpText"], h = /* @__PURE__ */ r({
  __name: "VueRating",
  props: {
    value: { default: 0 },
    max: { default: 5 },
    precision: { default: "whole" },
    readonly: { type: Boolean, default: !1 },
    allowClear: { type: Boolean, default: !1 },
    variant: { default: "" },
    size: { default: "md" },
    name: { default: "" },
    label: { default: "" },
    labelPosition: { default: "top" },
    labelHidden: { type: Boolean, default: !1 },
    noLabel: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    invalid: { type: Boolean, default: !1 },
    errorMessage: { default: "" },
    helpText: { default: "" }
  },
  emits: ["update:value", "change", "hover"],
  setup(e, { emit: n }) {
    const t = n, o = u(), i = (a) => {
      const l = a;
      t("change", l.detail), t("update:value", l.detail.newValue);
    }, d = (a) => {
      t("hover", a.detail);
    };
    return (a, l) => (f(), s("ag-rating", m({
      ref_key: "ratingRef",
      ref: o,
      ".value": e.value,
      ".readonly": e.readonly,
      ".allowClear": e.allowClear,
      ".required": e.required,
      ".invalid": e.invalid,
      max: e.max,
      precision: e.precision,
      variant: e.variant,
      size: e.size,
      name: e.name,
      label: e.label,
      labelPosition: e.labelPosition,
      labelHidden: e.labelHidden,
      noLabel: e.noLabel,
      errorMessage: e.errorMessage,
      helpText: e.helpText,
      onRatingChange: i,
      onRatingHover: d
    }, a.$attrs), [
      c(a.$slots, "default")
    ], 48, v));
  }
});
export {
  h as _
};
