// agnosticui-core@2.0.0-alpha.12/dist/VueRadio.vue_vue_type_script_setup_true_lang-C7Z-sWpa.js
import { defineComponent as o, createElementBlock as c, openBlock as u, mergeProps as s, renderSlot as f } from "vue";
import "./components/Radio/core/Radio.js";
const b = [".checked", ".disabled", ".required", ".invalid", "name", "value", "size", "theme", "labelText", "labelPosition", "label", "labelHidden", "noLabel", "errorMessage", "helpText"], m = /* @__PURE__ */ o({
  __name: "VueRadio",
  props: {
    name: { default: "" },
    value: { default: "" },
    checked: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    size: { default: "medium" },
    theme: { default: "default" },
    labelText: { default: "" },
    labelPosition: { default: "end" },
    label: { default: "" },
    labelHidden: { type: Boolean, default: !1 },
    noLabel: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    invalid: { type: Boolean, default: !1 },
    errorMessage: { default: "" },
    helpText: { default: "" }
  },
  emits: ["click", "change", "update:checked"],
  setup(e, { emit: d }) {
    const a = d, n = (l) => {
      a("click", l);
    }, i = (l) => {
      const t = l.detail;
      a("change", t), a("update:checked", t.checked);
    };
    return (l, t) => (u(), c("ag-radio", s({
      ".checked": e.checked,
      ".disabled": e.disabled,
      ".required": e.required,
      ".invalid": e.invalid,
      name: e.name,
      value: e.value,
      size: e.size,
      theme: e.theme,
      labelText: e.labelText,
      labelPosition: e.labelPosition,
      label: e.label,
      labelHidden: e.labelHidden,
      noLabel: e.noLabel,
      errorMessage: e.errorMessage,
      helpText: e.helpText,
      onClick: n,
      onChange: i
    }, l.$attrs), [
      f(l.$slots, "default")
    ], 48, b));
  }
});
export {
  m as _
};
