// agnosticui-core@2.0.0-alpha.12/dist/VueSelectionButtonGroup.vue_vue_type_script_setup_true_lang-CLuww2Ac.js
import { defineComponent as u, createElementBlock as s, openBlock as i, mergeProps as o, renderSlot as c } from "vue";
import "./components/SelectionButtonGroup/core/SelectionButtonGroup.js";
const m = ["type", "name", "legend", "legend-hidden", "theme", "size", "shape", "value", "values", ".disabled"], g = /* @__PURE__ */ u({
  __name: "VueSelectionButtonGroup",
  props: {
    type: { default: "radio" },
    name: { default: "" },
    legend: { default: "" },
    legendHidden: { type: Boolean, default: !1 },
    theme: { default: "" },
    size: { default: "md" },
    shape: { default: "" },
    value: { default: "" },
    values: { default: () => [] },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["selection-change", "update:value", "update:values"],
  setup(e, { emit: d }) {
    const l = d, n = (a) => {
      const t = a.detail;
      l("selection-change", t), t.selectedValues.length <= 1 && l("update:value", t.selectedValues[0] || ""), l("update:values", t.selectedValues);
    };
    return (a, t) => (i(), s("ag-selection-button-group", o({
      type: e.type,
      name: e.name,
      legend: e.legend,
      "legend-hidden": e.legendHidden,
      theme: e.theme,
      size: e.size,
      shape: e.shape,
      value: e.value,
      values: e.values,
      ".disabled": e.disabled,
      onSelectionChange: n
    }, a.$attrs), [
      c(a.$slots, "default")
    ], 48, m));
  }
});
export {
  g as _
};
