// agnosticui-core@2.0.0-alpha.12/dist/VueSlider.vue_vue_type_script_setup_true_lang-DrAG9Z0N.js
import { defineComponent as f, ref as c, createElementBlock as r, openBlock as m, mergeProps as b } from "vue";
import "./components/Slider/core/Slider.js";
const h = [".value", ".dual", ".vertical", ".filled", ".monochrome", ".labelHidden", ".labelPosition", ".noLabel", ".disabled", ".readonly", ".required", ".invalid", ".showTooltip", ".showTicks", "min", "max", "step", "size", "label", "ariaLabel", "errorMessage", "helpText", "name", "tickStep"], B = /* @__PURE__ */ f({
  __name: "VueSlider",
  props: {
    label: { default: "" },
    labelPosition: { default: "top" },
    labelHidden: { type: Boolean, default: !1 },
    noLabel: { type: Boolean, default: !1 },
    ariaLabel: { default: "" },
    min: { default: 0 },
    max: { default: 100 },
    step: { default: 1 },
    value: { default: 0 },
    dual: { type: Boolean, default: !1 },
    vertical: { type: Boolean, default: !1 },
    size: { default: "default" },
    filled: { type: Boolean, default: !1 },
    monochrome: { type: Boolean, default: !1 },
    disabled: { type: Boolean, default: !1 },
    readonly: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    invalid: { type: Boolean, default: !1 },
    errorMessage: { default: "" },
    helpText: { default: "" },
    name: { default: "" },
    showTooltip: { type: Boolean, default: !1 },
    showTicks: { type: Boolean, default: !1 },
    tickStep: { default: 25 }
  },
  emits: ["input", "change", "focus", "blur", "update:value"],
  setup(e, { emit: n }) {
    const a = n, o = c(), i = (l) => {
      const t = l;
      a("input", t.detail), a("update:value", t.detail.value);
    }, d = (l) => {
      a("change", l.detail);
    }, u = (l) => {
      a("focus", l);
    }, s = (l) => {
      a("blur", l);
    };
    return (l, t) => (m(), r("ag-slider", b({
      ref_key: "sliderRef",
      ref: o,
      ".value": e.value,
      ".dual": e.dual,
      ".vertical": e.vertical,
      ".filled": e.filled,
      ".monochrome": e.monochrome,
      ".labelHidden": e.labelHidden,
      ".labelPosition": e.labelPosition,
      ".noLabel": e.noLabel,
      ".disabled": e.disabled,
      ".readonly": e.readonly,
      ".required": e.required,
      ".invalid": e.invalid,
      ".showTooltip": e.showTooltip,
      ".showTicks": e.showTicks,
      min: e.min,
      max: e.max,
      step: e.step,
      size: e.size,
      label: e.label,
      ariaLabel: e.ariaLabel,
      errorMessage: e.errorMessage,
      helpText: e.helpText,
      name: e.name,
      tickStep: e.tickStep,
      onInput: i,
      onChange: d,
      onFocus: u,
      onBlur: s
    }, l.$attrs), null, 48, h));
  }
});
export {
  B as _
};
