// agnosticui-core@2.0.0-alpha.12/dist/VueSelectionCardGroup.vue_vue_type_script_setup_true_lang-B2uA4jdK.js
import { defineComponent as u, createElementBlock as s, openBlock as o, mergeProps as i, renderSlot as c } from "vue";
import "./components/SelectionCardGroup/core/SelectionCardGroup.js";
const g = ["type", "name", "legend", "legend-hidden", "value", "values", ".disabled"], r = /* @__PURE__ */ u({
  __name: "VueSelectionCardGroup",
  props: {
    type: { default: "radio" },
    name: { default: "" },
    legend: { default: "" },
    legendHidden: { type: Boolean, default: !1 },
    value: { default: "" },
    values: { default: () => [] },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["selection-change", "update:value", "update:values"],
  setup(e, { emit: d }) {
    const t = d, n = (a) => {
      const l = a.detail;
      t("selection-change", l), l.selectedValues.length <= 1 && t("update:value", l.selectedValues[0] || ""), t("update:values", l.selectedValues);
    };
    return (a, l) => (o(), s("ag-selection-card-group", i({
      type: e.type,
      name: e.name,
      legend: e.legend,
      "legend-hidden": e.legendHidden,
      value: e.value,
      values: e.values,
      ".disabled": e.disabled,
      onSelectionChange: n
    }, a.$attrs), [
      c(a.$slots, "default")
    ], 48, g));
  }
});
export {
  r as _
};
