// agnosticui-core@2.0.0-alpha.12/dist/VueToast.vue_vue_type_script_setup_true_lang-ZC6HDIL9.js
import { defineComponent as r, ref as l, createElementBlock as i, openBlock as f, mergeProps as p, renderSlot as m } from "vue";
import "./components/Toast/core/Toast.js";
const c = [".open", ".autoDismiss", ".showCloseButton", ".pauseOnHover", ".bordered", ".rounded", ".borderedLeft", "type", "position", "duration"], h = /* @__PURE__ */ r({
  __name: "VueToast",
  props: {
    open: { type: Boolean, default: !1 },
    type: { default: "default" },
    position: { default: "top-end" },
    duration: { default: 5e3 },
    autoDismiss: { type: Boolean, default: !0 },
    showCloseButton: { type: Boolean, default: !0 },
    pauseOnHover: { type: Boolean, default: !0 },
    bordered: { type: Boolean, default: !1 },
    rounded: { type: Boolean, default: !0 },
    borderedLeft: { type: Boolean, default: !1 },
    onToastOpen: {},
    onToastClose: {},
    onToastDismiss: {}
  },
  emits: ["toast-open", "toast-close", "toast-dismiss", "update:open"],
  setup(e, { emit: s }) {
    const t = s, a = l(), n = (o) => {
      t("toast-open", o), t("update:open", !0);
    }, d = (o) => {
      t("toast-close", o), t("update:open", !1);
    }, u = (o) => {
      t("toast-dismiss", o), t("update:open", !1);
    };
    return (o, B) => (f(), i("ag-toast", p({
      ref_key: "toastRef",
      ref: a,
      ".open": e.open,
      ".autoDismiss": e.autoDismiss,
      ".showCloseButton": e.showCloseButton,
      ".pauseOnHover": e.pauseOnHover,
      ".bordered": e.bordered,
      ".rounded": e.rounded,
      ".borderedLeft": e.borderedLeft,
      type: e.type,
      position: e.position,
      duration: e.duration,
      onToastOpen: n,
      onToastClose: d,
      onToastDismiss: u
    }, o.$attrs), [
      m(o.$slots, "default")
    ], 48, c));
  }
});
export {
  h as _
};
