// agnosticui-core@2.0.0-alpha.12/dist/VueTooltip.vue_vue_type_script_setup_true_lang-BGhqFxll.js
import { defineComponent as c, ref as r, createElementBlock as h, openBlock as g, mergeProps as f, renderSlot as m } from "vue";
import "./components/Tooltip/core/Tooltip.js";
const u = ["content", "placement", "distance", "skidding", "trigger", ".disabled"], b = /* @__PURE__ */ c({
  __name: "VueTooltip",
  props: {
    content: { default: "" },
    placement: { default: "top" },
    distance: { default: 8 },
    skidding: { default: 0 },
    trigger: { default: "hover focus" },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["show", "hide"],
  setup(e, { expose: i, emit: s }) {
    const n = s, o = r(), l = (t) => {
      n("show", t.detail);
    }, a = (t) => {
      n("hide", t.detail);
    };
    return i({
      show: () => {
        o.value?.show();
      },
      hide: () => {
        o.value?.hide();
      },
      toggle: () => {
        o.value?.toggle();
      }
    }), (t, d) => (g(), h("ag-tooltip", f({
      ref_key: "tooltipRef",
      ref: o,
      content: e.content,
      placement: e.placement,
      distance: e.distance,
      skidding: e.skidding,
      trigger: e.trigger,
      ".disabled": e.disabled,
      onShow: l,
      onHide: a
    }, t.$attrs), [
      m(t.$slots, "default")
    ], 48, u));
  }
});
export {
  b as _
};
