// agnosticui-core@2.0.0-alpha.12/dist/VueToggle.vue_vue_type_script_setup_true_lang-BxrvAsse.js
import { defineComponent as r, ref as u, watch as f, createElementBlock as s, openBlock as g, mergeProps as h } from "vue";
import "./components/Toggle/core/Toggle.js";
const b = ["label", ".labelPosition", ".labelHidden", ".noLabel", "size", "variant", ".checked", ".disabled", ".readonly", ".required", ".invalid", "errorMessage", "helpText", "name", "value"], v = /* @__PURE__ */ r({
  __name: "VueToggle",
  props: {
    label: {},
    labelPosition: { default: "top" },
    labelHidden: { type: Boolean, default: !1 },
    noLabel: { type: Boolean, default: !1 },
    checked: { type: Boolean, default: !1 },
    size: { default: "md" },
    variant: { default: "default" },
    disabled: { type: Boolean, default: !1 },
    readonly: { type: Boolean, default: !1 },
    required: { type: Boolean, default: !1 },
    invalid: { type: Boolean, default: !1 },
    errorMessage: { default: "" },
    helpText: { default: "" },
    name: {},
    value: {}
  },
  emits: ["click", "toggle-change", "update:checked"],
  setup(e, { emit: d }) {
    const o = e, t = d, a = u(), i = (l) => {
      t("click", l);
    }, c = (l) => {
      const n = l;
      t("toggle-change", n.detail), t("update:checked", n.detail.checked);
    };
    return f(
      () => o.checked,
      (l) => {
        a.value && a.value.checked !== l && (a.value.checked = l);
      }
    ), (l, n) => (g(), s("ag-toggle", h({
      ref_key: "toggleRef",
      ref: a,
      label: e.label,
      ".labelPosition": e.labelPosition,
      ".labelHidden": e.labelHidden,
      ".noLabel": e.noLabel,
      size: e.size,
      variant: e.variant,
      ".checked": e.checked,
      ".disabled": e.disabled,
      ".readonly": e.readonly,
      ".required": e.required,
      ".invalid": e.invalid,
      errorMessage: e.errorMessage,
      helpText: e.helpText,
      name: e.name,
      value: e.value,
      onClick: i,
      onToggleChange: c
    }, l.$attrs), null, 48, b));
  }
});
export {
  v as _
};
