// agnosticui-core@2.0.0-alpha.12/dist/components/Accordion/core/_Accordion.js
import { LitElement as s, css as d, html as t } from "lit";
import { generateUniqueId as l } from "../../../utils/unique-id.js";
const r = class r extends s {
  render() {
    return t`
      <slot></slot>
    `;
  }
};
r.styles = [d`
    :host {
      display: block;
    }
    /* Progressive enhancement: Hide only after web component is defined */
    :host(:not(:defined)) {
      visibility: visible; /* Content visible without JS */
    }
    :host(:defined) {
      visibility: visible; /* Explicit visibility once defined */
    }
  `];
let a = r;
const o = class o extends s {
  constructor() {
    super(), this._id = l("accordion-item"), this.open = !1, this.headingLevel = 3, this.disabled = !1, this.bordered = !1, this.background = !1, this.useChevron = !0, this.useX = !1, this.useMinus = !1, this.noIndicator = !1, this.addEventListener("keydown", this._handleKeydown.bind(this));
  }
  connectedCallback() {
    super.connectedCallback(), this.setAttribute("data-enhanced", "");
  }
  render() {
    return t`
      <div class="header" part="ag-accordion-header-wrapper" @click="${this.toggle}">
        ${this._renderHeading()}
      </div>
      <div
        class="content"
        part="ag-accordion-content"
        role="region"
        aria-labelledby="${this._id}-button"
        id="${this._id}-panel"
        ?hidden="${this.hasAttribute("data-enhanced") && !this.open}"
      >
        <slot name="content"></slot>
      </div>
    `;
  }
  _renderChevronIndicator() {
    return t`
      <svg
        width="20"
        height="20"
        viewBox="0 0 24 24"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        aria-hidden="true"
      >
        <path d="m6 9 6 6 6-6"></path>
      </svg>
    `;
  }
  _renderPlusIndicator() {
    return t`
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="20"
        height="20"
        viewBox="0 0 24 24"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        aria-hidden="true"
      >
        <path d="M12 5v14"></path>
        <path d="M5 12h14"></path>
      </svg>
    `;
  }
  _renderMinusIndicator() {
    return t`
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="20"
        height="20"
        viewBox="0 0 24 24"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        aria-hidden="true"
      >
        <path d="M5 12h14"></path>
      </svg>
    `;
  }
  _renderIndicator() {
    return this.noIndicator ? null : this.useX ? this._renderPlusIndicator() : this.useMinus ? this.open ? this._renderMinusIndicator() : this._renderPlusIndicator() : this._renderChevronIndicator();
  }
  _renderHeading() {
    const e = t`
      <button
        part="ag-accordion-header"
        aria-expanded="${this.open}"
        aria-controls="${this._id}-panel"
        id="${this._id}-button"
        ?disabled="${this.disabled}"
        aria-disabled="${this.disabled}"
        @keydown="${this._handleKeydown}"
      >
        <slot name="header"></slot>
        <span class="indicator" part="ag-accordion-indicator">
          <slot name="indicator">
            ${this._renderIndicator()}
          </slot>
        </span>
      </button>
    `;
    switch (Math.max(1, Math.min(6, this.headingLevel || 3))) {
      case 1:
        return t`<h1 class="heading" part="ag-accordion-heading">${e}</h1>`;
      case 2:
        return t`<h2 class="heading" part="ag-accordion-heading">${e}</h2>`;
      case 3:
        return t`<h3 class="heading" part="ag-accordion-heading">${e}</h3>`;
      case 4:
        return t`<h4 class="heading" part="ag-accordion-heading">${e}</h4>`;
      case 5:
        return t`<h5 class="heading" part="ag-accordion-heading">${e}</h5>`;
      case 6:
        return t`<h6 class="heading" part="ag-accordion-heading">${e}</h6>`;
    }
  }
  toggle() {
    if (this.disabled) return;
    this.open = !this.open;
    const e = new CustomEvent("toggle", {
      detail: { open: this.open },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(e), this.onToggle && this.onToggle(e);
  }
  /**
   * Focus the accordion's button (for programmatic focus management)
   */
  focus() {
    this.shadowRoot?.querySelector("button")?.focus();
  }
  /**
   * Get the button element (for external focus management)
   */
  get buttonElement() {
    return this.shadowRoot?.querySelector("button") || null;
  }
  /**
   * Handle keyboard interactions for individual accordion items
   */
  _handleKeydown(e) {
    if (this.disabled) return;
    const { key: n } = e;
    (n === "Enter" || n === " ") && e.target === this.buttonElement && (e.preventDefault(), this.toggle()), (n === "ArrowDown" || n === "ArrowUp" || n === "Home" || n === "End") && (e.preventDefault(), this.dispatchEvent(new CustomEvent("accordion-keydown", {
      detail: { key: n, accordionItem: this },
      bubbles: !0
    })));
  }
};
o.properties = {
  open: { type: Boolean, reflect: !0 },
  headingLevel: { type: Number, reflect: !0, attribute: "heading-level" },
  disabled: { type: Boolean, reflect: !0 },
  bordered: { type: Boolean, reflect: !0 },
  background: { type: Boolean, reflect: !0 },
  useChevron: { type: Boolean, reflect: !0, attribute: "use-chevron" },
  useX: { type: Boolean, reflect: !0, attribute: "use-x" },
  useMinus: { type: Boolean, reflect: !0, attribute: "use-minus" },
  noIndicator: { type: Boolean, reflect: !0, attribute: "no-indicator" },
  onToggle: { attribute: !1 }
}, o.styles = [d`
    :host {
      display: block;
    }
    .header {
      cursor: pointer;
    }
    :host([disabled]) .header {
      cursor: not-allowed;
      opacity: 0.6;
    }
    :host([background]) .header {
      background-color: var(--ag-background-secondary);
      color: var(--ag-text-primary);
    }
    .heading {
      margin: 0;
      padding: 0;
    }

    :host([background]) .heading {
      margin-inline: var(--ag-space-3);
    }
      
    :host([bordered]) .header button {
      border-bottom: 1px solid var(--ag-border);
    }

    .header button {
      background: none;
      border: none;
      padding-block: var(--ag-space-4);
      margin-block-end: var(--ag-space-2);
      font: inherit;
      /* Ensure button inherits color as we've seen in dark mode issues with useragent keeping it color: buttontext black */
      color: inherit;
      cursor: pointer;
      width: 100%;
      text-align: start;
      display: flex;
      align-items: center;
      justify-content: space-between;
    }
    .header button:focus-visible {
      /* Softer focus ring using alpha channel */
      outline: var(--ag-focus-width) solid rgba(var(--ag-focus), 0.5);
      outline-offset: var(--ag-focus-offset);
      transition: outline var(--ag-motion-medium) ease;
    }
    :host([disabled]) .header button {
      cursor: not-allowed;
      pointer-events: none;
    }

    /* Indicator wrapper - visible by default unless noIndicator is set */
    .indicator {
      display: block;
      flex-shrink: 0;
      transition: transform var(--ag-motion-slow) ease;
    }
    :host([no-indicator]) .indicator {
      display: none;
    }

    /* Chevron indicator (default): starts pointing down, rotates 180deg to point up when open */
    :host([use-chevron][open]) .indicator {
      transform: rotate(180deg);
    }

    /* X indicator: starts rotated 180deg (upside-down plus), rotates to 45deg (X) when open */
    :host([use-x]) .indicator {
      transform: rotate(180deg);
    }
    :host([use-x][open]) .indicator {
      transform: rotate(45deg);
    }

    /* Minus indicator: Plus swaps to minus icon when open - no rotation needed */
    :host([use-minus]) .indicator {
      /* No rotation - the icon swap from plus to minus provides the visual feedback */
      transform: none;
    }

    /* Respect prefers-reduced-motion */
    @media (prefers-reduced-motion: reduce) {
      .header button:focus-visible,
      .indicator {
        transition: none;
      }
    }

    /* Progressive enhancement: Content visible until JS enhanced */
    .content {
      display: block;
    }
    /* After JS enhancement, respect open/closed state */
    :host([data-enhanced]) .content {
      display: none;
    }
    :host([data-enhanced][open]) .content {
      display: block;
      margin-block-end: var(--ag-space-2);
    }
    /* Ensure hidden attribute works in all environments */
    .content[hidden] {
      display: none !important;
    }
  `];
let i = o;
customElements.get("ag-accordion-item") || customElements.define("ag-accordion-item", i);
export {
  a as Accordion,
  i as AccordionItem
};
