// agnosticui-core@2.0.0-alpha.12/dist/components/Accordion/core/AccordionGroup.js
import { LitElement as a, css as l, html as h } from "lit";
const c = class c extends a {
  constructor() {
    super(), this.handleAccordionKeydown = (n) => {
      const r = n, e = this.accordions;
      if (e.length === 0) return;
      const { key: i, accordionItem: d } = r.detail, o = e.indexOf(d);
      if (o === -1) return;
      let t = o;
      switch (i) {
        case "ArrowDown":
          t = (o + 1) % e.length;
          break;
        case "ArrowUp":
          t = (o - 1 + e.length) % e.length;
          break;
        case "Home":
          t = 0;
          break;
        case "End":
          t = e.length - 1;
          break;
        default:
          return;
      }
      t !== o && e[t].focus();
    }, this.addEventListener("accordion-keydown", this.handleAccordionKeydown);
  }
  get accordions() {
    return Array.from(this.children).filter((n) => "disabled" in n && "focus" in n && typeof n.focus == "function" && !n.disabled);
  }
  render() {
    return h`<slot></slot>`;
  }
};
c.styles = [l`
    :host {
      display: block;
      width: 100%;
    }
  `];
let s = c;
customElements.get("ag-accordion") || customElements.define("ag-accordion", s);
export {
  s as AccordionGroup
};
