// agnosticui-core@2.0.0-alpha.12/dist/components/Select/core/_Select.js
import { LitElement as u, css as g, html as n } from "lit";
import { n as a } from "../../../property-CemaeiRl.js";
import { e as b } from "../../../query-BykXNUlT.js";
import { isHorizontalLabel as h, createFormControlIds as f, buildAriaDescribedBy as m } from "../../../shared/form-control-utils.js";
import { formControlStyles as v } from "../../../shared/form-control-styles.js";
var y = Object.defineProperty, o = (p, e, r, i) => {
  for (var s = void 0, l = p.length - 1, d; l >= 0; l--)
    (d = p[l]) && (s = d(e, r, s) || s);
  return s && y(e, r, s), s;
};
const c = class c extends u {
  constructor() {
    super(...arguments), this.size = "", this.multiple = !1, this.disabled = !1, this.name = "", this.label = "", this.labelPosition = "top", this.labelHidden = !1, this.noLabel = !1, this.required = !1, this.invalid = !1, this.errorMessage = "", this.helpText = "";
  }
  firstUpdated() {
    this.handleSlotChange();
    const e = this.shadowRoot?.querySelector("slot");
    e && e.addEventListener("slotchange", () => this.handleSlotChange());
  }
  handleSlotChange() {
    if (!this.selectElement) return;
    const e = this.shadowRoot?.querySelector("slot");
    if (!e) return;
    e.assignedNodes().forEach((i) => {
      if (i.nodeType === Node.ELEMENT_NODE) {
        const s = i;
        (s.tagName === "OPTION" || s.tagName === "OPTGROUP") && this.selectElement.appendChild(i);
      }
    });
  }
  handleClick(e) {
    this.onClick && this.onClick(e);
  }
  handleFocus(e) {
    this.dispatchEvent(new FocusEvent("focus", {
      bubbles: !0,
      composed: !0
    })), this.onFocus && this.onFocus(e);
  }
  handleBlur(e) {
    this.dispatchEvent(new FocusEvent("blur", {
      bubbles: !0,
      composed: !0
    })), this.onBlur && this.onBlur(e);
  }
  handleChange(e) {
    const r = e.target;
    let i;
    this.multiple ? i = Array.from(r.selectedOptions).map((l) => l.value) : i = r.value;
    const s = new CustomEvent("change", {
      detail: { value: i },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(s), this.onChange && this.onChange(s);
  }
  renderLabel(e) {
    if (!this.label || this.noLabel) return "";
    const r = [];
    return h(this.labelPosition) ? (r.push("ag-form-control__label--horizontal"), r.push(`ag-form-control__label--${this.labelPosition}`)) : this.labelPosition === "bottom" && r.push(`ag-form-control__label--${this.labelPosition}`), n`
      <label
        for=${e.inputId}
        class="ag-form-control__label ${this.labelHidden ? "ag-form-control__label--hidden" : ""} ${this.required ? "ag-form-control__label--required" : ""} ${r.join(" ")}"
      >
        ${this.label}
      </label>
    `;
  }
  render() {
    const e = f("select"), r = m({
      helperId: e.helperId,
      errorId: e.errorId,
      hasHelper: !!this.helpText && !this.invalid,
      hasError: !!this.invalid && !!this.errorMessage
    }), i = h(this.labelPosition), s = n`
      <select
        id=${e.inputId}
        class="select"
        part="ag-select"
        ?multiple=${this.multiple}
        ?disabled=${this.disabled}
        ?required=${this.required}
        .name=${this.name}
        .size=${this.multipleSize ?? (this.multiple ? 4 : 1)}
        @click=${this.handleClick}
        @focus=${this.handleFocus}
        @blur=${this.handleBlur}
        @change=${this.handleChange}
        aria-disabled=${this.disabled ? "true" : "false"}
        aria-invalid=${this.invalid ? "true" : "false"}
        aria-describedby=${r || void 0}
      ></select>
      <slot></slot>
    `, l = this.helpText && !this.invalid ? n`<div id=${e.helperId} class="ag-form-control__helper">
          ${this.helpText}
        </div>` : "", d = this.invalid && this.errorMessage ? n`<div id=${e.errorId} class="ag-form-control__error">
          ${this.errorMessage}
        </div>` : "";
    return i ? n`
        <div class="ag-form-control--horizontal">
          ${this.renderLabel(e)}
          ${s}
        </div>
        ${l}
        ${d}
      ` : this.labelPosition === "bottom" ? n`
        ${s}
        ${l}
        ${d}
        ${this.renderLabel(e)}
      ` : n`
      ${this.renderLabel(e)}
      ${s}
      ${l}
      ${d}
    `;
  }
};
c.styles = [
  v,
  g`
      :host {
        display: block;
        width: 100%;
      }

      /* Hide the slot since options are moved into select */
      slot {
        display: none;
      }

      .select {
      display: block;
      width: 100%;
      /* Remove default browser styling */
      -webkit-appearance: none;
      -moz-appearance: none;
      appearance: none;

      /* Base styling */
      padding: var(--ag-space-2) var(--ag-space-8) var(--ag-space-2) var(--ag-space-3);
      -moz-padding-start: calc(var(--ag-space-3) - 3px);
      font-size: var(--ag-font-size-base);
      font-weight: 400;
      line-height: 1.5;
      font-family: inherit;
      color: var(--ag-text-primary);
      border: var(--ag-border-width-1) solid var(--ag-border);
      border-radius: var(--ag-radius-md);
      background-color: var(--ag-background-primary);
      transition:
        border-color var(--ag-transition-fast) ease-in-out,
        box-shadow var(--ag-transition-fast) ease-in-out;
    }

    /* Custom dropdown arrow for single select */
    .select:not([multiple]) {
      background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 16 16'%3e%3cpath fill='none' stroke='%23333330' stroke-linecap='round' stroke-linejoin='round' stroke-width='2' d='M2 5l6 6 6-6'/%3e%3c/svg%3e");
      background-repeat: no-repeat;
      background-position: right var(--ag-space-3) center;
      background-size: var(--ag-font-size-base) var(--ag-space-3);
    }

    /* Focus state */
    .select:focus {
      border-color: var(--ag-focus-ring-color);
      box-shadow: 0 0 0 var(--ag-focus-ring-outline-width) var(--ag-focus-ring-color);
      /* Needed for High Contrast mode */
      outline:
        var(--ag-focus-ring-outline-width) var(--ag-focus-ring-outline-style)
        var(--ag-focus-ring-outline-color);
      transition: box-shadow var(--ag-transition-fast) ease-out;
    }

    /* Disabled state */
    .select:disabled {
      background-color: var(--ag-disabled-bg);
      cursor: not-allowed;
      opacity: 0.6;
    }

    /* Firefox focusring fix */
    .select:-moz-focusring {
      color: transparent;
      text-shadow: 0 0 0 var(--ag-text-primary);
    }

    /* Small size variant */
    :host([size="small"]) .select {
      padding-block-start: var(--ag-space-1);
      padding-block-end: var(--ag-space-1);
      padding-inline-start: var(--ag-space-2);
      font-size: var(--ag-font-size-small);
    }

    /* Large size variant */
    :host([size="large"]) .select {
      padding-block-start: var(--ag-space-2);
      padding-block-end: var(--ag-space-2);
      padding-inline-start: var(--ag-space-4);
      font-size: var(--ag-font-size-large);
    }

    /* Reduced motion support */
    @media (prefers-reduced-motion), (update: slow) {
      .select,
      .select:focus {
        transition: none;
      }
    }
  `
];
let t = c;
o([
  a({ type: String, reflect: !0 })
], t.prototype, "size");
o([
  a({ type: Boolean, reflect: !0 })
], t.prototype, "multiple");
o([
  a({ type: Boolean, reflect: !0 })
], t.prototype, "disabled");
o([
  a({ type: String })
], t.prototype, "name");
o([
  a({ type: Number, attribute: "multiple-size" })
], t.prototype, "multipleSize");
o([
  a({ type: String })
], t.prototype, "label");
o([
  a({ type: String, attribute: "label-position" })
], t.prototype, "labelPosition");
o([
  a({ type: Boolean, attribute: "label-hidden" })
], t.prototype, "labelHidden");
o([
  a({ type: Boolean, attribute: "no-label" })
], t.prototype, "noLabel");
o([
  a({ type: Boolean })
], t.prototype, "required");
o([
  a({ type: Boolean })
], t.prototype, "invalid");
o([
  a({ type: String, attribute: "error-message" })
], t.prototype, "errorMessage");
o([
  a({ type: String, attribute: "help-text" })
], t.prototype, "helpText");
o([
  a({ attribute: !1 })
], t.prototype, "onClick");
o([
  a({ attribute: !1 })
], t.prototype, "onFocus");
o([
  a({ attribute: !1 })
], t.prototype, "onBlur");
o([
  a({ attribute: !1 })
], t.prototype, "onChange");
o([
  b("select")
], t.prototype, "selectElement");
customElements.get("ag-select") || customElements.define("ag-select", t);
export {
  t as Select
};
