// agnosticui-core@2.0.0-alpha.12/dist/components/Card/vue/VueCard.js
import { defineComponent as o, ref as n, onMounted as d, createElementBlock as r, openBlock as s, mergeProps as l, renderSlot as a } from "vue";
import "../core/Card.js";
import { _ as i } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const u = o({
  name: "VueCard",
  props: {
    stacked: {
      type: Boolean,
      default: !1
    },
    shadow: {
      type: Boolean,
      default: !1
    },
    animated: {
      type: Boolean,
      default: !1
    },
    rounded: {
      type: [String, Boolean],
      default: ""
    },
    variant: {
      type: String,
      default: ""
    }
  },
  setup(e) {
    const t = n(null);
    return d(async () => {
      await customElements.whenDefined("ag-card");
    }), {
      agComponent: t,
      stacked: e.stacked,
      shadow: e.shadow,
      animated: e.animated,
      rounded: e.rounded,
      variant: e.variant
    };
  }
}), f = [".stacked", ".shadow", ".animated", "rounded", "variant"];
function m(e, t, c, p, h, k) {
  return s(), r("ag-card", l({
    ref: "agComponent",
    ".stacked": e.stacked,
    ".shadow": e.shadow,
    ".animated": e.animated,
    rounded: e.rounded,
    variant: e.variant
  }, e.$attrs), [
    a(e.$slots, "header", { slot: "header" }),
    a(e.$slots, "default"),
    a(e.$slots, "footer", { slot: "footer" })
  ], 48, f);
}
const v = /* @__PURE__ */ i(u, [["render", m]]);
export {
  v as default
};
