// agnosticui-core@2.0.0-alpha.12/dist/components/Select/vue/VueSelect.js
import { defineComponent as r, ref as i, createElementBlock as s, openBlock as p, mergeProps as f, renderSlot as y } from "vue";
import "../core/Select.js";
import { _ as h } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const b = r({
  name: "VueSelect",
  props: {
    size: {
      type: String,
      default: ""
    },
    multiple: {
      type: Boolean,
      default: !1
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    name: {
      type: String,
      default: ""
    },
    multipleSize: {
      type: Number,
      default: void 0
    },
    value: {
      type: [String, Array],
      default: void 0
    },
    // External label support
    label: {
      type: String,
      default: ""
    },
    labelPosition: {
      type: String,
      default: "top"
    },
    labelHidden: {
      type: Boolean,
      default: !1
    },
    noLabel: {
      type: Boolean,
      default: !1
    },
    required: {
      type: Boolean,
      default: !1
    },
    invalid: {
      type: Boolean,
      default: !1
    },
    errorMessage: {
      type: String,
      default: ""
    },
    helpText: {
      type: String,
      default: ""
    }
  },
  emits: {
    // Custom event - emit detail payload
    change: (e) => !0,
    // v-model support
    "update:value": (e) => !0,
    // Native events - emit full event
    focus: (e) => !0,
    blur: (e) => !0,
    click: (e) => !0,
    keydown: (e) => !0,
    keyup: (e) => !0
  },
  setup(e, { emit: l }) {
    return {
      agComponent: i(null),
      handleChange: (a) => {
        l("change", a.detail), l("update:value", a.detail.value);
      },
      handleFocus: (a) => {
        l("focus", a);
      },
      handleBlur: (a) => {
        l("blur", a);
      },
      handleClick: (a) => {
        l("click", a);
      },
      handleKeydown: (a) => {
        l("keydown", a);
      },
      handleKeyup: (a) => {
        l("keyup", a);
      }
    };
  }
}), g = [".size", ".multiple", ".disabled", ".name", ".multipleSize", ".label", ".labelPosition", ".labelHidden", ".noLabel", ".required", ".invalid", ".errorMessage", ".helpText"];
function m(e, l, o, d, t, u) {
  return p(), s("ag-select", f({
    ref: "agComponent",
    ".size": e.size,
    ".multiple": e.multiple,
    ".disabled": e.disabled,
    ".name": e.name,
    ".multipleSize": e.multipleSize,
    ".label": e.label,
    ".labelPosition": e.labelPosition,
    ".labelHidden": e.labelHidden,
    ".noLabel": e.noLabel,
    ".required": e.required,
    ".invalid": e.invalid,
    ".errorMessage": e.errorMessage,
    ".helpText": e.helpText,
    onChange: l[0] || (l[0] = (...n) => e.handleChange && e.handleChange(...n)),
    onFocus: l[1] || (l[1] = (...n) => e.handleFocus && e.handleFocus(...n)),
    onBlur: l[2] || (l[2] = (...n) => e.handleBlur && e.handleBlur(...n)),
    onClick: l[3] || (l[3] = (...n) => e.handleClick && e.handleClick(...n)),
    onKeydown: l[4] || (l[4] = (...n) => e.handleKeydown && e.handleKeydown(...n)),
    onKeyup: l[5] || (l[5] = (...n) => e.handleKeyup && e.handleKeyup(...n))
  }, e.$attrs), [
    y(e.$slots, "default")
  ], 48, g);
}
const K = /* @__PURE__ */ h(b, [["render", m]]);
export {
  K as default
};
