// agnosticui-core@2.0.0-alpha.12/dist/components/Tooltip/core/_Tooltip.js
import { LitElement as g, css as _, html as w } from "lit";
import { n } from "../../../property-CemaeiRl.js";
import { r as E } from "../../../state-CovhUvdr.js";
import { e as c } from "../../../query-BykXNUlT.js";
import { a as y, c as b, o as L, f as k, s as x, b as C } from "../../../floating-ui.dom-muqvZJIo.js";
var S = Object.defineProperty, s = (d, t, e, r) => {
  for (var o = void 0, h = d.length - 1, l; h >= 0; h--)
    (l = d[h]) && (o = l(t, e, o) || o);
  return o && S(t, e, o), o;
}, a;
let i = (a = class extends g {
  constructor() {
    super(), this._handleMouseEnter = () => {
      this.disabled || (clearTimeout(this._hideTimeout), this.show());
    }, this._handleMouseLeave = () => {
      this.disabled || (this._hideTimeout = window.setTimeout(() => {
        this.hide();
      }, 100));
    }, this._handleFocus = () => {
      this.disabled || this.show();
    }, this._handleBlur = () => {
      this.disabled || this.hide();
    }, this._handleClick = () => {
      this.disabled || (this._open ? this.hide() : this.show());
    }, this._handleDocumentKeyDown = (t) => {
      t.key === "Escape" && this._open && this.hide();
    }, this.show = () => {
      this._open || this.disabled || (this._open = !0);
    }, this.hide = () => {
      this._open && (this._open = !1);
    }, this.toggle = () => {
      this._open ? this.hide() : this.show();
    }, this.content = "", this.placement = "top", this.distance = 8, this.skidding = 0, this.trigger = "hover focus", this.disabled = !1, this._open = !1;
  }
  firstUpdated() {
    this._setupEventListeners(), this._setupARIA();
  }
  _setupEventListeners() {
    const t = this.firstElementChild;
    t && (this.trigger.includes("hover") && (this.addEventListener("mouseenter", this._handleMouseEnter), this.addEventListener("mouseleave", this._handleMouseLeave)), this.trigger.includes("focus") && (t.addEventListener("focus", this._handleFocus), t.addEventListener("blur", this._handleBlur)), this.trigger.includes("click") && t.addEventListener("click", this._handleClick));
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this._cleanup?.(), this._removeEventListeners(), document.removeEventListener("keydown", this._handleDocumentKeyDown);
  }
  _setupARIA() {
  }
  _removeEventListeners() {
    const t = this.firstElementChild;
    t && (this.removeEventListener("mouseenter", this._handleMouseEnter), this.removeEventListener("mouseleave", this._handleMouseLeave), t.removeEventListener("focus", this._handleFocus), t.removeEventListener("blur", this._handleBlur), t.removeEventListener("click", this._handleClick));
  }
  updated(t) {
    if (super.updated(t), t.has("_open"))
      if (this._open) {
        document.addEventListener("keydown", this._handleDocumentKeyDown), this.updateComplete.then(() => {
          this._startPositioning();
        });
        const e = new CustomEvent("show", {
          detail: { visible: !0 },
          bubbles: !0,
          composed: !0
        });
        this.dispatchEvent(e), this.onShow && this.onShow(e);
      } else {
        this._stopPositioning(), document.removeEventListener("keydown", this._handleDocumentKeyDown);
        const e = new CustomEvent("hide", {
          detail: { visible: !1 },
          bubbles: !0,
          composed: !0
        });
        this.dispatchEvent(e), this.onHide && this.onHide(e);
      }
  }
  _startPositioning() {
    const t = this.shadowRoot?.querySelector("#tooltip"), e = this.shadowRoot?.querySelector("#arrow");
    if (!t || !e) return;
    const r = this.firstElementChild;
    r && (this._updatePosition(), this._cleanup = y(r, t, () => {
      this._updatePosition();
    }));
  }
  _stopPositioning() {
    this._cleanup?.(), this._cleanup = void 0;
  }
  async _updatePosition() {
    const t = this.shadowRoot?.querySelector("#tooltip"), e = this.shadowRoot?.querySelector("#arrow");
    if (!t || !e) return;
    const r = this.firstElementChild;
    if (!r) return;
    const { x: o, y: h, placement: l, middlewareData: m } = await b(
      r,
      t,
      {
        placement: this.placement,
        middleware: [
          L({ mainAxis: this.distance, crossAxis: this.skidding }),
          k(),
          x({ padding: 8 }),
          C({ element: e })
        ]
      }
    );
    Object.assign(t.style, {
      left: `${o}px`,
      top: `${h}px`
    });
    const { x: p, y: u } = m.arrow ?? {}, f = {
      top: "bottom",
      right: "left",
      bottom: "top",
      left: "right"
    }[l.split("-")[0]], v = `-${8 / 2}px`;
    Object.assign(e.style, {
      left: p != null ? `${p}px` : "",
      top: u != null ? `${u}px` : "",
      right: "",
      bottom: "",
      [f]: v
    });
  }
  render() {
    return w`
      <slot></slot>
      <div
        id="tooltip"
        part="ag-tooltip"
        role="tooltip"
        class="tooltip"
        ?data-show=${this._open}
        @mouseenter="${() => clearTimeout(this._hideTimeout)}"
        @mouseleave="${this.hide}"
      >
        <slot name="content">${this.content}</slot>
        <div id="arrow" part="ag-tooltip-arrow"></div>
      </div>
    `;
  }
}, a.styles = _`
    :host {
      display: inline-block;
    }

    #tooltip {
      display: none;
      position: absolute;
      z-index: var(--ag-z-index-modal);
      pointer-events: none;
      opacity: 0;
      transition: opacity var(--ag-motion-fast) ease-in-out;
      background: var(--ag-neutral-900);
      color: var(--ag-white);
      padding: var(--ag-space-2) var(--ag-space-3);
      border-radius: var(--ag-radius-md);
      font-size: var(--ag-font-size-sm);
      max-inline-size: 300px;
    }

    #tooltip[data-show] {
      display: block;
      opacity: 1;
      pointer-events: auto;
    }

    #arrow {
      position: absolute;
      background: inherit;
      width: var(--ag-space-2);
      height: var(--ag-space-2);
      transform: rotate(45deg);
    }
  `, a);
s([
  n({ type: String })
], i.prototype, "content");
s([
  n({ type: String, reflect: !0 })
], i.prototype, "placement");
s([
  n({ type: Number })
], i.prototype, "distance");
s([
  n({ type: Number })
], i.prototype, "skidding");
s([
  n({ type: String })
], i.prototype, "trigger");
s([
  n({ type: Boolean, reflect: !0 })
], i.prototype, "disabled");
s([
  n({ attribute: !1 })
], i.prototype, "onShow");
s([
  n({ attribute: !1 })
], i.prototype, "onHide");
s([
  E()
], i.prototype, "_open");
s([
  c("#tooltip")
], i.prototype, "tooltipElement");
s([
  c("#arrow")
], i.prototype, "arrowElement");
customElements.get("ag-tooltip") || customElements.define("ag-tooltip", i);
export {
  i as Tooltip
};
