// agnosticui-core@2.0.0-alpha.12/dist/components/ScrollProgress/vue/VueScrollProgress.js
import { defineComponent as o, ref as a, onMounted as i, nextTick as s, watchEffect as l, createElementBlock as d, openBlock as u, mergeProps as g, renderSlot as c } from "vue";
import "../core/ScrollProgress.js";
import { _ as f } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const m = o({
  name: "VueScrollProgress",
  props: {
    mode: { type: String, default: "bar" },
    target: { type: Object, default: null },
    orientation: { type: String, default: "top" },
    dots: { type: Number, default: 5 },
    badgeVariant: { type: String, default: "info" },
    ringSize: { type: Number, default: 32 },
    ringStrokeWidth: { type: Number, default: 3 },
    ringVariant: { type: String, default: "info" },
    precision: { type: Number, default: 0 }
  },
  setup(e) {
    const r = a(null), n = () => {
      const t = r.value;
      t && (t.mode = e.mode, t.target = e.target, t.orientation = e.orientation, t.dots = e.dots, t.badgeVariant = e.badgeVariant, t.ringSize = e.ringSize, t.ringStrokeWidth = e.ringStrokeWidth, t.ringVariant = e.ringVariant, t.precision = e.precision);
    };
    return i(async () => {
      await customElements.whenDefined("ag-scroll-progress"), await s(), n();
    }), l(() => {
      r.value && n();
    }), { agComponent: r };
  }
});
function p(e, r, n, t, S, y) {
  return u(), d("ag-scroll-progress", g({ ref: "agComponent" }, e.$attrs), [
    c(e.$slots, "default")
  ], 16);
}
const h = /* @__PURE__ */ f(m, [["render", p]]);
export {
  h as default
};
