// agnosticui-core@2.0.0-alpha.12/dist/components/ScrollToButton/core/_ScrollToButton.js
import { LitElement as p, css as d, html as s } from "lit";
import { n as o } from "../../../property-CemaeiRl.js";
import "../../Button/core/Button.js";
var g = Object.defineProperty, i = (c, t, r, u) => {
  for (var n = void 0, l = c.length - 1, h; l >= 0; l--)
    (h = c[l]) && (n = h(t, r, n) || n);
  return n && g(t, r, n), n;
};
const a = class a extends p {
  constructor() {
    super(), this._targetElement = null, this._handleScroll = () => {
      this.visible = window.scrollY > this.scrollThreshold;
    }, this.label = "Back to Top", this.showLabel = !1, this.icon = !0, this.scrollThreshold = 400, this.target = "top", this.direction = "auto", this.smoothScroll = !0, this.visible = !1, this.size = "md", this.shape = "";
  }
  _getScrollBehavior() {
    const t = window.matchMedia("(prefers-reduced-motion: reduce)").matches;
    return this.smoothScroll && !t ? "smooth" : "auto";
  }
  _getScrollTarget() {
    if (this._targetElement)
      return this._targetElement.offsetTop;
    if (typeof this.target == "string")
      switch (this.target) {
        case "top":
          return 0;
        case "bottom":
          return document.documentElement.scrollHeight - window.innerHeight;
        default: {
          const t = document.getElementById(this.target);
          if (t)
            return t.offsetTop;
          const r = document.querySelector(this.target);
          return r instanceof HTMLElement ? r.offsetTop : (console.warn(`ScrollToButton: Target "${this.target}" not found, scrolling to top`), 0);
        }
      }
    return 0;
  }
  _getDirection() {
    if (this.direction !== "auto")
      return this.direction;
    if (this._targetElement)
      return this._targetElement.offsetTop < window.scrollY ? "up" : "down";
    if (typeof this.target == "string")
      switch (this.target) {
        case "bottom":
          return "down";
        case "top":
        default:
          return "up";
      }
    return "up";
  }
  _scrollToTarget() {
    const t = this._getScrollTarget(), r = this._getScrollBehavior();
    window.scrollTo({
      top: t,
      behavior: r
    });
  }
  // Public method to set HTMLElement target
  setTargetElement(t) {
    this._targetElement = t;
  }
  connectedCallback() {
    super.connectedCallback(), window.addEventListener("scroll", this._handleScroll, { passive: !0 }), this._handleScroll();
  }
  disconnectedCallback() {
    window.removeEventListener("scroll", this._handleScroll), super.disconnectedCallback();
  }
  _renderDefaultIcon() {
    return this._getDirection() === "down" ? s`
        <svg
          xmlns="http://www.w3.org/2000/svg"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          fill="currentColor"
          aria-hidden="true"
        >
          <path d="M0 0h24v24H0V0z" fill="none" />
          <path d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z" />
        </svg>
      ` : s`
      <svg
        xmlns="http://www.w3.org/2000/svg"
        height="20"
        viewBox="0 0 24 24"
        width="20"
        fill="currentColor"
        aria-hidden="true"
      >
        <path d="M0 0h24v24H0V0z" fill="none" />
        <path d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z" />
      </svg>
    `;
  }
  render() {
    const t = this.showLabel || !this.icon, r = this.icon && !t;
    return s`
      <ag-button
        mode="primary"
        size="${this.size}"
        shape="${this.shape || (r ? "circle" : "")}"
        @click="${this._scrollToTarget}"
        aria-label="${this.label}"
        part="ag-scrollto-button"
        exportparts="ag-button"
        role="button"
      >
        <div class="button-content" part="ag-button-content">
          ${this.icon ? s`
            <span class="ag-icon-wrapper" part="ag-icon-wrapper">
              <slot name="icon">
                ${this._renderDefaultIcon()}
              </slot>
            </span>
          ` : ""}
          ${t ? s`
            <span class="label" part="ag-label">${this.label}</span>
          ` : ""}
        </div>
      </ag-button>
    `;
  }
};
a.styles = d`
    :host {
      display: inline-block;
      position: fixed;
      bottom: var(--ag-space-8, 2rem);
      inset-inline-end: var(--ag-space-8, 2rem);
      z-index: var(--ag-z-index-sticky, 900);
      opacity: 0;
      visibility: hidden;
      transition: opacity var(--ag-motion-slow, 0.4s) ease-in-out,
                  visibility var(--ag-motion-slow, 0.4s) ease-in-out;
    }

    :host([visible]) {
      opacity: 1;
      visibility: visible;
    }

    @media (prefers-reduced-motion: reduce) {
      :host {
        transition: none;
      }
    }

    @media print {
      :host {
        display: none !important;
      }
    }

    /* Ensure button is large enough for touch targets */
    ag-button {
      min-width: 44px;
      min-height: 44px;
    }

    /* Default flex layout for icon + label */
    .button-content {
      display: flex;
      align-items: center;
      gap: var(--ag-space-2, 0.5rem);
      flex-direction: row;
    }

    /* Icon wrapper for part targeting */
    .icon-wrapper {
      display: inline-flex;
      align-items: center;
      justify-content: center;
    }

    /* Label wrapper for part targeting */
    .label {
      display: inline-block;
    }
  `;
let e = a;
i([
  o({ type: String })
], e.prototype, "label");
i([
  o({ type: Boolean })
], e.prototype, "showLabel");
i([
  o({ type: Boolean })
], e.prototype, "icon");
i([
  o({ type: Number })
], e.prototype, "scrollThreshold");
i([
  o({ type: String })
], e.prototype, "target");
i([
  o({ type: String })
], e.prototype, "direction");
i([
  o({ type: Boolean })
], e.prototype, "smoothScroll");
i([
  o({ type: Boolean, reflect: !0 })
], e.prototype, "visible");
i([
  o({ type: String })
], e.prototype, "size");
i([
  o({ type: String })
], e.prototype, "shape");
export {
  e as ScrollToButton
};
