// agnosticui-core@2.0.0-alpha.12/dist/components/Mark/core/_Mark.js
import { LitElement as k, css as v, html as n } from "lit";
import { n as g } from "../../../property-CemaeiRl.js";
import { e as f } from "../../../query-BykXNUlT.js";
var x = Object.defineProperty, h = (m, t, o, d) => {
  for (var r = void 0, i = m.length - 1, a; i >= 0; i--)
    (a = m[i]) && (r = a(t, o, r) || r);
  return r && x(t, o, r), r;
};
const p = class p extends k {
  get _textContent() {
    return this._slot ? this._slot.assignedNodes({ flatten: !0 }).map((t) => t.textContent ?? "").join("") : "";
  }
  constructor() {
    super(), this.variant = "warning", this.search = void 0, this.caseSensitive = !1, this.matchAll = !1;
  }
  _onSlotChange() {
    this.requestUpdate();
  }
  _renderStatic() {
    return n`
      <mark class="mark mark-${this.variant}" part="ag-mark">
        <slot @slotchange=${this._onSlotChange}></slot>
      </mark>
    `;
  }
  _renderReactive() {
    const t = this._textContent, o = n`<slot @slotchange=${this._onSlotChange} style="display: none;"></slot>`;
    if (!t)
      return o;
    if (!this.search)
      return n`${o}${t}`;
    const d = this.search.replace(/[\\^$.*+?()[\]{}|]/g, "\\$&");
    if (!d)
      return n`${o}${t}`;
    let r = this.matchAll ? "g" : "";
    this.caseSensitive || (r += "i");
    const i = new RegExp(d, r), a = [];
    if (this.matchAll) {
      const e = [...t.matchAll(i)];
      if (!e.length) return n`${o}${t}`;
      let l = 0;
      for (const c of e)
        a.push(t.substring(l, c.index)), a.push(n`<mark class="mark mark-${this.variant}">${c[0]}</mark>`), l = (c.index ?? 0) + c[0].length;
      a.push(t.substring(l));
    } else {
      const e = t.match(i);
      if (e && e.index !== void 0) {
        const l = t.substring(0, e.index), c = n`<mark class="mark mark-${this.variant}">${e[0]}</mark>`, u = t.substring(e.index + e[0].length);
        a.push(l, c, u);
      } else
        a.push(t);
    }
    return n`${o}${a}`;
  }
  render() {
    return this.search ? this._renderReactive() : this._renderStatic();
  }
};
p.styles = v`
    :host {
      display: inline;
    }
    .mark {
      border-radius: var(--ag-radius-xs);
      background-color: var(--ag-warning-background);
      color: var(--ag-warning-text);
    }
    .mark::before,
    .mark::after {
      clip-path: inset(100%);
      clip: rect(1px, 1px, 1px, 1px);
      height: 1px;
      overflow: hidden;
      position: absolute;
      white-space: nowrap;
      width: 1px;
    }
    .mark::before {
      content: " [highlight start] ";
    }
    .mark::after {
      content: " [highlight end] ";
    }
    .mark-warning {
      background-color: var(--ag-warning-background);
      color: var(--ag-warning-text);
    }
    .mark-info {
      background-color: var(--ag-info-background);
      color: var(--ag-info-text);
    }
    .mark-success {
      background-color: var(--ag-success-background);
      color: var(--ag-success-text);
    }
    .mark-error {
      background-color: var(--ag-danger-background);
      color: var(--ag-danger-text);
    }
    .mark-primary {
      background-color: var(--ag-primary-background);
      color: var(--ag-primary-text);
    }
    .mark-secondary {
      background-color: var(--ag-background-secondary);
      color: var(--ag-text-primary);
    }
    .mark-monochrome {
      background-color: var(--ag-background-secondary-inverted);
      color: var(--ag-text-primary-inverted);
    }
  `;
let s = p;
h([
  g({ type: String })
], s.prototype, "variant");
h([
  g({ type: String })
], s.prototype, "search");
h([
  g({ type: Boolean, attribute: "case-sensitive" })
], s.prototype, "caseSensitive");
h([
  g({ type: Boolean, attribute: "match-all" })
], s.prototype, "matchAll");
h([
  f("slot")
], s.prototype, "_slot");
export {
  s as Mark
};
