// agnosticui-core@2.0.0-alpha.12/dist/components/SidebarNav/core/_SidebarNavSubmenu.js
import { LitElement as l, css as p, html as c } from "lit";
import { n as d } from "../../../property-CemaeiRl.js";
import { NAV_SUBLINK_STYLES as m } from "../../../styles/nav-sublink-styles.js";
var f = Object.defineProperty, b = (a, n, o, g) => {
  for (var t = void 0, e = a.length - 1, i; e >= 0; e--)
    (i = a[e]) && (t = i(n, o, t) || t);
  return t && f(n, o, t), t;
};
const r = class r extends l {
  constructor() {
    super(...arguments), this.open = !1;
  }
  render() {
    return c`
      <slot></slot>
    `;
  }
};
r.hostAttributes = {
  class: "ag-sidebar-nav-submenu",
  part: "ag-sidebar-nav-submenu",
  "aria-label": "Sidebar Navigation Menu"
}, r.styles = [
  m,
  p`
      :host {
        display: none;
        margin-inline-start: 0.875rem;
        border-inline-start: 1px solid var(--ag-border-subtle);
        padding-inline-start: 0.625rem;
        margin-block-start: var(--ag-space-1);
        margin-block-end: var(--ag-space-1);
      }

      :host([open]) {
        display: block;
      }
    `
];
let s = r;
b([
  d({ type: Boolean, reflect: !0 })
], s.prototype, "open");
export {
  s as AgSidebarNavSubmenu
};
