// agnosticui-core@2.0.0-alpha.12/dist/components/CopyButton/core/_CopyButton.js
import { LitElement as u, css as m, html as c, nothing as y } from "lit";
import { n } from "../../../property-CemaeiRl.js";
import { r as l } from "../../../state-CovhUvdr.js";
import "../../IconButton/core/IconButton.js";
var b = Object.defineProperty, i = (p, o, t, r) => {
  for (var s = void 0, a = p.length - 1, d; a >= 0; a--)
    (d = p[a]) && (s = d(o, t, s) || s);
  return s && b(o, t, s), s;
};
const h = class h extends u {
  constructor() {
    super(), this._isCopied = !1, this._hasError = !1, this._announcement = "", this.text = "", this.label = "Copy to clipboard", this.successLabel = "Copied!", this.errorLabel = "Copy failed", this.timeout = 1e3, this.size = "md", this.variant = "ghost";
  }
  disconnectedCallback() {
    super.disconnectedCallback(), this._timeoutId && clearTimeout(this._timeoutId);
  }
  firstUpdated() {
    this._validateSlots();
  }
  _validateSlots() {
    const o = this.querySelector('[slot="icon-copy"]'), t = this.querySelector('[slot="icon-copied"]'), r = !!o;
    if (r !== !!t)
      throw new Error(
        `AgCopyButton: Both "icon-copy" and "icon-copied" slots must be provided together. Currently provided: ${r ? "icon-copy" : "icon-copied"} only.`
      );
  }
  async _copyWithFallback(o) {
    if (navigator.clipboard && window.isSecureContext) {
      await navigator.clipboard.writeText(o);
      return;
    }
    const t = document.createElement("textarea");
    t.value = o, t.style.position = "fixed", t.style.left = "-999999px", t.style.top = "-999999px", document.body.appendChild(t), t.focus(), t.select();
    try {
      if (!document.execCommand("copy"))
        throw new Error("execCommand copy failed");
    } finally {
      document.body.removeChild(t);
    }
  }
  _handleCopy() {
    this.text && (this._timeoutId && clearTimeout(this._timeoutId), this._hasError = !1, this._copyWithFallback(this.text).then(() => {
      this._isCopied = !0, this._announcement = this.successLabel, this.dispatchEvent(new CustomEvent("copy", {
        detail: { text: this.text },
        bubbles: !0,
        composed: !0
      })), this._timeoutId = window.setTimeout(() => {
        this._isCopied = !1, this._announcement = "";
      }, this.timeout);
    }).catch((o) => {
      this._hasError = !0, this._announcement = this.errorLabel, this.dispatchEvent(new CustomEvent("copy-error", {
        detail: { error: o },
        bubbles: !0,
        composed: !0
      })), this._timeoutId = window.setTimeout(() => {
        this._hasError = !1, this._announcement = "";
      }, this.timeout);
    }));
  }
  _renderIcon() {
    return c`
      <span class="ag-icon-stack" part="icon-stack">
        <slot name="icon-error" part="icon icon-error" class="ag-icon-slot ${this._hasError ? "visible" : ""}">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="100%" height="100%" fill="currentColor">
            <path d="M0 0h24v24H0z" fill="none"/>
            <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"/>
          </svg>
        </slot>

        <slot name="icon-copied" part="icon icon-copied" class="ag-icon-slot ${this._isCopied && !this._hasError ? "visible" : ""}">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="100%" height="100%" fill="currentColor">
            <path d="M0 0h24v24H0z" fill="none"/>
            <path d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"/>
          </svg>
        </slot>

        <slot name="icon-copy" part="icon icon-copy" class="ag-icon-slot ${!this._isCopied && !this._hasError ? "visible" : ""}">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="100%" height="100%" fill="currentColor">
            <path d="M0 0h24v24H0z" fill="none"/>
            <path d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm3 4H8c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h11c1.1 0 2-.9 2-2V7c0-1.1-.9-2-2-2zm0 16H8V7h11v14z"/>
          </svg>
        </slot>
      </span>
    `;
  }
  render() {
    const o = this._hasError ? this.errorLabel : this._isCopied ? this.successLabel : this.label;
    return c`
      <ag-icon-button
        part="button"
        .size="${this.size}"
        .variant="${this.variant}"
        .label="${o}"
        @click="${this._handleCopy}"
      >
        ${this._renderIcon()}
      </ag-icon-button>
      
      ${this._announcement ? c`<span class="sr-only" role="status" aria-live="polite" aria-atomic="true">
            ${this._announcement}
          </span>` : y}
    `;
  }
};
h.styles = m`
    :host {
      display: inline-block;
    }

    .sr-only {
      position: absolute;
      width: 1px;
      height: 1px;
      padding: 0;
      margin: -1px;
      overflow: hidden;
      clip: rect(0, 0, 0, 0);
      white-space: nowrap;
      border-width: 0;
    }

    /* Stacked icon slots for smooth cross-fade between states */
    .ag-icon-stack {
      position: relative;
      /* Make this a block-level box so the parent IconButton's flex centering
         can reliably center the stacked slots. Inline alignment can interfere
         with flex layout in some browsers, causing the "pushed down" effect. */
      display: block;
      width: 1em;
      height: 1em;
    }

    /* Each slot occupies the full stack and is centered */
    .ag-icon-slot {
      position: absolute;
      inset: 0;
      /* Use grid centering to avoid baseline/line-height shifts during transitions */
      display: grid;
      place-items: center;
      opacity: 0;
      transition: opacity var(--ag-motion-fast, 150ms) ease-in-out;
      will-change: opacity;
      pointer-events: none;
    }

    .ag-icon-slot.visible {
      opacity: 1;
      pointer-events: auto;
    }

    /* Ensure slotted SVGs and fallback svgs size consistently to the IconButton's em-based icon box
       Use 1em so icons scale with the IconButton font-size and avoid layout shifts during cross-fade */
    .ag-icon-slot::slotted(svg),
    .ag-icon-slot svg {
      width: 1em;
      height: 1em;
      max-inline-size: 100%;
      max-block-size: 100%;
      display: block;
    }
  `;
let e = h;
i([
  n({ type: String })
], e.prototype, "text");
i([
  n({ type: String })
], e.prototype, "label");
i([
  n({ type: String, attribute: "success-label" })
], e.prototype, "successLabel");
i([
  n({ type: String, attribute: "error-label" })
], e.prototype, "errorLabel");
i([
  n({ type: Number })
], e.prototype, "timeout");
i([
  n({ type: String })
], e.prototype, "size");
i([
  n({ type: String })
], e.prototype, "variant");
i([
  l()
], e.prototype, "_isCopied");
i([
  l()
], e.prototype, "_hasError");
i([
  l()
], e.prototype, "_announcement");
export {
  e as AgCopyButton
};
