// agnosticui-core@2.0.0-alpha.12/dist/components/Collapsible/core/_Collapsible.js
import { LitElement as c, css as h, html as i } from "lit";
import { n as t } from "../../../property-CemaeiRl.js";
var m = Object.defineProperty, o = (u, s, n, a) => {
  for (var r = void 0, d = u.length - 1, p; d >= 0; d--)
    (p = u[d]) && (r = p(s, n, r) || r);
  return r && m(s, n, r), r;
};
const l = class l extends c {
  constructor() {
    super(), this.open = !1, this.bordered = !1, this.shadow = !1, this.useChevron = !0, this.useX = !1, this.useMinus = !1, this.noIndicator = !1;
  }
  _renderChevronIndicator() {
    return i`
      <svg
        width="20"
        height="20"
        viewBox="0 0 24 24"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        aria-hidden="true"
      >
        <path d="m6 9 6 6 6-6"></path>
      </svg>
    `;
  }
  _renderPlusIndicator() {
    return i`
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="20"
        height="20"
        viewBox="0 0 24 24"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        aria-hidden="true"
      >
        <path d="M12 5v14"></path>
        <path d="M5 12h14"></path>
      </svg>
    `;
  }
  _renderMinusIndicator() {
    return i`
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="20"
        height="20"
        viewBox="0 0 24 24"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        stroke-linecap="round"
        stroke-linejoin="round"
        aria-hidden="true"
      >
        <path d="M5 12h14"></path>
      </svg>
    `;
  }
  _renderIndicator() {
    return this.noIndicator ? null : this.useX ? this._renderPlusIndicator() : this.useMinus ? this.open ? this._renderMinusIndicator() : this._renderPlusIndicator() : this._renderChevronIndicator();
  }
  _onToggle(s) {
    const n = s.target;
    this.open = n.open;
    const a = new CustomEvent("toggle", {
      detail: { open: this.open },
      bubbles: !0,
      composed: !0
    });
    this.dispatchEvent(a), this.onToggle && this.onToggle(a);
  }
  render() {
    return i`
      <details
        part="ag-collapsible-details"
        ?open=${this.open}
        @toggle=${this._onToggle}
      >
        <summary part="ag-collapsible-summary">
          <slot name="summary"></slot>
          <span class="indicator" part="ag-collapsible-indicator">
            <slot name="indicator">
              ${this._renderIndicator()}
            </slot>
          </span>
        </summary>
        <div class="collapsible-content" part="ag-collapsible-content">
          <slot></slot>
        </div>
      </details>
    `;
  }
};
l.styles = h`
    :host {
      display: block;
    }

    /* Base details element */
    details {
      border: none;
      border-radius: var(--ag-radius-md);
    }

    :host([bordered]) details {
      border: var(--ag-border-width-1) solid var(--ag-border);
    }

    :host([shadow]) details {
      box-shadow: var(--ag-shadow-md);
    }

    summary {
      cursor: pointer;
      padding: var(--ag-space-4);
      display: flex;
      justify-content: space-between;
      align-items: center;
      list-style: none; /* Remove default marker */
      background: none;
      color: var(--ag-text-primary);
    }

    summary::-webkit-details-marker {
      display: none; /* Chrome */
    }

    summary:focus-visible {
      outline: var(--ag-focus-width) solid var(--ag-focus);
      outline-offset: var(--ag-focus-offset);
      transition: outline var(--ag-motion-medium) ease;
    }

    .collapsible-content {
      padding: var(--ag-space-4);
      margin-block-end: var(--ag-space-2);
    }

    /* Indicator wrapper - visible by default unless noIndicator is set */
    .indicator {
      display: block;
      flex-shrink: 0;
      transition: transform var(--ag-motion-slow) ease;
    }

    :host([no-indicator]) .indicator {
      display: none;
    }

    /* Chevron indicator (default): starts pointing down, rotates 180deg to point up when open */
    :host([use-chevron]) details[open] > summary .indicator {
      transform: rotate(180deg);
    }

    /* X indicator: starts rotated 180deg (upside-down plus), rotates to 45deg (X) when open */
    :host([use-x]) .indicator {
      transform: rotate(180deg);
    }
    :host([use-x]) details[open] > summary .indicator {
      transform: rotate(45deg);
    }

    /* Minus indicator: Plus swaps to minus icon when open - no rotation needed */
    :host([use-minus]) .indicator {
      /* No rotation - the icon swap from plus to minus provides the visual feedback */
      transform: none;
    }

    /* Respect prefers-reduced-motion */
    @media (prefers-reduced-motion: reduce) {
      summary:focus-visible,
      .indicator {
        transition: none;
      }
    }
  `;
let e = l;
o([
  t({ type: Boolean, reflect: !0 })
], e.prototype, "open");
o([
  t({ type: Boolean, reflect: !0 })
], e.prototype, "bordered");
o([
  t({ type: Boolean, reflect: !0 })
], e.prototype, "shadow");
o([
  t({ type: Boolean, reflect: !0, attribute: "use-chevron" })
], e.prototype, "useChevron");
o([
  t({ type: Boolean, reflect: !0, attribute: "use-x" })
], e.prototype, "useX");
o([
  t({ type: Boolean, reflect: !0, attribute: "use-minus" })
], e.prototype, "useMinus");
o([
  t({ type: Boolean, reflect: !0, attribute: "no-indicator" })
], e.prototype, "noIndicator");
o([
  t({ attribute: !1 })
], e.prototype, "onToggle");
export {
  e as AgCollapsible
};
