// agnosticui-core@2.0.0-alpha.12/dist/components/Breadcrumb/core/_Breadcrumb.js
import { LitElement as u, css as m, html as i } from "lit";
import { n as s } from "../../../property-CemaeiRl.js";
var p = Object.defineProperty, c = (b, r, a, l) => {
  for (var e = void 0, n = b.length - 1, d; n >= 0; n--)
    (d = b[n]) && (e = d(r, a, e) || e);
  return e && p(r, a, e), e;
};
const o = class o extends u {
  constructor() {
    super(), this._handleItemClick = (r, a, l) => {
      const e = new CustomEvent("breadcrumb-click", {
        detail: {
          item: a,
          index: l,
          event: r
        },
        bubbles: !0,
        composed: !0
      });
      this.dispatchEvent(e), this.onBreadcrumbClick && this.onBreadcrumbClick(e);
    }, this.items = [], this.type = "default", this.primary = !1, this.ariaLabel = "Breadcrumb";
  }
  _renderBreadcrumbItem(r, a) {
    const l = a === this.items.length - 1, e = r.current || l;
    return i`
      <li class="ag-breadcrumb__item" part="ag-breadcrumb-item">
        ${r.href && !e ? i`
              <a
                href="${r.href}"
                class="ag-breadcrumb__link"
                part="ag-breadcrumb-item-link"
                @click="${(n) => this._handleItemClick(n, r, a)}"
              >
                ${r.label}
              </a>
            ` : e ? i`
                <span class="ag-breadcrumb__text" aria-current="page" part="ag-breadcrumb-item-current">
                  ${r.label}
                </span>
              ` : i`
                <span class="ag-breadcrumb__text" part="ag-breadcrumb-item-text">
                  ${r.label}
                </span>
              `}
      </li>
    `;
  }
  render() {
    return !this.items || this.items.length === 0 ? i`` : i`
      <nav class="ag-breadcrumb" part="ag-breadrumb" aria-label="${this.ariaLabel}">
        <ol class="ag-breadcrumb__list">
          ${this.items.map((r, a) => this._renderBreadcrumbItem(r, a))}
        </ol>
      </nav>
    `;
  }
};
o.styles = m`
    /* MINIMALIST & THEMEABLE - Styling via --ag-* design tokens */
    :host {
      display: block;
    }

    .ag-breadcrumb__list {
      display: flex;
      flex-wrap: wrap;
      padding: 0;
      margin: 0;
      list-style: none;
      white-space: nowrap;
    }

    .ag-breadcrumb__item {
      display: inline-flex;
      align-items: center;
    }

    .ag-breadcrumb__item + .ag-breadcrumb__item {
      padding-inline-start: var(--ag-space-2);
    }

    .ag-breadcrumb__item + .ag-breadcrumb__item::before {
      content: "›";
      padding-inline-end: var(--ag-space-2);
      color: var(--ag-text-secondary);
      speak: none;
      user-select: none;
    }

    /* Separator variants */
    :host([type="slash"]) .ag-breadcrumb__item + .ag-breadcrumb__item::before {
      content: "/";
    }

    :host([type="arrow"]) .ag-breadcrumb__item + .ag-breadcrumb__item::before {
      content: "→";
    }

    :host([type="bullet"]) .ag-breadcrumb__item + .ag-breadcrumb__item::before {
      content: "•";
    }

    /* Default link styling - uses text color */
    .ag-breadcrumb__link {
      text-decoration: none;
      color: var(--ag-text-primary);
      padding: var(--ag-space-1);
      margin: calc(var(--ag-space-1) * -1);
      border-radius: var(--ag-border-radius);
      transition: all var(--ag-transition-duration) ease;
      cursor: pointer;
    }

    .ag-breadcrumb__link:hover {
      text-decoration: underline;
    }

    /* Primary variant - blue link styling (opt-in) */
    :host([primary]) .ag-breadcrumb__link {
      color: var(--ag-primary-text);
    }

    :host([primary]) .ag-breadcrumb__link:hover {
      color: var(--ag-primary-text);
      text-decoration: underline;
    }

    .ag-breadcrumb__link:focus-visible {
      outline: var(--ag-focus-width) solid var(--ag-focus);
      outline-offset: var(--ag-focus-offset);
    }

    /* Current page styling - de-emphasized */
    .ag-breadcrumb__link[aria-current="page"] {
      color: var(--ag-text-secondary);
      text-decoration: none;
      cursor: default;
    }

    .ag-breadcrumb__link[aria-current="page"]:hover {
      text-decoration: none;
    }

    /* Non-linked current page */
    .ag-breadcrumb__text {
      color: var(--ag-text-primary);
      padding: var(--ag-space-1);
      margin: calc(var(--ag-space-1) * -1);
    }
  `;
let t = o;
c([
  s({ type: Array })
], t.prototype, "items");
c([
  s({ type: String, reflect: !0 })
], t.prototype, "type");
c([
  s({ type: Boolean, reflect: !0 })
], t.prototype, "primary");
c([
  s({ type: String, reflect: !0, attribute: "aria-label" })
], t.prototype, "ariaLabel");
c([
  s({ attribute: !1 })
], t.prototype, "onBreadcrumbClick");
export {
  t as AgBreadcrumb
};
