// agnosticui-core@2.0.0-alpha.12/dist/components/Radio/core/_Radio.js
import { LitElement as u, css as b, nothing as l, html as c } from "lit";
import { n as i } from "../../../property-CemaeiRl.js";
import { formControlStyles as v } from "../../../shared/form-control-styles.js";
import { createFormControlIds as f, buildAriaDescribedBy as w } from "../../../shared/form-control-utils.js";
var m = Object.defineProperty, o = (g, r, d, s) => {
  for (var e = void 0, t = g.length - 1, h; t >= 0; t--)
    (h = g[t]) && (e = h(r, d, e) || e);
  return e && m(r, d, e), e;
}, n;
const a = (n = class extends u {
  constructor() {
    super(...arguments), this.name = "", this.value = "", this.checked = !1, this.disabled = !1, this.size = "medium", this.theme = "primary", this.labelText = "", this.labelPosition = "end", this.label = "", this.labelHidden = !1, this.noLabel = !1, this.required = !1, this.invalid = !1, this.errorMessage = "", this.helpText = "", this._ids = f("ag-radio");
  }
  handleClick(r) {
    this.onClick && this.onClick(r);
  }
  handleKeyDown(r) {
    if (!["ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight", "Home", "End"].includes(r.key))
      return;
    r.preventDefault();
    const d = this.getRadiosInGroup();
    if (d.length === 0) return;
    const s = d.indexOf(this);
    if (s === -1) return;
    let e;
    r.key === "Home" ? e = 0 : r.key === "End" ? e = d.length - 1 : r.key === "ArrowDown" || r.key === "ArrowRight" ? e = (s + 1) % d.length : e = (s - 1 + d.length) % d.length;
    const t = d[e];
    if (t && !t.disabled) {
      const h = t.shadowRoot?.querySelector(".radio-input");
      if (h) {
        h.focus(), t.checked = !0, t.uncheckOtherRadiosInGroup();
        const p = new CustomEvent(
          "change",
          {
            detail: {
              checked: !0,
              value: t.value,
              name: t.name
            },
            bubbles: !0,
            composed: !0
          }
        );
        t.dispatchEvent(p), t.onChange && t.onChange(p);
      }
    }
  }
  getRadiosInGroup() {
    if (!this.name) return [];
    let r = this.getRootNode();
    for (; r && "host" in r; ) {
      const e = r.host?.getRootNode();
      if (e && e !== r)
        r = e;
      else
        break;
    }
    const s = (r instanceof Document ? r : document).querySelectorAll(`ag-radio[name="${this.name}"]`);
    return Array.from(s).filter((e) => e instanceof n && !e.disabled);
  }
  handleChange(r) {
    if (this.disabled) {
      r.preventDefault();
      return;
    }
    const d = r.target, s = this.checked;
    this.checked = d.checked, this.checked && !s && this.name && this.uncheckOtherRadiosInGroup();
    const e = new CustomEvent(
      "change",
      {
        detail: {
          checked: this.checked,
          value: this.value,
          name: this.name
        },
        bubbles: !0,
        composed: !0
      }
    );
    this.dispatchEvent(e), this.onChange && this.onChange(e);
  }
  uncheckOtherRadiosInGroup() {
    let r = this.getRootNode();
    for (; r && "host" in r; ) {
      const e = r.host?.getRootNode();
      if (e && e !== r)
        r = e;
      else
        break;
    }
    (r instanceof Document ? r : document).querySelectorAll(`ag-radio[name="${this.name}"]`).forEach((e) => {
      e !== this && e instanceof n && (e.checked = !1);
    });
  }
  /**
   * Render optional external label (for groups or above radio)
   */
  _renderExternalLabel() {
    return !this.label || this.noLabel ? l : c`
      <label
        id="${this._ids.labelId}"
        for="${this._ids.inputId}"
        class="ag-form-control__label ag-radio__external-label ${this.labelHidden ? "ag-form-control__label--hidden" : ""}"
        part="ag-radio-external-label"
      >
        ${this.label}
        ${this.required ? c`
              <span class="ag-form-control__required" part="ag-required" aria-hidden="true">*</span>
            ` : l}
      </label>
    `;
  }
  /**
   * Render helper text
   */
  _renderHelper() {
    return this.helpText ? c`
      <div
        id="${this._ids.helperId}"
        class="ag-form-control__helper ag-radio__helper"
        part="ag-radio-helper"
      >
        ${this.helpText}
      </div>
    ` : l;
  }
  /**
   * Render error text
   */
  _renderError() {
    return c`
      <div
        id="${this._ids.errorId}"
        class="ag-form-control__error ag-radio__error"
        part="ag-radio-error"
        ?hidden="${!this.invalid || !this.errorMessage}"
      >
        ${this.errorMessage || ""}
      </div>
    `;
  }
  /**
   * Build ARIA describedby attribute
   */
  _getAriaDescribedBy() {
    return w({
      helperId: this._ids.helperId,
      errorId: this._ids.errorId,
      hasHelper: !!this.helpText,
      hasError: this.invalid && !!this.errorMessage
    });
  }
  render() {
    const r = `
      radio-wrapper
      ${this.labelPosition === "start" ? "radio-wrapper--label-start" : ""}
    `, d = `
      radio-indicator
      radio-indicator--${this.size}
      radio-indicator--${this.theme}
    `, s = `
      radio-label
      radio-label--${this.size}
    `, e = this._getAriaDescribedBy(), t = c`
      <label class=${r.trim()} part="ag-radio-wrapper">
        <input
          type="radio"
          id="${this._ids.inputId}"
          class="radio-input"
          part="ag-radio-input"
          name=${this.name}
          value=${this.value}
          .checked=${this.checked}
          ?disabled=${this.disabled}
          ?required=${this.required}
          aria-required="${this.required ? "true" : "false"}"
          aria-invalid="${this.invalid ? "true" : "false"}"
          aria-describedby="${e || l}"
          @click=${this.handleClick}
          @change=${this.handleChange}
          @keydown=${this.handleKeyDown}
          aria-checked=${this.checked ? "true" : "false"}
        />
        <span class=${d.trim()} part="ag-radio-indicator"></span>
        <span class=${s.trim()} part="ag-radio-label">
          ${this.labelText ? this.labelText : c`<slot></slot>`}
        </span>
      </label>
    `;
    return this.label || this.helpText || this.errorMessage ? c`
        <div class="ag-radio-wrapper-with-external">
          ${this._renderExternalLabel()}
          ${t}
          ${this._renderHelper()}
          ${this._renderError()}
        </div>
      ` : t;
  }
}, n.styles = [
  v,
  b`
      :host {
        display: inline-block;
      }

    .radio-wrapper {
      display: inline-flex;
      align-items: center;
      cursor: pointer;
      user-select: none;
      gap: var(--ag-space-3);
    }

    :host([disabled]) .radio-wrapper {
      cursor: not-allowed;
      opacity: 0.6;
    }

    .radio-wrapper--label-start {
      flex-direction: row-reverse;
    }

    .radio-input {
      position: absolute;
      width: 1px;
      height: 1px;
      opacity: 0;
      margin: 0;
      padding: 0;
    }

    .radio-indicator {
      position: relative;
      display: inline-block;
      flex-shrink: 0;
      border-radius: var(--ag-radius-full);
      transition: all var(--ag-motion-fast) ease-in-out;
    }

    /* Size variants */
    .radio-indicator--small {
      width: 12px;
      height: 12px;
    }

    .radio-indicator--medium {
      width: 14px;
      height: 14px;
    }

    .radio-indicator--large {
      width: 16px;
      height: 16px;
    }

    /* Default theme - alias to primary */
    .radio-indicator--default {
      border: var(--ag-border-width-2) solid var(--ag-primary-border);
      background: var(--ag-white);
      box-shadow: 0 0 0 var(--ag-border-width-2) transparent;
    }

    .radio-input:checked + .radio-indicator--default {
      background: var(--ag-primary);
      border-color: var(--ag-primary);
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-white) inset;
    }

    .radio-input:focus + .radio-indicator--default {
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-primary-border),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2)) var(--ag-white),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2) + var(--ag-focus-offset)) rgba(var(--ag-focus), 0.5);
    }

    .radio-input:checked:focus + .radio-indicator--default {
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-white) inset,
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2)) var(--ag-white),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2) + var(--ag-focus-offset)) rgba(var(--ag-focus), 0.5);
    }

    /* Primary theme */
    .radio-indicator--primary {
      border: var(--ag-border-width-2) solid var(--ag-primary-border);
      background: var(--ag-white);
      box-shadow: 0 0 0 var(--ag-border-width-2) transparent;
    }

    .radio-input:checked + .radio-indicator--primary {
      background: var(--ag-primary);
      border-color: var(--ag-primary);
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-white) inset;
    }

    .radio-input:focus + .radio-indicator--primary {
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-primary-border),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2)) var(--ag-white),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2) + var(--ag-focus-offset)) rgba(var(--ag-focus), 0.5);
    }

    .radio-input:checked:focus + .radio-indicator--primary {
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-white) inset,
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2)) var(--ag-white),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2) + var(--ag-focus-offset)) rgba(var(--ag-focus), 0.5);
    }

    /* Success theme - green */
    .radio-indicator--success {
      border: var(--ag-border-width-2) solid var(--ag-border);
      background: var(--ag-white);
      box-shadow: 0 0 0 var(--ag-border-width-2) transparent;
    }

    .radio-input:checked + .radio-indicator--success {
      background: var(--ag-success);
      border-color: var(--ag-success);
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-white) inset;
    }

    .radio-input:focus + .radio-indicator--success {
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-border),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2)) var(--ag-white),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2) + var(--ag-focus-offset)) rgba(var(--ag-focus), 0.5);
    }

    .radio-input:checked:focus + .radio-indicator--success {
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-white) inset,
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2)) var(--ag-white),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2) + var(--ag-focus-offset)) rgba(var(--ag-focus), 0.5);
    }

    /* Monochrome theme */
    .radio-indicator--monochrome {
      border: var(--ag-border-width-2) solid var(--ag-black);
      background: var(--ag-white);
      box-shadow: 0 0 0 var(--ag-border-width-2) transparent;
    }

    .radio-input:checked + .radio-indicator--monochrome {
      background: var(--ag-black);
      border-color: var(--ag-black);
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-white) inset;
    }

    .radio-input:focus + .radio-indicator--monochrome {
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-black),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2)) var(--ag-white),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2) + var(--ag-focus-offset)) rgba(var(--ag-focus), 0.5);
    }

    .radio-input:checked:focus + .radio-indicator--monochrome {
      box-shadow: 0 0 0 var(--ag-border-width-2) var(--ag-white) inset,
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2)) var(--ag-white),
                  0 0 0 calc(var(--ag-focus-width) + var(--ag-border-width-2) + var(--ag-focus-offset)) rgba(var(--ag-focus), 0.5);
    }

    .radio-label {
      display: inline-block;
    }

    .radio-label--small {
      font-size: 0.875rem;
    }

    .radio-label--medium {
      font-size: 1rem;
    }

    .radio-label--large {
      font-size: 1.125rem;
    }

    /* Respect reduced motion preferences */
    @media (prefers-reduced-motion: reduce) {
      .radio-indicator {
        transition: none;
      }
    }

    /* High contrast mode support */
    @media (prefers-contrast: high) {
      .radio-indicator {
        outline: var(--ag-border-width-1) solid;
      }
    }
  `
], n);
o([
  i({ type: String, reflect: !0 })
], a.prototype, "name");
o([
  i({ type: String, reflect: !0 })
], a.prototype, "value");
o([
  i({ type: Boolean, reflect: !0 })
], a.prototype, "checked");
o([
  i({ type: Boolean, reflect: !0 })
], a.prototype, "disabled");
o([
  i({ type: String })
], a.prototype, "size");
o([
  i({ type: String })
], a.prototype, "theme");
o([
  i({ type: String })
], a.prototype, "labelText");
o([
  i({ type: String })
], a.prototype, "labelPosition");
o([
  i({ type: String })
], a.prototype, "label");
o([
  i({ type: Boolean, attribute: "label-hidden" })
], a.prototype, "labelHidden");
o([
  i({ type: Boolean, attribute: "no-label" })
], a.prototype, "noLabel");
o([
  i({ type: Boolean, reflect: !0 })
], a.prototype, "required");
o([
  i({ type: Boolean, reflect: !0 })
], a.prototype, "invalid");
o([
  i({ type: String, attribute: "error-message" })
], a.prototype, "errorMessage");
o([
  i({ type: String, attribute: "help-text" })
], a.prototype, "helpText");
o([
  i({ attribute: !1 })
], a.prototype, "onClick");
o([
  i({ attribute: !1 })
], a.prototype, "onChange");
let _ = a;
export {
  _ as AgRadio
};
