// agnosticui-core@2.0.0-alpha.12/dist/components/Fieldset/core/_Fieldset.js
import { LitElement as p, css as f, html as g } from "lit";
import { n as r } from "../../../property-CemaeiRl.js";
var c = Object.defineProperty, d = (n, a, i, h) => {
  for (var e = void 0, l = n.length - 1, s; l >= 0; l--)
    (s = n[l]) && (e = s(a, i, e) || e);
  return e && c(a, i, e), e;
};
const o = class o extends p {
  constructor() {
    super(...arguments), this.legend = "", this.bordered = !1, this.layout = "vertical", this.legendHidden = !1;
  }
  /**
   * Validate layout prop on connection
   */
  connectedCallback() {
    super.connectedCallback(), this.layout !== "vertical" && this.layout !== "horizontal" && (this.layout = "vertical");
  }
  render() {
    const a = [
      "ag-fieldset",
      this.bordered ? "ag-fieldset--bordered" : "",
      this.layout === "horizontal" ? "ag-fieldset--horizontal" : ""
    ].filter(Boolean).join(" "), i = [
      "ag-fieldset-legend",
      this.legendHidden ? "ag-fieldset-legend--hidden" : ""
    ].filter(Boolean).join(" ");
    return g`
      <fieldset
        class="${a}"
        part="ag-fieldset"
      >
        ${this.legend ? g`
          <legend class="${i}" part="ag-legend">
            ${this.legend}
          </legend>
        ` : ""}

        <div class="ag-fieldset-content" part="ag-content">
          <slot></slot>
        </div>
      </fieldset>
    `;
  }
};
o.styles = f`
    :host {
      display: block;
    }

    .ag-fieldset {
      border: 0;
      padding: 0;
      margin: 0;
      min-inline-size: auto;
    }

    .ag-fieldset--bordered {
      border: var(--ag-border-width-1) solid var(--ag-border);
      border-radius: var(--ag-radius-md);
      padding: var(--ag-fieldset-padding, var(--ag-space-4));
    }

    .ag-fieldset--horizontal .ag-fieldset-content {
      display: flex;
      flex-wrap: wrap;
      gap: var(--ag-fieldset-gap, var(--ag-space-4));
      align-items: flex-start;
    }

    .ag-fieldset-legend {
      font-weight: 600;
      font-size: var(--ag-font-size-base);
      color: var(--ag-text-primary);
      padding: 0;
    }

    .ag-fieldset-legend--hidden {
      position: absolute !important;
      width: 1px !important;
      height: 1px !important;
      padding: 0 !important;
      margin: -1px !important;
      overflow: hidden !important;
      clip: rect(0, 0, 0, 0) !important;
      white-space: nowrap !important;
      border: 0 !important;
    }

    .ag-fieldset-content {
      display: block;
    }
  `;
let t = o;
d([
  r({ type: String })
], t.prototype, "legend");
d([
  r({ type: Boolean, reflect: !0 })
], t.prototype, "bordered");
d([
  r({ type: String, reflect: !0 })
], t.prototype, "layout");
d([
  r({ type: Boolean, attribute: "legend-hidden" })
], t.prototype, "legendHidden");
export {
  t as AgFieldset
};
