// typed-htmx@0.3.1/dist/typed-html/jsx-runtime.js
"use strict";
/// <reference types="typed-html" />
/// <reference path="../jsx.d.ts" />
Object.defineProperty(exports, "__esModule", { value: true });
exports.jsxs = exports.jsx = exports.Fragment = exports.Node = void 0;
const typed_html_1 = require("typed-html");
const index_js_1 = require("../index.js");
class Node {
    constructor(children) {
        this.children = children;
    }
    toString() {
        if (Array.isArray(this.children))
            return this.children.join("\n");
        return this.children.toString();
    }
}
exports.Node = Node;
function Fragment({ children }) {
    if (Array.isArray(children)) {
        const elts = children.map(sanitizer);
        return index_js_1.jsxConfig.trusted ? elts.join("\n") : new Node(elts);
    }
    const elt = sanitizer(children);
    return index_js_1.jsxConfig.trusted ? elt : new Node(elt);
}
exports.Fragment = Fragment;
function sanitizer(value) {
    const str = value || value === 0 ? value.toString() : "";
    if (!index_js_1.jsxConfig.sanitize || index_js_1.jsxConfig.trusted)
        return str;
    if (value instanceof Node)
        return value;
    return index_js_1.jsxConfig.sanitize(str, typeof value);
}
function expandLiterals(props) {
    for (const attr of index_js_1.jsxConfig.jsonAttributes) {
        if (!(attr in props))
            continue;
        const value = props[attr];
        if (typeof value === "object") {
            props[attr] = { toString: () => JSON.stringify(value) };
        }
    }
}
function jsx(tag, { children, ...props }) {
    expandLiterals(props);
    const contents = Array.isArray(children) ? children.map(sanitizer) : [sanitizer(children)];
    const elt = (0, typed_html_1.createElement)(tag, props, ...contents);
    return index_js_1.jsxConfig.trusted ? elt : new Node(elt);
}
exports.jsx = jsx;
function jsxs(tag, { children, ...props }) {
    expandLiterals(props);
    const elt = (0, typed_html_1.createElement)(tag, props, ...children.map(sanitizer));
    return index_js_1.jsxConfig.trusted ? elt : new Node(elt);
}
exports.jsxs = jsxs;
