// agnosticui-core@2.0.0-alpha.12/dist/components/AspectRatio/core/_AspectRatio.js
import { LitElement as p, css as c, html as m } from "lit";
import { n as r } from "../../../property-CemaeiRl.js";
var u = Object.defineProperty, h = (n, e, s, l) => {
  for (var t = void 0, o = n.length - 1, d; o >= 0; o--)
    (d = n[o]) && (t = d(e, s, t) || t);
  return t && u(e, s, t), t;
};
const a = class a extends p {
  constructor() {
    super(), this.width = 1, this.height = 1, this.maxWidth = void 0;
  }
  get _aspectStyle() {
    const e = this.width || 1, s = this.height || 1, l = `${e} / ${s}`, t = [];
    return t.push(`aspect-ratio: ${l};`), this.maxWidth !== void 0 && this.maxWidth !== null && (t.push(`max-inline-size: ${this.maxWidth}px;`), t.push("margin-inline-start: auto; margin-inline-end: auto;")), t.join(" ");
  }
  render() {
    return m`
      <div part="ag-aspect-ratio" class="ratio" style=${this._aspectStyle}>
        <slot></slot>
      </div>
    `;
  }
};
a.styles = c`
    :host {
      display: block;
      width: 100%;
    }

    /* wrapper that enforces aspect ratio */
    .ratio {
      position: relative;
      width: 100%;
      height: auto;
      overflow: hidden;
      display: block;
    }

    /* Make slotted child fill the container */
    ::slotted(*) {
      position: absolute;
      inset: 0;
      width: 100%;
      height: 100%;
      object-fit: cover;
      border: 0;
    }
  `;
let i = a;
h([
  r({ type: Number })
], i.prototype, "width");
h([
  r({ type: Number })
], i.prototype, "height");
h([
  r({ type: Number, attribute: "max-width" })
], i.prototype, "maxWidth");
export {
  i as AgAspectRatio
};
