// agnosticui-core@2.0.0-alpha.12/dist/components/Popover/vue/VuePopover.js
import { defineComponent as i, ref as d, onMounted as l, nextTick as c, watchEffect as u, createElementBlock as g, openBlock as p, mergeProps as h, renderSlot as m } from "vue";
import "../core/Popover.js";
import { _ as f } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const w = i({
  name: "VuePopover",
  props: {
    placement: {
      type: String,
      default: "bottom"
    },
    distance: {
      type: Number,
      default: 8
    },
    skidding: {
      type: Number,
      default: 0
    },
    arrow: {
      type: Boolean,
      default: !0
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    triggerType: {
      type: String,
      default: "click"
    },
    matchTriggerWidth: {
      type: Boolean,
      default: !1
    },
    showCloseButton: {
      type: Boolean,
      default: !0
    },
    closeLabel: {
      type: String,
      default: "Close popover"
    },
    trapFocus: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["show", "hide"],
  setup(e, { emit: a }) {
    const o = d(null), n = () => {
      const t = o.value;
      t && (t.placement = e.placement, t.distance = e.distance, t.skidding = e.skidding, t.arrow = e.arrow, t.disabled = e.disabled, t.triggerType = e.triggerType, t.matchTriggerWidth = e.matchTriggerWidth, t.showCloseButton = e.showCloseButton, t.closeLabel = e.closeLabel, t.trapFocus = e.trapFocus);
    }, r = (t) => {
      a("show", t);
    }, s = (t) => {
      a("hide", t);
    };
    return l(async () => {
      await customElements.whenDefined("ag-popover"), await c(), n();
      const t = o.value;
      t && (t.addEventListener("show", r), t.addEventListener("hide", s));
    }), u(() => {
      o.value && n();
    }), { agComponent: o };
  }
}), y = [".arrow", ".disabled", ".matchTriggerWidth", ".showCloseButton", ".trapFocus", "placement", "distance", "skidding", "triggerType", "closeLabel"];
function b(e, a, o, n, r, s) {
  return p(), g("ag-popover", h({
    ref: "agComponent",
    ".arrow": e.arrow,
    ".disabled": e.disabled,
    ".matchTriggerWidth": e.matchTriggerWidth,
    ".showCloseButton": e.showCloseButton,
    ".trapFocus": e.trapFocus,
    placement: e.placement,
    distance: e.distance,
    skidding: e.skidding,
    triggerType: e.triggerType,
    closeLabel: e.closeLabel
  }, e.$attrs), [
    m(e.$slots, "default")
  ], 48, y);
}
const k = /* @__PURE__ */ f(w, [["render", b]]);
export {
  k as default
};
