// agnosticui-core@2.0.0-alpha.12/dist/components/IntlFormatter/core/_IntlFormatter.js
import { LitElement as p, html as c } from "lit";
import { n as t } from "../../../property-CemaeiRl.js";
var y = Object.defineProperty, e = (h, i, o, u) => {
  for (var a = void 0, n = h.length - 1, s; n >= 0; n--)
    (s = h[n]) && (a = s(i, o, a) || a);
  return a && y(i, o, a), a;
};
class r extends p {
  constructor() {
    super(), this._formatterCache = /* @__PURE__ */ new Map(), this.type = "date", this.hourFormat = "auto", this.noGrouping = !1, this.currency = "USD", this.currencyDisplay = "symbol";
  }
  /**
   * Get the locale to use for formatting
   */
  _getLocale() {
    return this.lang || navigator.language;
  }
  /**
   * Generate a cache key based on formatter type and options
   */
  _getCacheKey() {
    const i = [this.type, this._getLocale()];
    return this.type === "date" ? i.push(
      this.dateStyle || "",
      this.timeStyle || "",
      this.weekday || "",
      this.era || "",
      this.year || "",
      this.month || "",
      this.day || "",
      this.hour || "",
      this.minute || "",
      this.second || "",
      this.timeZoneName || "",
      this.timeZone || "",
      this.hourFormat || ""
    ) : i.push(
      String(this.noGrouping),
      this.currency || "",
      this.currencyDisplay || "",
      String(this.minimumIntegerDigits || ""),
      String(this.minimumFractionDigits || ""),
      String(this.maximumFractionDigits || ""),
      String(this.minimumSignificantDigits || ""),
      String(this.maximumSignificantDigits || "")
    ), i.join("|");
  }
  /**
   * Dispatch a format error event
   */
  _dispatchError(i, o) {
    this.dispatchEvent(new CustomEvent("format-error", {
      detail: { type: i, error: o.message },
      bubbles: !0,
      composed: !0
    }));
  }
  /**
   * Validate component properties based on type
   */
  _validate() {
    if (this.type === "currency" && !this.currency)
      return { valid: !1, error: "Currency type requires a currency code" };
    if (this.type === "date") {
      if (!(this.date || this.value))
        return { valid: !1, error: "Date type requires a date value" };
    } else if (this.value === void 0 || this.value === null)
      return { valid: !1, error: `${this.type} type requires a value` };
    return { valid: !0 };
  }
  /**
   * Format a date value using Intl.DateTimeFormat
   */
  _formatDate() {
    const i = this._validate();
    if (!i.valid)
      return this._dispatchError("date", new Error(i.error)), "";
    const o = this.date || this.value, u = new Date(o);
    if (isNaN(u.getTime()))
      return this._dispatchError("date", new Error("Invalid date")), "";
    const a = this.hourFormat === "auto" ? void 0 : this.hourFormat === "12", n = {
      dateStyle: this.dateStyle,
      timeStyle: this.timeStyle,
      weekday: this.weekday,
      era: this.era,
      year: this.year,
      month: this.month,
      day: this.day,
      hour: this.hour,
      minute: this.minute,
      second: this.second,
      timeZoneName: this.timeZoneName,
      timeZone: this.timeZone,
      hour12: a
    };
    Object.keys(n).forEach((s) => {
      n[s] === void 0 && delete n[s];
    });
    try {
      const s = this._getCacheKey();
      let m = this._formatterCache.get(s);
      return m || (m = new Intl.DateTimeFormat(this._getLocale(), n), this._formatterCache.set(s, m)), m.format(u);
    } catch (s) {
      return this._dispatchError("date", s), "";
    }
  }
  /**
   * Format a number value using Intl.NumberFormat
   */
  _formatNumber() {
    const i = this._validate();
    if (!i.valid)
      return this._dispatchError("number", new Error(i.error)), "";
    const o = typeof this.value == "string" ? parseFloat(this.value) : Number(this.value);
    if (isNaN(o))
      return this._dispatchError("number", new Error("Invalid number")), "";
    const a = {
      style: this.type === "percent" ? "percent" : this.type === "currency" ? "currency" : "decimal",
      currency: this.currency,
      currencyDisplay: this.currencyDisplay,
      useGrouping: !this.noGrouping,
      minimumIntegerDigits: this.minimumIntegerDigits,
      minimumFractionDigits: this.minimumFractionDigits,
      maximumFractionDigits: this.maximumFractionDigits,
      minimumSignificantDigits: this.minimumSignificantDigits,
      maximumSignificantDigits: this.maximumSignificantDigits
    };
    Object.keys(a).forEach((n) => {
      a[n] === void 0 && delete a[n];
    });
    try {
      const n = this._getCacheKey();
      let s = this._formatterCache.get(n);
      return s || (s = new Intl.NumberFormat(this._getLocale(), a), this._formatterCache.set(n, s)), s.format(o);
    } catch (n) {
      return this._dispatchError("number", n), "";
    }
  }
  /**
   * Render the formatted value
   */
  render() {
    if (!this._validate().valid)
      return c`<span part="error"></span>`;
    let o = "";
    if (this.type === "date") {
      o = this._formatDate();
      const u = this.date || this.value, a = new Date(u);
      if (!isNaN(a.getTime()))
        return c`<time part="date-time" datetime="${a.toISOString()}">${o}</time>`;
    } else
      return o = this._formatNumber(), c`<span part="${this.type}">${o}</span>`;
    return c`${o}`;
  }
  /**
   * Clean up formatter cache when component is removed
   */
  disconnectedCallback() {
    super.disconnectedCallback(), this._formatterCache.clear();
  }
}
e([
  t({ type: String, reflect: !0 })
], r.prototype, "type");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "value");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "lang");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "date");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "weekday");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "era");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "year");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "month");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "day");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "hour");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "minute");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "second");
e([
  t({ type: String, reflect: !0, attribute: "time-zone-name" })
], r.prototype, "timeZoneName");
e([
  t({ type: String, reflect: !0, attribute: "time-zone" })
], r.prototype, "timeZone");
e([
  t({ type: String, reflect: !0, attribute: "hour-format" })
], r.prototype, "hourFormat");
e([
  t({ type: String, reflect: !0, attribute: "date-style" })
], r.prototype, "dateStyle");
e([
  t({ type: String, reflect: !0, attribute: "time-style" })
], r.prototype, "timeStyle");
e([
  t({ type: Boolean, reflect: !0, attribute: "no-grouping" })
], r.prototype, "noGrouping");
e([
  t({ type: String, reflect: !0 })
], r.prototype, "currency");
e([
  t({ type: String, reflect: !0, attribute: "currency-display" })
], r.prototype, "currencyDisplay");
e([
  t({ type: Number, reflect: !0, attribute: "minimum-integer-digits" })
], r.prototype, "minimumIntegerDigits");
e([
  t({ type: Number, reflect: !0, attribute: "minimum-fraction-digits" })
], r.prototype, "minimumFractionDigits");
e([
  t({ type: Number, reflect: !0, attribute: "maximum-fraction-digits" })
], r.prototype, "maximumFractionDigits");
e([
  t({ type: Number, reflect: !0, attribute: "minimum-significant-digits" })
], r.prototype, "minimumSignificantDigits");
e([
  t({ type: Number, reflect: !0, attribute: "maximum-significant-digits" })
], r.prototype, "maximumSignificantDigits");
export {
  r as IntlFormatter
};
