// agnosticui-core@2.0.0-alpha.12/dist/components/Image/core/_Image.js
import { LitElement as g, css as y, nothing as a, html as o } from "lit";
import { n as r } from "../../../property-CemaeiRl.js";
import { r as n } from "../../../state-CovhUvdr.js";
var u = Object.defineProperty, i = (p, t, l, f) => {
  for (var s = void 0, h = p.length - 1, c; h >= 0; h--)
    (c = p[h]) && (s = c(t, l, s) || s);
  return s && u(t, l, s), s;
};
const d = class d extends g {
  // ──────────────────────────────────────────────────────────────
  // Constructor
  // ──────────────────────────────────────────────────────────────
  constructor() {
    super(), this._handleLoad = () => {
      this._loaded = !0, this._errored = !1, this.dispatchEvent(
        new CustomEvent("ag-load", {
          bubbles: !0,
          composed: !0
        })
      );
    }, this._handleError = () => {
      if (this._loaded = !1, this._errored = !0, this.fallbackSrc && this._currentSrc !== this.fallbackSrc) {
        this._currentSrc = this.fallbackSrc;
        return;
      }
      this.dispatchEvent(
        new CustomEvent("ag-error", {
          bubbles: !0,
          composed: !0
        })
      );
    }, this.src = "", this.alt = "", this.sources = [], this.fit = "cover", this.position = "center", this.loading = "lazy", this.fade = !1, this.duration = 200, this._loaded = !1, this._errored = !1, this._currentSrc = "";
  }
  // ──────────────────────────────────────────────────────────────
  // Lifecycle
  // ──────────────────────────────────────────────────────────────
  willUpdate(t) {
    super.willUpdate(t), t.has("fit") && this.style.setProperty("--ag-image-fit", this.fit), t.has("position") && this.style.setProperty("--ag-image-position", this.position), t.has("width") && (this.style.width = this.width != null ? `${this.width}px` : ""), t.has("height") && (this.style.height = this.height != null ? `${this.height}px` : ""), (t.has("aspectRatio") || t.has("width") || t.has("height")) && (this.aspectRatio ? this.style.aspectRatio = this.aspectRatio : this.width != null && this.height != null ? this.style.aspectRatio = `${this.width} / ${this.height}` : this.style.aspectRatio = ""), t.has("src") && t.get("src") !== this.src && (this._loaded = !1, this._errored = !1, this._currentSrc = this.src);
  }
  // ──────────────────────────────────────────────────────────────
  // Render Helpers
  // ──────────────────────────────────────────────────────────────
  _renderImage() {
    const t = this._errored && this.fallbackSrc ? this.fallbackSrc : this._currentSrc;
    return o`
      <img
        part="ag-img"
        class=${this._loaded ? "loaded" : ""}
        src=${t}
        alt=${this.alt}
        loading=${this.loading}
        decoding="async"
        width=${this.width ?? a}
        height=${this.height ?? a}
        @load=${this._handleLoad}
        @error=${this._handleError}
      />
    `;
  }
  _renderPicture() {
    return !this.sources || this.sources.length === 0 ? this._renderImage() : o`
      <picture>
        ${this.sources.map(
      (t) => o`
            <source
              srcset=${t.srcset}
              type=${t.type ?? a}
              media=${t.media ?? a}
              sizes=${t.sizes ?? a}
            />
          `
    )}
        ${this._renderImage()}
      </picture>
    `;
  }
  _renderOverlay() {
    return this._loaded ? a : this._errored ? o`
        <div class="overlay error" part="ag-error" role="alert" aria-label="Image failed to load">
          <slot name="error">Failed to load image</slot>
        </div>
      ` : o`
      <div class="overlay" part="ag-placeholder" aria-label="Image loading">
        <slot name="placeholder">Loading…</slot>
      </div>
    `;
  }
  // ──────────────────────────────────────────────────────────────
  // Main Render
  // ──────────────────────────────────────────────────────────────
  render() {
    return o`
      <div class="wrapper">
        ${this._renderPicture()}
        ${this._renderOverlay()}
      </div>
      <slot></slot>
    `;
  }
};
d.styles = y`
    :host {
      display: inline-block;
      position: relative;
      overflow: hidden;
      /* Remove inline spacing around image element - images are inline by default and can create unwanted gaps. This doesn't affect slotted content. */
      line-height: 0;
      background: var(--ag-image-bg-color, var(--ag-background-secondary));
    }

    .wrapper {
      display: block;
      width: 100%;
      height: 100%;
      position: relative;
    }

    img {
      display: block;
      width: 100%;
      height: 100%;
      object-fit: var(--ag-image-fit, cover);
      object-position: var(--ag-image-position, center);
    }

    /* Fade transition when enabled */
    :host([fade]) img {
      opacity: 0;
      transition: opacity var(--ag-image-transition-duration, var(--ag-motion-medium, 200ms)) ease;
    }

    :host([fade]) img.loaded {
      opacity: 1;
    }

    /* Non-fade components are always visible */
    :host(:not([fade])) img {
      opacity: 1;
    }

    /* Overlay styles for placeholder and error states */
    .overlay {
      position: absolute;
      inset: 0;
      display: flex;
      align-items: center;
      justify-content: center;
      font-size: var(--ag-font-size-sm);
      background: var(--ag-image-bg-color, var(--ag-background-secondary));
      color: var(--ag-text-muted);
      z-index: 1;
      /* Reset line-height for slotted content so text displays properly. The line-height: 0 on :host would otherwise be inherited here. */line-height: normal;
    }

    .overlay.error {
      background: var(--ag-danger-background);
      color: var(--ag-danger-text);
    }

    /* Hide overlay when image is loaded */
    .overlay.hidden {
      display: none;
    }
  `;
let e = d;
i([
  r({ type: String })
], e.prototype, "src");
i([
  r({ type: String })
], e.prototype, "alt");
i([
  r({ type: Array })
], e.prototype, "sources");
i([
  r({ type: Number })
], e.prototype, "width");
i([
  r({ type: Number })
], e.prototype, "height");
i([
  r({ type: String, attribute: "aspect-ratio" })
], e.prototype, "aspectRatio");
i([
  r({ type: String })
], e.prototype, "fit");
i([
  r({ type: String })
], e.prototype, "position");
i([
  r({ type: String })
], e.prototype, "loading");
i([
  r({ type: Boolean, reflect: !0 })
], e.prototype, "fade");
i([
  r({ type: Number })
], e.prototype, "duration");
i([
  r({ type: String, attribute: "fallback-src" })
], e.prototype, "fallbackSrc");
i([
  n()
], e.prototype, "_loaded");
i([
  n()
], e.prototype, "_errored");
i([
  n()
], e.prototype, "_currentSrc");
export {
  e as AgImage
};
