// agnosticui-core@2.0.0-alpha.12/dist/components/Divider/core/_Divider.js
import { LitElement as l, css as c, html as h } from "lit";
import { n as t } from "../../../property-CemaeiRl.js";
import { hasSlotContent as g } from "../../../utils/slot.js";
var f = Object.defineProperty, a = (s, e, d, p) => {
  for (var i = void 0, o = s.length - 1, v; o >= 0; o--)
    (v = s[o]) && (i = v(e, d, i) || i);
  return i && f(e, d, i), i;
};
const n = class n extends l {
  constructor() {
    super(), this.vertical = !1, this.justify = "center", this.size = "default", this.variant = "default", this.hasContent = !1;
  }
  handleSlotChange(e) {
    this.hasContent = g(e.target);
  }
  getClasses() {
    const e = ["divider"];
    return this.vertical && e.push("divider-vertical"), this.justify !== "center" && !this.vertical && e.push(`divider-justify-${this.justify}`), this.size !== "default" && e.push(`divider-${this.size}`), this.variant !== "default" && e.push(`divider-${this.variant}`), e.join(" ");
  }
  render() {
    const e = this.vertical ? "vertical" : "horizontal", d = this.hasContent ? "divider-content divider-has-content" : "divider-content";
    return h`
      <div
        class="${this.getClasses()}"
        part="ag-divider"
        role="separator"
        aria-orientation="${e}"
      >
        <span class="${d}" part="ag-divider-content">
          <slot @slotchange=${this.handleSlotChange}></slot>
        </span>
      </div>
    `;
  }
};
n.styles = c`
    :host {
      display: block;
    }

    :host([is-vertical]) {
      align-self: stretch;
    }

    .divider {
      display: flex;
      justify-content: center;
      align-items: center;
      white-space: nowrap;
      width: 100%;
    }

    .divider::before,
    .divider::after {
      content: '';
      background-color: var(--ag-border);
      height: var(--ag-border-width-2);
      flex-grow: 1;
    }

    /* Sizes - Horizontal */
    .divider-small::before,
    .divider-small::after {
      height: var(--ag-border-width-1);
    }

    .divider-large::before,
    .divider-large::after {
      height: var(--ag-border-width-3);
    }

    .divider-xlarge::before,
    .divider-xlarge::after {
      height: 4px;
    }

    /* Justify */
    .divider-justify-end::after,
    .divider-justify-start::before {
      flex-grow: 0;
      flex-basis: 3%;
    }

    /* Content */
    .divider-has-content {
      padding-inline-start: var(--ag-space-4);
      padding-inline-end: var(--ag-space-4);
    }

    /* Vertical */
    .divider-vertical {
      height: 100%;
      margin: 0 var(--ag-space-4);
      width: var(--ag-space-4);
      flex-direction: column;
    }

    .divider-vertical::before,
    .divider-vertical::after {
      width: var(--ag-border-width-2);
      height: auto;  /* Override the fixed height for vertical growth */
    }

    /* Sizes - Vertical */
    .divider-vertical.divider-small::before,
    .divider-vertical.divider-small::after {
      width: var(--ag-border-width-1);
    }

    .divider-vertical.divider-large::before,
    .divider-vertical.divider-large::after {
      width: var(--ag-border-width-3);
    }

    .divider-vertical.divider-xlarge::before,
    .divider-vertical.divider-xlarge::after {
      width: 4px;
    }

    .divider-vertical .divider-has-content {
      padding-inline-start: var(--ag-space-6);
      padding-inline-end: var(--ag-space-6);
      padding-block-start: var(--ag-space-3);
      padding-block-end: var(--ag-space-3);
    }

    /* Variants */
    .divider-warning::before,
    .divider-warning::after {
      background-color: var(--ag-warning);
    }

    .divider-warning .divider-content {
      color: var(--ag-text-primary);
    }

    .divider-error::before,
    .divider-error::after {
      background-color: var(--ag-danger);
    }

    .divider-error .divider-content {
      color: var(--ag-text-primary);
    }

    .divider-success::before,
    .divider-success::after {
      background-color: var(--ag-success);
    }

    .divider-success .divider-content {
      color: var(--ag-text-primary);
    }

    .divider-info::before,
    .divider-info::after {
      background-color: var(--ag-info);
    }

    .divider-info .divider-content {
      color: var(--ag-text-primary);
    }

    .divider-monochrome::before,
    .divider-monochrome::after {
      background-color: var(--ag-text-primary);
    }

    .divider-monochrome .divider-content {
      color: var(--ag-text-primary);
    }
  `;
let r = n;
a([
  t({ type: Boolean, attribute: "is-vertical", reflect: !0 })
], r.prototype, "vertical");
a([
  t({ type: String, reflect: !0 })
], r.prototype, "justify");
a([
  t({ type: String, reflect: !0 })
], r.prototype, "size");
a([
  t({ type: String, reflect: !0 })
], r.prototype, "variant");
a([
  t({ type: Boolean, state: !0 })
], r.prototype, "hasContent");
export {
  r as AgDivider
};
