// agnosticui-core@2.0.0-alpha.12/dist/components/SkeletonLoader/core/_SkeletonLoader.js
import { LitElement as c, css as l, html as u } from "lit";
import { n as r } from "../../../property-CemaeiRl.js";
var f = Object.defineProperty, s = (n, t, o, p) => {
  for (var i = void 0, a = n.length - 1, d; a >= 0; a--)
    (d = n[a]) && (i = d(t, o, i) || i);
  return i && f(t, o, i), i;
};
const h = class h extends c {
  constructor() {
    super(), this.variant = "text", this.effect = "pulse", this.intensity = "light";
  }
  connectedCallback() {
    super.connectedCallback(), this.hasAttribute("aria-hidden") || this.setAttribute("aria-hidden", "true");
  }
  updated(t) {
    super.updated(t), (t.has("width") || t.has("height") || t.has("variant")) && this._updateDimensions();
  }
  _updateDimensions() {
    this.variant === "circular" ? this.width && !this.height ? (this.style.height = this.width, this.style.width = this.width) : this.height && !this.width ? (this.style.width = this.height, this.style.height = this.height) : this.width && this.height && (this.style.width = this.width, this.style.height = this.height) : (this.width && (this.style.width = this.width), this.height && (this.style.height = this.height));
  }
  render() {
    return u`<div class="skeleton-wrapper" part="ag-skeleton-wrapper"></div>`;
  }
};
h.styles = l`
    :host {
      display: block;
      background-color: var(--ag-background-secondary);
      position: relative;
      overflow: hidden;
      width: 100%;
      height: 1em;
    }

    /* Intensity variants */
    :host([intensity='light']) {
      background-color: var(--ag-background-secondary);
    }

    :host([intensity='medium']) {
      background-color: var(--ag-background-tertiary);
    }

    /* Variants */
    :host([variant='text']) {
      border-radius: var(--ag-radius-sm);
    }

    :host([variant='circular']) {
      border-radius: 50%;
    }

    :host([variant='rectangular']) {
      border-radius: 0;
    }

    :host([variant='rounded']) {
      border-radius: var(--ag-radius-md);
    }

    /* Effects */
    @keyframes pulse {
      0%, 100% {
        opacity: 1;
      }
      50% {
        opacity: 0.5;
      }
    }

    :host([effect='pulse']) {
      animation: pulse 1.5s infinite ease-in-out;
    }

    :host([effect='sheen'])::after {
      content: "";
      position: absolute;
      inset: 0;
      transform: translateX(-100%);
      background-image: linear-gradient(
        90deg,
        transparent 0%,
        var(--ag-background-primary) 20%,
        var(--ag-background-primary) 60%,
        transparent 100%
      );
      opacity: 0.3;
      animation: sheen 1.5s infinite;
    }

    @keyframes sheen {
      100% {
        transform: translateX(100%);
      }
    }

    :host([effect='none']) {
      animation: none;
    }

    /* Respect prefers-reduced-motion */
    @media (prefers-reduced-motion: reduce) {
      :host([effect='pulse']),
      :host([effect='sheen'])::after {
        animation: none;
      }
    }
  `;
let e = h;
s([
  r({ type: String, reflect: !0 })
], e.prototype, "variant");
s([
  r({ type: String, reflect: !0 })
], e.prototype, "effect");
s([
  r({ type: String, reflect: !0 })
], e.prototype, "intensity");
s([
  r({ type: String, reflect: !0 })
], e.prototype, "width");
s([
  r({ type: String, reflect: !0 })
], e.prototype, "height");
export {
  e as AgSkeleton
};
