// agnosticui-core@2.0.0-alpha.12/dist/components/Sidebar/vue/VueSidebar.js
import { defineComponent as i, createElementBlock as p, openBlock as d, renderSlot as a } from "vue";
import "../core/Sidebar.js";
import { _ as r } from "../../../_plugin-vue_export-helper-CHgC5LLL.js";
const g = i({
  name: "VueSidebar",
  props: {
    open: { type: Boolean, default: !1 },
    collapsed: { type: Boolean, default: !1 },
    position: { type: String, default: "left" },
    ariaLabel: { type: String, default: "Navigation" },
    variant: {
      type: String,
      default: "default"
    },
    noTransition: { type: Boolean, default: !1 },
    width: { type: String, default: void 0 },
    disableCompactMode: { type: Boolean, default: !1 },
    showMobileToggle: { type: Boolean, default: !0 },
    mobileTogglePosition: {
      type: String,
      default: "top-left"
    },
    showHeaderToggle: { type: Boolean, default: !1 }
  },
  emits: ["update:open", "update:collapsed"],
  setup(e, { emit: o }) {
    return {
      onToggle: (l) => {
        o("update:open", l.detail.open);
      },
      onCollapse: (l) => {
        o("update:collapsed", l.detail.collapsed);
      }
    };
  }
}), f = [".open", ".collapsed", "position", "aria-label", "variant", ".noTransition", "width", ".disableCompactMode", ".showMobileToggle", "mobile-toggle-position", ".showHeaderToggle"];
function u(e, o, n, s, l, b) {
  return d(), p("ag-sidebar", {
    ".open": e.open,
    ".collapsed": e.collapsed,
    position: e.position,
    "aria-label": e.ariaLabel,
    variant: e.variant,
    ".noTransition": e.noTransition,
    width: e.width,
    ".disableCompactMode": e.disableCompactMode,
    ".showMobileToggle": e.showMobileToggle,
    "mobile-toggle-position": e.mobileTogglePosition,
    ".showHeaderToggle": e.showHeaderToggle,
    onAgSidebarToggle: o[0] || (o[0] = (...t) => e.onToggle && e.onToggle(...t)),
    onAgSidebarCollapse: o[1] || (o[1] = (...t) => e.onCollapse && e.onCollapse(...t))
  }, [
    a(e.$slots, "default"),
    a(e.$slots, "ag-header"),
    a(e.$slots, "ag-header-start"),
    a(e.$slots, "ag-header-end"),
    a(e.$slots, "ag-header-toggle"),
    a(e.$slots, "ag-toggle-icon"),
    a(e.$slots, "ag-footer")
  ], 40, f);
}
const w = /* @__PURE__ */ r(g, [["render", u]]);
export {
  w as default
};
